--- a/Code/gml_Object_obj_lightemup_controller_Create_0.gml
+++ b/Code/gml_Object_obj_lightemup_controller_Create_0.gml
@@ -37,25 +37,47 @@
 cursor_xstart = cursor_x;
 cursor_ystart = cursor_y;
 shooting_enabled = true;
+charnames = ["", "kris", "susie", "ralsei", "noelle"];
+charsprites[0] = [4131, 4131, 2051, 821, 4994];
+charsprites[1] = [147, 147, 4260, 3911, 4995];
+charsprites[2] = [2306, 2306, 4770, 239, 4993];
+charsprites[3] = [2489, 2489, 4226, 4119, 1619];
+charsprites[4] = [4898, 4898, 3151, 1776, 1619];
 with (obj_tenna_enemy)
 {
     var xx = camerax();
     var yy = cameray();
-    obj_herokris.x -= 8;
-    obj_herokris.y -= 12;
-    obj_herosusie.x -= 8;
-    obj_herosusie.y -= 6;
-    obj_heroralsei.x -= 4;
-    obj_herosusie.x = xx + 80;
-    obj_herosusie.y = yy + 92;
-    obj_herokris.x = xx + 116;
-    obj_herokris.y = yy + 132;
-    obj_heroralsei.x = xx + 64;
-    obj_heroralsei.y = yy + 145;
-    obj_herosusie.depth = obj_herokris.depth + 1;
-    scr_act_charsprite("ralsei", spr_ralsei_zoosuit, 0, 0);
-    scr_act_charsprite("kris", spr_kris_zoosuit, 0, 0);
-    scr_act_charsprite("susie", spr_susie_zoosuit, 0, 0);
+    if (instance_exists(global.charinstance[1]))
+    {
+        global.charinstance[1].x = xx + 80;
+        global.charinstance[1].y = yy + 92;
+        if (instance_exists(global.charinstance[0]))
+        {
+            global.charinstance[1].depth = global.charinstance[0].depth + 1;
+        }
+    }
+    if (instance_exists(global.charinstance[0]))
+    {
+        global.charinstance[0].x = xx + 116;
+        global.charinstance[0].y = yy + 132;
+    }
+    if (instance_exists(global.charinstance[2]))
+    {
+        global.charinstance[2].x = xx + 64;
+        global.charinstance[2].y = yy + 145;
+    }
+    if (global.char[2] > 0)
+    {
+        scr_act_charsprite(other.charnames[global.char[2]], other.charsprites[1][global.char[2]], 0, 0);
+    }
+    if (global.char[0] > 0)
+    {
+        scr_act_charsprite(other.charnames[global.char[0]], other.charsprites[1][global.char[0]], 0, 0);
+    }
+    if (global.char[1] > 0)
+    {
+        scr_act_charsprite(other.charnames[global.char[1]], other.charsprites[1][global.char[1]], 0, 0);
+    }
     with (obj_afterimage)
     {
         instance_destroy();
