--- a/Code/gml_GlobalScript_scr_endturn.gml
+++ b/Code/gml_GlobalScript_scr_endturn.gml
@@ -59,12 +59,12 @@
         }
     }
     __noactors = 1;
-    if (global.acting[0] == 1)
-    {
-        __noactors = 0;
-    }
     for (__noactorsi = 0; __noactorsi < 3; __noactorsi++)
     {
+        if (global.acting[__noactorsi] == 1)
+        {
+            __noactors = 0;
+        }
         if (global.actingsingle[__noactorsi] == 1)
         {
             __noactors = 0;
