--- a/Code/gml_GlobalScript_scr_monstersetup.gml
+++ b/Code/gml_GlobalScript_scr_monstersetup.gml
@@ -1488,7 +1488,7 @@
         global.canact[myself][2] = 1;
         global.actname[myself][2] = stringsetloc("Sharpshoot", "scr_monstersetup_slash_scr_monstersetup_gml_1885_0");
         global.actdesc[myself][2] = " ";
-        global.actactor[myself][2] = 4;
+        global.actactor[myself][2] = 6;
         global.actdesc[myself][2] = stringsetloc("Light#'em up", "scr_monstersetup_slash_scr_monstersetup_gml_1888_0");
         global.canactsus[myself][0] = 1;
         global.actnamesus[myself][0] = stringsetloc("S-Action", "scr_monstersetup_slash_scr_monstersetup_gml_1891_0");
@@ -1496,6 +1496,9 @@
         global.canactral[myself][0] = 1;
         global.actnameral[myself][0] = stringsetloc("R-Action", "scr_monstersetup_slash_scr_monstersetup_gml_1895_0");
         global.actsimulral[myself][0] = 1;
+        global.canactnoe[myself][0] = 1;
+        global.actnamenoe[myself][0] = stringset("N-Action");
+        global.actsimulnoe[myself][0] = 1;
     }
     if (global.monstertype[myself] == 55)
     {
@@ -1521,24 +1524,46 @@
         global.canact[myself][0] = 1;
         global.actname[myself][0] = stringsetloc("Check", "scr_monstersetup_slash_scr_monstersetup_gml_1913_0");
         global.actdesc[myself][0] = stringsetloc("Useless#analysis", "scr_monstersetup_slash_scr_monstersetup_gml_1877_0");
+        var actnum = 1;
+        if (scr_havechar(3))
+        {
-        global.canact[myself][1] = 1;
+            global.canact[myself][actnum] = 1;
-        global.actname[myself][1] = stringsetloc("EasyPhoto", "scr_monstersetup_slash_scr_monstersetup_gml_1916_0");
+            global.actname[myself][actnum] = stringsetloc("EasyPhoto", "scr_monstersetup_slash_scr_monstersetup_gml_1916_0");
-        global.actactor[myself][1] = 3;
+            global.actactor[myself][actnum] = 3;
-        global.actdesc[myself][1] = stringsetloc("Timely#photo", "scr_monstersetup_slash_scr_monstersetup_gml_1918_0");
+            global.actdesc[myself][actnum] = stringsetloc("Timely#photo", "scr_monstersetup_slash_scr_monstersetup_gml_1918_0");
+            actnum++;
+        }
-        global.canact[myself][2] = 1;
+        global.canact[myself][actnum] = 1;
-        global.actname[myself][2] = stringsetloc("ToughPhoto", "scr_monstersetup_slash_scr_monstersetup_gml_1921_0");
+        global.actname[myself][actnum] = stringsetloc("ToughPhoto", "scr_monstersetup_slash_scr_monstersetup_gml_1921_0");
-        global.actdesc[myself][2] = stringsetloc("Fast#photo", "scr_monstersetup_slash_scr_monstersetup_gml_1922_0");
+        global.actdesc[myself][actnum] = stringsetloc("Fast#photo", "scr_monstersetup_slash_scr_monstersetup_gml_1922_0");
+        actnum++;
+        if (scr_havechar(2))
+        {
-        global.canact[myself][3] = 1;
+            global.canact[myself][actnum] = 1;
-        global.actname[myself][3] = stringsetloc("PowerPhoto", "scr_monstersetup_slash_scr_monstersetup_gml_1930_0");
+            global.actname[myself][actnum] = stringsetloc("PowerPhoto", "scr_monstersetup_slash_scr_monstersetup_gml_1930_0");
-        global.actdesc[myself][3] = stringsetloc("Extra#Mercy", "scr_monstersetup_slash_scr_monstersetup_gml_1931_0");
+            global.actdesc[myself][actnum] = stringsetloc("Extra#Mercy", "scr_monstersetup_slash_scr_monstersetup_gml_1931_0");
-        global.actcost[myself][3] = 62.5;
+            global.actcost[myself][actnum] = 62.5;
-        global.actactor[myself][3] = 2;
+            global.actactor[myself][actnum] = 2;
+            actnum++;
+        }
+        if (scr_havechar(4))
+        {
+            global.canact[myself][actnum] = 1;
+            global.actname[myself][actnum] = stringset("ClassyPhoto");
+            global.actdesc[myself][actnum] = stringset("Nice#Photo");
+            global.actcost[myself][actnum] = 30;
+            global.actactor[myself][actnum] = 5;
+            actnum++;
+        }
         global.canactsus[myself][0] = 1;
         global.actnamesus[myself][0] = stringsetloc("S-Action", "scr_monstersetup_slash_scr_monstersetup_gml_1925_0");
         global.actsimulsus[myself][0] = 1;
         global.canactral[myself][0] = 1;
         global.actnameral[myself][0] = stringsetloc("R-Action", "scr_monstersetup_slash_scr_monstersetup_gml_1929_0");
         global.actsimulral[myself][0] = 1;
+        global.canactnoe[myself][0] = 1;
+        global.actnamenoe[myself][0] = stringset("N-Action");
+        global.actsimulnoe[myself][0] = 1;
     }
     if (global.monstertype[myself] == 56)
     {
@@ -1560,7 +1585,7 @@
         global.actdesc[myself][1] = stringsetloc("Bullets#give big#TP...", "scr_monstersetup_slash_scr_monstersetup_gml_1956_0");
         if (global.encounterno == 135)
         {
-            global.actsimul[myself][1] = 4;
+            global.actsimul[myself][1] = 7;
         }
         global.canact[myself][2] = 1;
         global.actname[myself][2] = stringsetloc("Mute", "scr_monstersetup_slash_scr_monstersetup_gml_1960_0");
@@ -1568,11 +1593,11 @@
         global.actdesc[myself][2] = stringsetloc("Tire#enemies", "scr_monstersetup_slash_scr_monstersetup_gml_1962_0");
         if (global.encounterno == 135)
         {
-            global.actsimul[myself][1] = 4;
+            global.actsimul[myself][1] = 7;
         }
         global.canact[myself][3] = 1;
         global.actname[myself][3] = stringsetloc("OffButton", "scr_monstersetup_slash_scr_monstersetup_gml_1966_0");
-        global.actactor[myself][3] = 4;
+        global.actactor[myself][3] = 6;
         global.actcost[myself][3] = 250;
         global.actdesc[myself][3] = stringsetloc("Turn#it off", "scr_monstersetup_slash_scr_monstersetup_gml_1969_0");
         global.canactsus[myself][0] = 1;
@@ -1581,6 +1606,9 @@
         global.canactral[myself][0] = 1;
         global.actnameral[myself][0] = stringsetloc("R-Action", "scr_monstersetup_slash_scr_monstersetup_gml_1976_0");
         global.actsimulral[myself][0] = 0;
+        global.canactnoe[myself][0] = 1;
+        global.actnamenoe[myself][0] = stringset("N-Action");
+        global.actsimulnoe[myself][0] = 1;
     }
     if (global.monstertype[myself] == 57)
     {
@@ -1631,10 +1659,10 @@
         global.canact[myself][1] = 1;
         global.actname[myself][1] = stringsetloc("BegForMercy", "scr_monstersetup_slash_scr_monstersetup_gml_2044_0_b");
         global.actdesc[myself][1] = stringsetloc("", "scr_monstersetup_slash_scr_monstersetup_gml_2045_0");
-        if (global.flag[1119] > 0)
+        if (global.flag[1119] > 0 && room != room_battletest)
         {
             global.canact[myself][2] = 1;
-            global.actactor[myself][2] = 4;
+            global.actactor[myself][2] = 6;
             global.actname[myself][2] = stringsetloc("ActCool", "scr_monstersetup_slash_scr_monstersetup_gml_2052_0");
             if (global.encounterno == 140)
             {
@@ -1654,6 +1682,10 @@
         global.actnameral[myself][0] = stringsetloc("R-Action", "scr_monstersetup_slash_scr_monstersetup_gml_2067_0");
         global.actdescral[myself][0] = stringsetloc("", "scr_monstersetup_slash_scr_monstersetup_gml_2068_0");
         global.actsimulral[myself][0] = 1;
+        global.canactnoe[myself][0] = 1;
+        global.actnamenoe[myself][0] = stringset("N-Action");
+        global.actdescnoe[myself][0] = stringset("");
+        global.actsimulnoe[myself][0] = 1;
     }
     if (global.monstertype[myself] == 59)
     {
@@ -1675,22 +1707,43 @@
         global.canact[myself][1] = 1;
         global.actdesc[myself][1] = stringsetloc("Touch#Green#4 Mercy", "scr_monstersetup_slash_scr_monstersetup_gml_265_0");
         global.actsimul[myself][1] = 1;
+        var actnum = 2;
+        if (scr_havechar(2))
+        {
-        global.actname[myself][2] = stringsetloc("Cheat", "scr_monstersetup_slash_scr_monstersetup_gml_274_0");
+            global.actname[myself][actnum] = stringsetloc("Cheat", "scr_monstersetup_slash_scr_monstersetup_gml_274_0");
-        global.canact[myself][2] = 1;
+            global.canact[myself][actnum] = 1;
-        global.actactor[myself][2] = 2;
+            global.actactor[myself][actnum] = 2;
-        global.actdesc[myself][2] = stringsetloc("TIRE#Enemies,#but...", "scr_monstersetup_slash_scr_monstersetup_gml_277_0");
+            global.actdesc[myself][actnum] = stringsetloc("TIRE#Enemies,#but...", "scr_monstersetup_slash_scr_monstersetup_gml_277_0");
-        global.actsimul[myself][2] = 0;
+            global.actsimul[myself][actnum] = 0;
+            actnum++;
+        }
+        if (scr_havechar(3))
+        {
-        global.actname[myself][3] = stringsetloc("Bribe", "scr_monstersetup_slash_scr_monstersetup_gml_280_0_b");
+            global.actname[myself][actnum] = stringsetloc("Bribe", "scr_monstersetup_slash_scr_monstersetup_gml_280_0_b");
-        global.canact[myself][3] = 1;
+            global.canact[myself][actnum] = 1;
-        global.actactor[myself][3] = 3;
+            global.actactor[myself][actnum] = 3;
-        global.actdesc[myself][3] = stringsetloc("Give 150#POINTs#to SPARE", "scr_monstersetup_slash_scr_monstersetup_gml_283_0");
+            global.actdesc[myself][actnum] = stringsetloc("Give 150#POINTs#to SPARE", "scr_monstersetup_slash_scr_monstersetup_gml_283_0");
-        global.actsimul[myself][3] = 0;
+            global.actsimul[myself][actnum] = 0;
+            actnum++;
+        }
+        if (scr_havechar(4))
+        {
+            global.actname[myself][actnum] = stringset("HighRoll");
+            global.canact[myself][actnum] = 1;
+            global.actactor[myself][actnum] = 5;
+            global.actdesc[myself][actnum] = stringset("50% MERCY#to all,#but...");
+            global.actsimul[myself][actnum] = 0;
+            actnum++;
+        }
         global.canactsus[myself][0] = 1;
         global.actnamesus[myself][0] = stringsetloc("S-Action", "scr_monstersetup_slash_scr_monstersetup_gml_287_0");
         global.actsimulsus[myself][0] = 1;
         global.canactral[myself][0] = 1;
         global.actnameral[myself][0] = stringsetloc("R-Action", "scr_monstersetup_slash_scr_monstersetup_gml_291_0_b");
         global.actsimulral[myself][0] = 1;
+        global.canactnoe[myself][0] = 1;
+        global.actnamenoe[myself][0] = stringset("N-Action");
+        global.actsimulnoe[myself][0] = 1;
     }
     if (global.monstertype[myself] == 60)
     {
@@ -1711,13 +1764,16 @@
         global.actname[myself][1] = stringsetloc("ILoveTV", "scr_monstersetup_slash_scr_monstersetup_gml_2033_0");
         global.canact[myself][2] = 1;
         global.actname[myself][2] = stringsetloc("ILoveTVX", "scr_monstersetup_slash_scr_monstersetup_gml_2036_0");
-        global.actactor[myself][2] = 4;
+        global.actactor[myself][2] = 6;
         global.canactsus[myself][0] = 1;
         global.actnamesus[myself][0] = stringsetloc("IloveTV", "scr_monstersetup_slash_scr_monstersetup_gml_2044_0");
         global.actsimulsus[myself][0] = 1;
         global.canactral[myself][0] = 1;
         global.actnameral[myself][0] = stringsetloc("IloveTV", "scr_monstersetup_slash_scr_monstersetup_gml_2048_0");
         global.actsimulral[myself][0] = 1;
+        global.canactnoe[myself][0] = 1;
+        global.actnamenoe[myself][0] = stringset("IloveTV");
+        global.actsimulnoe[myself][0] = 1;
     }
     if (global.monstertype[myself] == 61)
     {
@@ -1738,13 +1794,16 @@
         global.actname[myself][1] = stringsetloc("ILoveTV", "scr_monstersetup_slash_scr_monstersetup_gml_2069_0");
         global.canact[myself][2] = 1;
         global.actname[myself][2] = stringsetloc("ILoveTVX", "scr_monstersetup_slash_scr_monstersetup_gml_2072_0");
-        global.actactor[myself][2] = 4;
+        global.actactor[myself][2] = 6;
         global.canactsus[myself][0] = 1;
         global.actnamesus[myself][0] = stringsetloc("IloveTV", "scr_monstersetup_slash_scr_monstersetup_gml_2080_0");
         global.actsimulsus[myself][0] = 1;
         global.canactral[myself][0] = 1;
         global.actnameral[myself][0] = stringsetloc("IloveTV", "scr_monstersetup_slash_scr_monstersetup_gml_2084_0");
         global.actsimulral[myself][0] = 1;
+        global.canactnoe[myself][0] = 1;
+        global.actnamenoe[myself][0] = stringset("IloveTV");
+        global.actsimulnoe[myself][0] = 1;
     }
     if (global.monstertype[myself] == 102)
     {
@@ -1786,34 +1845,73 @@
         var _minigame_string = stringsetloc("Minigame", "scr_monstersetup_slash_scr_monstersetup_gml_2239_0");
         var _playgame_string = stringsetloc("Play#game", "scr_monstersetup_slash_scr_monstersetup_gml_2240_0");
         var _earnscore_string = stringsetloc("Earn#score", "scr_monstersetup_slash_scr_monstersetup_gml_2241_0");
+        var charcount = 0;
+        var char_r = 0;
+        var char_l = 0;
+        for (i = 0; i < 3; i++)
+        {
+            if (global.char[i] != 0)
+            {
+                charcount++;
+            }
+            if (global.char[i] > 1)
+            {
+                if (char_l == 0)
+                {
+                    char_l = global.char[i];
+                }
+                char_r = global.char[i];
+            }
+        }
+        if (char_l == 4)
+        {
+            char_l = 5;
+        }
+        if (char_r == 4)
+        {
+            char_r = 5;
+        }
+        var solonum = 1;
+        if (charcount > 1)
+        {
+            global.canact[myself][1] = 1;
+            global.actname[myself][1] = _minigame_string;
+            global.actdesc[myself][1] = _playgame_string;
+            global.actsimul[myself][1] = 0;
+            global.actactor[myself][1] = 6;
+            global.actcost[myself][1] = 65;
+            solonum = 2;
-        global.canact[myself][1] = 1;
+        }
-        global.actname[myself][1] = _minigame_string;
+        if (charcount == 3)
-        global.actdesc[myself][1] = _playgame_string;
+        {
-        global.actsimul[myself][1] = 0;
+            global.actcost[myself][1] = 20;
-        global.actactor[myself][1] = 4;
+            if (char_l != char_r)
-        global.actcost[myself][1] = 20;
+            {
-        global.canact[myself][2] = 1;
+                global.canact[myself][2] = 1;
-        global.actname[myself][2] = _minigame_string;
+                global.actname[myself][2] = _minigame_string;
-        global.actdesc[myself][2] = _playgame_string;
+                global.actdesc[myself][2] = _playgame_string;
-        global.actsimul[myself][2] = 0;
+                global.actsimul[myself][2] = 0;
-        global.actactor[myself][2] = 2;
+                global.actactor[myself][2] = char_l;
-        global.actcost[myself][2] = 65;
+                global.actcost[myself][2] = 65;
-        global.canact[myself][3] = 1;
+                global.canact[myself][3] = 1;
-        global.actname[myself][3] = _minigame_string;
+                global.actname[myself][3] = _minigame_string;
-        global.actdesc[myself][3] = _playgame_string;
+                global.actdesc[myself][3] = _playgame_string;
-        global.actsimul[myself][3] = 0;
+                global.actsimul[myself][3] = 0;
-        global.actactor[myself][3] = 3;
+                global.actactor[myself][3] = char_r;
-        global.actcost[myself][3] = 65;
+                global.actcost[myself][3] = 65;
+                solonum = 4;
+            }
+        }
-        global.canact[myself][4] = 1;
+        global.canact[myself][solonum] = 1;
-        global.actname[myself][4] = _minigame_string;
+        global.actname[myself][solonum] = _minigame_string;
-        global.actdesc[myself][4] = _playgame_string;
+        global.actdesc[myself][solonum] = _playgame_string;
-        global.actsimul[myself][4] = 0;
+        global.actsimul[myself][solonum] = 0;
-        global.actactor[myself][4] = 0;
+        global.actactor[myself][solonum] = 0;
-        global.actcost[myself][4] = 110;
+        global.actcost[myself][solonum] = 110;
-        global.canact[myself][5] = 1;
+        global.canact[myself][solonum + 1] = 1;
-        global.actname[myself][5] = stringsetloc("ILoveTV", "scr_monstersetup_slash_scr_monstersetup_gml_2148_0");
+        global.actname[myself][solonum + 1] = stringsetloc("ILoveTV", "scr_monstersetup_slash_scr_monstersetup_gml_2148_0");
-        global.actdesc[myself][5] = _earnscore_string;
+        global.actdesc[myself][solonum + 1] = _earnscore_string;
-        global.actsimul[myself][5] = 1;
+        global.actsimul[myself][solonum + 1] = 1;
         var havejeviltail = false;
         if (global.chararmor1[1] == 7 || global.chararmor2[1] == 7 || global.chararmor1[2] == 7 || global.chararmor2[2] == 7 || global.chararmor1[3] == 7 || global.chararmor2[3] == 7)
         {
@@ -1839,6 +1937,10 @@
         global.actnameral[myself][1] = stringsetloc("PopularBoy", "scr_monstersetup_slash_scr_monstersetup_gml_2158_0");
         global.actdescral[myself][1] = stringsetloc("Ralsei#HPChange#=score", "scr_monstersetup_slash_scr_monstersetup_gml_2321_0");
         global.actsimulral[myself][1] = 0;
+        global.canactnoe[myself][0] = 1;
+        global.actnamenoe[myself][0] = stringset("N-Action");
+        global.actdescnoe[myself][0] = _earnscore_string;
+        global.actsimulnoe[myself][0] = 1;
     }
     if (global.monstertype[myself] == 104)
     {
@@ -1863,6 +1965,9 @@
         global.canactral[myself][0] = 1;
         global.actnameral[myself][0] = stringsetloc("R-Action", "scr_monstersetup_slash_scr_monstersetup_gml_2187_0");
         global.actsimulral[myself][0] = 0;
+        global.canactnoe[myself][0] = 1;
+        global.actnamenoe[myself][0] = stringset("N-Action");
+        global.actsimulnoe[myself][0] = 0;
     }
     if (global.monstertype[myself] == 105)
     {
@@ -1878,12 +1983,12 @@
         global.mercymax[myself] = 100;
         global.canact[myself][0] = 1;
         global.actname[myself][0] = stringsetloc("ILOVETV", "scr_monstersetup_slash_scr_monstersetup_gml_2205_0");
-        global.actactor[myself][0] = 4;
+        global.actactor[myself][0] = 7;
         global.canact[myself][1] = 1;
         global.actname[myself][1] = stringsetloc("Resist", "scr_monstersetup_slash_scr_monstersetup_gml_2209_0");
         global.actdesc[myself][1] = stringsetloc("", "scr_monstersetup_slash_scr_monstersetup_gml_2210_0");
         global.actsimul[myself][1] = 0;
-        global.actactor[myself][1] = 4;
+        global.actactor[myself][1] = 7;
     }
     if (global.monstertype[myself] == 106)
     {
