--- a/Code/gml_Object_obj_heroparent_Step_0.gml
+++ b/Code/gml_Object_obj_heroparent_Step_0.gml
@@ -365,15 +365,15 @@
                 damage = ceil(damage * obj_knight_enemy.damagereduction);
                 if (object_index == obj_herokris)
                 {
-                    if (global.hp[2] < 0 && global.hp[3] < 0)
+                    if (scr_heroupcount() <= 1)
                     {
                         damage *= 2;
                     }
-                    else if (global.hp[2] < 0 || global.hp[3] < 0)
+                    if (scr_heroupcount() == 2)
                     {
                         damage *= 1;
                     }
-                    else
+                    if (scr_heroupcount() == 3)
                     {
                         damage = round(damage * 0.5);
                     }
