--- a/Code/gml_Object_obj_shadowman_sharpshoot_cursor_Step_0.gml
+++ b/Code/gml_Object_obj_shadowman_sharpshoot_cursor_Step_0.gml
@@ -54,17 +54,24 @@
     if (button3_h() == 1 && shoottimer > 0 && ammo > 0 && stopshooting == false)
     {
         shoottimer = -3;
-        instance_create(obj_herokris.x + 27 + 34, obj_herokris.y + 33 + 12, obj_shadowman_sharpshoot_heart);
-        ammo -= 1;
-        if (i_ex(obj_herosusie))
-        {
-            instance_create(obj_herosusie.x + 27 + 49, obj_herosusie.y + 42, obj_shadowman_sharpshoot_heart);
-            ammo -= 1;
-        }
-        if (i_ex(obj_heroralsei))
-        {
-            instance_create(obj_heroralsei.x + 47 + 32, (obj_heroralsei.y + 50) - 3, obj_shadowman_sharpshoot_heart);
-            ammo -= 1;
+        spriteoffx[0] = 0;
+        spriteoffx[1] = 61;
+        spriteoffx[2] = 76;
+        spriteoffx[3] = 79;
+        spriteoffx[4] = 65;
+        spriteoffy[0] = 0;
+        spriteoffy[1] = 45;
+        spriteoffy[2] = 42;
+        spriteoffy[3] = 47;
+        spriteoffy[4] = 50;
+        for (var i = 0; i < 3; i++)
+        {
+            if (i_ex(global.charinstance[i]))
+            {
+                inst = instance_create(global.charinstance[i].x + spriteoffx[global.char[i]], global.charinstance[i].y + spriteoffy[global.char[i]], obj_shadowman_sharpshoot_heart);
+                inst.depth = global.charinstance[i].depth - 10;
+                ammo -= 1;
+            }
         }
         snd_play_x(snd_noise, 0.8, 1.2);
     }
