--- a/Code/gml_Object_obj_shutta_photo_controller_Draw_77.gml
+++ b/Code/gml_Object_obj_shutta_photo_controller_Draw_77.gml
@@ -63,7 +63,7 @@
         offside_subject = false;
         have_subject = false;
         var _comment = -4;
-        if (got_kris || got_susie || got_ralsei)
+        if (got_kris || got_susie || got_ralsei || got_noelle)
         {
             have_subject = true;
             var _avg_x = 0;
@@ -102,6 +102,10 @@
             {
                 scr_battle_sprite_set("ralsei", spr_ralsei_pose, 0, 0);
             }
+            if (got_noelle && subact != "krs")
+            {
+                scr_battle_sprite_set("noelle", spr_noelleb_actready, 0, 0);
+            }
         }
         subact_state = "capture";
         switch (subact)
@@ -139,6 +143,12 @@
             case "bullets":
                 event_user(11);
                 break;
+            case "xmas":
+                event_user(12);
+                break;
+            case "findluigi":
+                event_user(13);
+                break;
         }
     }
 }
@@ -147,52 +157,57 @@
     case "intro":
         if (introtimer == 10)
         {
-            if (i_ex(obj_herosusie))
-            {
-                if (i_ex(obj_heroralsei))
-                {
-                    difficulty = 0;
-                    if (obj_shutta_enemy.photoactcount == 1)
-                    {
-                        subact = "free";
-                    }
-                    if (obj_shutta_enemy.photoactcount == 2)
-                    {
-                        subact = "rotation";
-                    }
-                    if (obj_shutta_enemy.photoactcount == 3)
-                    {
-                        subact = "lancer";
-                    }
-                    if (obj_shutta_enemy.photoactcount == 4)
-                    {
+            difficulty = 0;
+            if (obj_shutta_enemy.photoactcount == 1)
+            {
+                subact = "free";
+            }
+            if (obj_shutta_enemy.photoactcount == 2)
+            {
+                subact = "rotation";
+            }
+            if (obj_shutta_enemy.photoactcount == 3)
+            {
+                subact = "lancer";
+            }
+            if (obj_shutta_enemy.photoactcount == 4 || obj_shutta_enemy.photoactcount == 5)
+            {
+                var index = 0;
+                if (obj_shutta_enemy.photoactcount == 5)
+                {
+                    index = 1;
+                }
+                switch (obj_shutta_enemy.curpartyphoto)
+                {
+                    case 2:
                         subact = "susieitems";
-                    }
-                    if (obj_shutta_enemy.photoactcount == 5)
-                    {
+                        break;
+                    case 3:
                         subact = "ralsei";
-                    }
-                    if (obj_shutta_enemy.photoactcount == 6)
-                    {
-                        subact = "rotation";
-                        if (obj_shutta_enemy.acedrotationphoto == true)
-                        {
-                            difficulty = 1;
-                        }
-                    }
-                    if (obj_shutta_enemy.photoactcount == 7)
-                    {
-                        subact = "lancer";
-                    }
-                    if (obj_shutta_enemy.photoactcount == 8)
-                    {
-                        subact = "krs";
-                    }
-                }
-            }
-            else
-            {
-                subact = scr_weighted_choice("enemyjump", 1);
+                        break;
+                    case 4:
+                        subact = "xmas";
+                        break;
+                    default:
+                        subact = "findluigi";
+                        break;
+                }
+            }
+            if (obj_shutta_enemy.photoactcount == 6)
+            {
+                subact = "rotation";
+                if (obj_shutta_enemy.acedrotationphoto == true)
+                {
+                    difficulty = 1;
+                }
+            }
+            if (obj_shutta_enemy.photoactcount == 7)
+            {
+                subact = "lancer";
+            }
+            if (obj_shutta_enemy.photoactcount == 8)
+            {
+                subact = "krs";
             }
             if (i_ex(obj_tenna_enemy))
             {
@@ -273,6 +288,12 @@
             case "bullets":
                 event_user(11);
                 break;
+            case "xmas":
+                event_user(12);
+                break;
+            case "findluigi":
+                event_user(13);
+                break;
         }
         if (tookaphoto == true)
         {
@@ -452,6 +473,12 @@
             break;
         case "bullets":
             event_user(11);
+            break;
+        case "xmas":
+            event_user(12);
+            break;
+        case "findluigi":
+            event_user(13);
             break;
     }
 }
