--- a/Code/gml_Object_obj_shutta_photo_controller_Other_18.gml
+++ b/Code/gml_Object_obj_shutta_photo_controller_Other_18.gml
@@ -91,6 +91,20 @@
             }
             global.flag[515] = 1;
         }
+        else if (got_noelle && closest_hero == "noelle")
+        {
+            enemy.photosuccesscount++;
+            endmessage = message_nicephoto;
+            if (enemy.amiabossfight == true)
+            {
+                obj_shutta_photo_controller.endmessage = message_noelle;
+                if (global.flag[516] > 0)
+                {
+                    obj_shutta_photo_controller.endmessage = message_noelle2;
+                }
+            }
+            global.flag[516] = 1;
+        }
         else if (_trueshutta != -1 && place_meeting(x, y, _trueshutta) && collision_circle(x, y, reticle_size + 40, _trueshutta, false, true))
         {
             with (obj_shutta_enemy)
