--- a/Code/gml_Object_obj_shutta_photo_controller_Other_21.gml
+++ b/Code/gml_Object_obj_shutta_photo_controller_Other_21.gml
@@ -247,6 +247,20 @@
             }
             global.flag[515] = 1;
         }
+        else if (got_noelle && closest_hero == "noelle")
+        {
+            enemy.photosuccesscount++;
+            endmessage = message_nicephoto;
+            if (enemy.amiabossfight == true)
+            {
+                obj_shutta_photo_controller.endmessage = message_noelle;
+                if (global.flag[516] > 0)
+                {
+                    obj_shutta_photo_controller.endmessage = message_noelle2;
+                }
+            }
+            global.flag[516] = 1;
+        }
         else if (place_meeting(x, y, obj_shutta_reticle_bullet) || place_meeting(x, y, obj_shutta_reticle_bullet2) || place_meeting(x, y, obj_shutta_nobyact_fakebullet))
         {
             with (obj_shutta_reticle_bullet)
