--- a/Code/gml_Object_obj_shutta_photo_controller_CleanUp_0.gml
+++ b/Code/gml_Object_obj_shutta_photo_controller_CleanUp_0.gml
@@ -3,15 +3,7 @@
 {
     sprite_delete(screenshot);
 }
-with (obj_herokris)
+with (obj_heroparent)
-{
-    mask_index = -1;
-}
-with (obj_herosusie)
-{
-    mask_index = -1;
-}
-with (obj_heroralsei)
 {
     mask_index = -1;
 }
@@ -19,26 +11,40 @@
 {
     instance_destroy();
 }
+with (obj_shutta_nobyact_findluigi)
+{
+    instance_destroy();
+}
 with (obj_shutta_nobyact_tv)
 {
     instance_destroy();
 }
+scr_battle_sprite_reset("noelle");
 scr_battle_sprite_reset("ralsei");
 scr_battle_sprite_reset("susie");
 scr_battle_sprite_reset("kris");
-with (obj_herokris)
-{
-    state = other.krisstate;
-    global.faceaction[myself] = other.krisfaceaction;
-}
-with (obj_herosusie)
-{
-    state = other.susiestate;
-    global.faceaction[myself] = other.susiefaceaction;
-}
-with (obj_heroralsei)
-{
-    state = other.ralseistate;
-    global.faceaction[myself] = other.ralseifaceaction;
+if (i_ex(global.charinstance[0]))
+{
+    with (global.charinstance[0])
+    {
+        state = other.charstate[0];
+        global.faceaction[myself] = other.charfaceaction[0];
+    }
+}
+if (i_ex(global.charinstance[1]))
+{
+    with (global.charinstance[1])
+    {
+        state = other.charstate[1];
+        global.faceaction[myself] = other.charfaceaction[1];
+    }
+}
+if (i_ex(global.charinstance[2]))
+{
+    with (global.charinstance[2])
+    {
+        state = other.charstate[2];
+        global.faceaction[myself] = other.charfaceaction[2];
+    }
 }
 
