--- a/Code/gml_Object_obj_shutta_photo_controller_Create_0.gml
+++ b/Code/gml_Object_obj_shutta_photo_controller_Create_0.gml
@@ -48,9 +48,11 @@
 message_kris = stringsetloc("* Nice! That one's going in my Kris compilation!", "obj_shutta_photo_controller_slash_Create_0_gml_55_0");
 message_susie = stringsetloc("* Oh! A wild Susie photo! ... Arms inside the ride, folks!", "obj_shutta_photo_controller_slash_Create_0_gml_56_0");
 message_ralsei = stringsetloc("* Ralsei photo!? We can sell that! Here's 100PTs!", "obj_shutta_photo_controller_slash_Create_0_gml_57_0");
+message_noelle = stringset("* Ooh, a Noelle photo! ... It's been a while since we had one of those!");
 message_kris2 = stringsetloc("* Stop! My Kris compilation is full!", "obj_shutta_photo_controller_slash_Create_0_gml_59_0");
 message_susie2 = stringsetloc("* What is this, a Susie safari?", "obj_shutta_photo_controller_slash_Create_0_gml_60_0");
 message_ralsei2 = stringsetloc("* More Ralsei!? No, we need the scarcity!!", "obj_shutta_photo_controller_slash_Create_0_gml_61_0");
+message_noelle2 = stringset("* Can you believe it folks? The Holiday specials! Just a week away!");
 message_toilet = stringsetloc("* Photo of toilet!?! Are we still TV-Y7!?", "obj_shutta_photo_controller_slash_Create_0_gml_63_0");
 message_toilet2 = stringsetloc("* A perfect shot!! But are we still TV-Y7!?", "obj_shutta_photo_controller_slash_Create_0_gml_64_0");
 endmessage = message_default;
@@ -75,9 +77,11 @@
 got_kris = false;
 got_susie = false;
 got_ralsei = false;
+got_noelle = false;
 closest_hero = "none";
 fastphoto = false;
 powerphoto = false;
+fancyphoto = false;
 timerb = 0;
 posemax = 60 + irandom(30);
 posesiner = 0;
@@ -114,6 +118,10 @@
 {
     mask_index = spr_ralsei_funnyface;
 }
+with (obj_heronoelle)
+{
+    mask_index = spr_noelleb_actready;
+}
 screenshot = -1;
 reticle_over_good_shot = false;
 reticle_over_good_shot_prev = false;
@@ -123,25 +131,29 @@
 bullettarget = -1;
 secrettargetx = 0;
 secrettargety = 0;
-krisstate = -1;
-krisfaceaction = -1;
-susiestate = -1;
-susiefaceaction = -1;
-ralseistate = -1;
-ralseifaceaction = -1;
-if (i_ex(obj_herokris))
-{
-    krisstate = obj_herokris.state;
-    krisfaceaction = global.faceaction[obj_herokris.myself];
-}
-if (i_ex(obj_herosusie))
-{
-    susiestate = obj_herosusie.state;
-    susiefaceaction = global.faceaction[obj_herosusie.myself];
-}
-if (i_ex(obj_heroralsei))
-{
-    ralseistate = obj_heroralsei.state;
-    ralseifaceaction = global.faceaction[obj_heroralsei.myself];
+charstate[0] = -1;
+charfaceaction[0] = -1;
+charstate[1] = -1;
+charfaceaction[1] = -1;
+charstate[2] = -1;
+charfaceaction[2] = -1;
+activekris = (scr_findchar(1) > -1) ? global.charinstance[scr_findchar(1)] : -4;
+activesusie = (scr_findchar(2) > -1) ? global.charinstance[scr_findchar(2)] : -4;
+activeralsei = (scr_findchar(3) > -1) ? global.charinstance[scr_findchar(3)] : -4;
+activenoelle = (scr_findchar(4) > -1) ? global.charinstance[scr_findchar(4)] : -4;
+if (i_ex(global.charinstance[0]))
+{
+    charstate[0] = global.charinstance[0].state;
+    charfaceaction[0] = global.faceaction[global.charinstance[0].myself];
+}
+if (i_ex(global.charinstance[1]))
+{
+    charstate[1] = global.charinstance[1].state;
+    charfaceaction[1] = global.faceaction[global.charinstance[1].myself];
+}
+if (i_ex(global.charinstance[2]))
+{
+    charstate[2] = global.charinstance[2].state;
+    charfaceaction[2] = global.faceaction[global.charinstance[2].myself];
 }
 
