--- a/Code/gml_Object_obj_shutta_photo_controller_Draw_0.gml
+++ b/Code/gml_Object_obj_shutta_photo_controller_Draw_0.gml
@@ -1,7 +1,7 @@
 siner++;
 if (state == "intro" || state == "aiming")
 {
-    if (i_ex(obj_shutta_nobyacttest) && obj_shutta_nobyacttest.photocooldowntimer > 0)
+    if ((i_ex(obj_shutta_nobyacttest) && obj_shutta_nobyacttest.photocooldowntimer > 0) || (i_ex(obj_shutta_nobyact_findluigi) && obj_shutta_nobyact_findluigi.photocooldowntimer > 0))
     {
         draw_sprite_ext(sprite_index, 1, x, y, image_xscale, image_yscale, image_angle, image_blend, image_alpha);
     }
