--- a/Code/gml_Object_obj_shutta_photo_controller_Draw_77.gml
+++ b/Code/gml_Object_obj_shutta_photo_controller_Draw_77.gml
@@ -63,7 +63,7 @@
         offside_subject = false;
         have_subject = false;
         var _comment = -4;
-        if (got_kris || got_susie || got_ralsei)
+        if (got_kris || got_susie || got_ralsei || got_noelle)
         {
             have_subject = true;
             var _avg_x = 0;
@@ -92,15 +92,31 @@
             _avg_y /= _hero_num;
             if (got_kris && subact != "krs")
             {
+                with (obj_herokris)
+                {
-                scr_battle_sprite_set("kris", spr_kris_peace, 0, 0);
+                    scr_battle_sprite_set(myself, spr_kris_peace, 0, 0);
+                }
             }
             if (got_susie && (subact != "susieitems" && subact != "krs"))
             {
+                with (obj_herosusie)
+                {
-                scr_battle_sprite_set("susie", spr_susie_pose, 0, 0);
+                    scr_battle_sprite_set(myself, spr_susie_pose, 0, 0);
+                }
             }
             if (got_ralsei && subact != "krs")
             {
+                with (obj_heroralsei)
+                {
-                scr_battle_sprite_set("ralsei", spr_ralsei_pose, 0, 0);
+                    scr_battle_sprite_set(myself, spr_ralsei_pose, 0, 0);
+                }
+            }
+            if (got_noelle && subact != "krs")
+            {
+                with (obj_heronoelle)
+                {
+                    scr_battle_sprite_set(myself, spr_noelleb_actready, 0, 0);
+                }
             }
         }
         subact_state = "capture";
@@ -139,6 +155,12 @@
             case "bullets":
                 event_user(11);
                 break;
+            case "xmas":
+                event_user(12);
+                break;
+            case "findluigi":
+                event_user(13);
+                break;
         }
     }
 }
@@ -147,52 +169,57 @@
     case "intro":
         if (introtimer == 10)
         {
-            if (i_ex(obj_herosusie))
-            {
-                if (i_ex(obj_heroralsei))
-                {
-                    difficulty = 0;
-                    if (obj_shutta_enemy.photoactcount == 1)
-                    {
-                        subact = "free";
-                    }
-                    if (obj_shutta_enemy.photoactcount == 2)
-                    {
-                        subact = "rotation";
-                    }
-                    if (obj_shutta_enemy.photoactcount == 3)
-                    {
-                        subact = "lancer";
-                    }
-                    if (obj_shutta_enemy.photoactcount == 4)
-                    {
+            difficulty = 0;
+            if (obj_shutta_enemy.photoactcount == 1)
+            {
+                subact = "free";
+            }
+            if (obj_shutta_enemy.photoactcount == 2)
+            {
+                subact = "rotation";
+            }
+            if (obj_shutta_enemy.photoactcount == 3)
+            {
+                subact = "lancer";
+            }
+            if (obj_shutta_enemy.photoactcount == 4 || obj_shutta_enemy.photoactcount == 5)
+            {
+                var index = 0;
+                if (obj_shutta_enemy.photoactcount == 5)
+                {
+                    index = 1;
+                }
+                switch (obj_shutta_enemy.curpartyphoto)
+                {
+                    case 2:
                         subact = "susieitems";
-                    }
-                    if (obj_shutta_enemy.photoactcount == 5)
-                    {
+                        break;
+                    case 3:
                         subact = "ralsei";
-                    }
-                    if (obj_shutta_enemy.photoactcount == 6)
-                    {
-                        subact = "rotation";
-                        if (obj_shutta_enemy.acedrotationphoto == true)
-                        {
-                            difficulty = 1;
-                        }
-                    }
-                    if (obj_shutta_enemy.photoactcount == 7)
-                    {
-                        subact = "lancer";
-                    }
-                    if (obj_shutta_enemy.photoactcount == 8)
-                    {
-                        subact = "krs";
-                    }
-                }
-            }
-            else
-            {
-                subact = scr_weighted_choice("enemyjump", 1);
+                        break;
+                    case 4:
+                        subact = "xmas";
+                        break;
+                    default:
+                        subact = "findluigi";
+                        break;
+                }
+            }
+            if (obj_shutta_enemy.photoactcount == 6)
+            {
+                subact = "rotation";
+                if (obj_shutta_enemy.acedrotationphoto == true)
+                {
+                    difficulty = 1;
+                }
+            }
+            if (obj_shutta_enemy.photoactcount == 7)
+            {
+                subact = "lancer";
+            }
+            if (obj_shutta_enemy.photoactcount == 8)
+            {
+                subact = "krs";
             }
             if (i_ex(obj_tenna_enemy))
             {
@@ -224,10 +251,9 @@
             }
             screenshot = sprite_create_from_surface(photo_surface, 0, 0, photo_width, photo_height, 0, 0, 0, 0);
         }
-        scr_battle_sprite_reset("kris");
+        scr_battle_sprite_reset(0);
-        scr_battle_sprite_reset("noelle");
+        scr_battle_sprite_reset(1);
-        scr_battle_sprite_reset("ralsei");
+        scr_battle_sprite_reset(2);
-        scr_battle_sprite_reset("susie");
         filter1.x = -9999;
         filter1.y = -9999;
         filter2.x = -9999;
@@ -273,6 +299,12 @@
             case "bullets":
                 event_user(11);
                 break;
+            case "xmas":
+                event_user(12);
+                break;
+            case "findluigi":
+                event_user(13);
+                break;
         }
         if (tookaphoto == true)
         {
@@ -452,6 +484,12 @@
             break;
         case "bullets":
             event_user(11);
+            break;
+        case "xmas":
+            event_user(12);
+            break;
+        case "findluigi":
+            event_user(13);
             break;
     }
 }
