--- a/Code/gml_Object_obj_shutta_photo_controller_Other_16.gml
+++ b/Code/gml_Object_obj_shutta_photo_controller_Other_16.gml
@@ -36,6 +36,10 @@
             {
                 mercygained += 10;
             }
+            if (fancyphoto)
+            {
+                mercygained += 5;
+            }
             mercygained += 10;
             if (!i_ex(obj_shutta_text))
             {
@@ -60,6 +64,10 @@
             {
                 mercygained += 10;
             }
+            if (fancyphoto)
+            {
+                mercygained += 5;
+            }
             mercygained += 10;
             if (!i_ex(obj_shutta_text))
             {
@@ -86,6 +94,41 @@
             if (powerphoto)
             {
                 mercygained += 10;
+            }
+            if (fancyphoto)
+            {
+                mercygained += 5;
+            }
+            mercygained += 10;
+            if (!i_ex(obj_shutta_text))
+            {
+                var _comment = instance_create(x, y, obj_shutta_text);
+                _comment.text = comment_center;
+                _comment.color = c_yellow;
+            }
+        }
+        else if (got_noelle && closest_hero == "noelle")
+        {
+            enemy.photosuccessballoon = 1;
+            if (enemy.amiabossfight == true)
+            {
+                if (global.flag[516] == 0)
+                {
+                    obj_shutta_photo_controller.endmessage = message_noelle;
+                }
+                if (global.flag[516] > 0)
+                {
+                    obj_shutta_photo_controller.endmessage = message_noelle2;
+                }
+            }
+            global.flag[516] = 1;
+            if (powerphoto)
+            {
+                mercygained += 10;
+            }
+            if (fancyphoto)
+            {
+                mercygained += 5;
             }
             mercygained += 10;
             if (!i_ex(obj_shutta_text))
