--- a/Code/gml_Object_obj_battlecontroller_Step_0.gml
+++ b/Code/gml_Object_obj_battlecontroller_Step_0.gml
@@ -317,7 +317,7 @@
                 }
                 spadesuccesscon = 1;
                 var ralseiyarn = false;
-                if (global.charturn == 2 && (obj_heroralsei.sprite_index == spr_ralseib_yarn_1 || obj_heroralsei.sprite_index == spr_ralseib_yarn_2) && obj_rouxls_ch3_enemy.yarnendturn == 0)
+                if (global.charturn == 2 && i_ex(obj_heroralsei) && (obj_heroralsei.sprite_index == spr_ralseib_yarn_1 || obj_heroralsei.sprite_index == spr_ralseib_yarn_2) && obj_rouxls_ch3_enemy.yarnendturn == 0)
                 {
                     selnoise = 0;
                     global.bmenuno = 0;
@@ -1104,6 +1104,25 @@
             if (global.actactor[global.bmenucoord[11][global.charturn]][global.bmenucoord[9][global.charturn]] == 5)
             {
                 if (havechar[3] == 0 || global.hp[4] <= 0)
+                {
+                    canpress = 0;
+                }
+            }
+            if (global.actactor[global.bmenucoord[11][global.charturn]][global.bmenucoord[9][global.charturn]] == 6)
+            {
+                if (havechar[1] == 1 && global.hp[2] <= 0)
+                {
+                    canpress = 0;
+                }
+                if (havechar[2] == 1 && global.hp[3] <= 0)
+                {
+                    canpress = 0;
+                }
+                if (havechar[3] == 1 && global.hp[4] <= 0)
+                {
+                    canpress = 0;
+                }
+                if (havechar[1] != 1 && havechar[2] != 1 && havechar[3] != 1)
                 {
                     canpress = 0;
                 }
