--- a/Code/gml_Object_obj_tenna_enemy_Step_0.gml
+++ b/Code/gml_Object_obj_tenna_enemy_Step_0.gml
@@ -1,12 +1,3 @@
-if (!i_ex(obj_herosusie) || !i_ex(obj_heroralsei))
-{
-    if (debugvar == false && scr_debug())
-    {
-        scr_debug_print("You need Susie and Ralsei for this battle");
-        debugvar = true;
-    }
-    exit;
-}
 if (audience_goes_away == true)
 {
     with (obj_tenna_enemy_bg)
@@ -693,6 +684,7 @@
             smashcutwithouttakingdamage = 1;
         }
         popularboy = false;
+        popularboycount = 0;
         gothit = false;
         bet = false;
         minigameactivated = false;
@@ -701,38 +693,50 @@
         smashcuthurt = 0;
         pointsmultiplierthisturn = 5;
         totalfails = 0;
-        scr_battle_sprite_reset("ralsei");
+        for (var i = 0; i < 3; i++)
+        {
+            if (global.char[i] == 3)
+            {
+                scr_battle_sprite_reset(i);
+            }
+        }
         if (resetralsei == true)
         {
             resetralsei = false;
             snd_play(snd_wing);
-            obj_heroralsei.x = obj_heroralsei.xstart;
-            obj_heroralsei.y = obj_heroralsei.ystart;
-            with (obj_heroralsei)
-            {
-                normalsprite = spr_ralsei_walk_right;
-                idlesprite = spr_ralsei_idle;
-                defendsprite = spr_ralsei_defend;
-                hurtsprite = spr_ralsei_hurt_fixed;
-                attackreadysprite = spr_ralsei_attackready;
-                attacksprite = spr_ralsei_attack;
-                itemsprite = spr_ralsei_item;
-                itemreadysprite = spr_ralsei_itemready;
-                spellreadysprite = spr_ralsei_spellready;
-                spellsprite = spr_ralsei_spell;
-                defeatsprite = spr_ralsei_defeat;
-                victorysprite = spr_ralsei_victory;
-                actreadysprite = spr_ralsei_actready;
-                actsprite = spr_ralsei_act;
-            }
-            repeat (26)
-            {
-                puff = instance_create(((obj_heroralsei.x + 50) - 60) + irandom(80), (obj_heroralsei.y - 90) + 90 + irandom(100), obj_animation);
-                puff.depth = obj_heroralsei.depth - 1;
-                puff.sprite_index = spr_susiezilla_houseexplosion;
-                puff.image_speed = 0.5;
-                puff.image_xscale = 1;
-                puff.image_yscale = 1;
+            if (instance_exists(obj_heroralsei))
+            {
+                with (obj_heroralsei)
+                {
+                    x = xstart;
+                    y = ystart;
+                    if (idlesprite == spr_gameshow_drowningRalsei_ralsei_origin_edit)
+                    {
+                        repeat (26)
+                        {
+                            puff = instance_create(((x + 50) - 60) + irandom(80), (y - 90) + 90 + irandom(100), obj_animation);
+                            puff.depth = depth - 1;
+                            puff.sprite_index = spr_susiezilla_houseexplosion;
+                            puff.image_speed = 0.5;
+                            puff.image_xscale = 1;
+                            puff.image_yscale = 1;
+                        }
+                    }
+                    normalsprite = spr_ralsei_walk_right;
+                    idlesprite = spr_ralsei_idle;
+                    defendsprite = spr_ralsei_defend;
+                    hurtsprite = spr_ralsei_hurt_fixed;
+                    attackreadysprite = spr_ralsei_attackready;
+                    attacksprite = spr_ralsei_attack;
+                    itemsprite = spr_ralsei_item;
+                    itemreadysprite = spr_ralsei_itemready;
+                    spellreadysprite = spr_ralsei_spellready;
+                    spellsprite = spr_ralsei_spell;
+                    defeatsprite = spr_ralsei_defeat;
+                    victorysprite = spr_ralsei_victory;
+                    actreadysprite = spr_ralsei_actready;
+                    actsprite = spr_ralsei_act;
+                }
             }
         }
         if (lancercheat == true)
@@ -752,7 +756,7 @@
 }
 if (popularboy == true && ralseihpprev != global.hp[3])
 {
-    obj_tenna_enemy_bg.addscore += round(abs(ralseihpprev - global.hp[3]) * 0.17);
+    obj_tenna_enemy_bg.addscore += (round(abs(ralseihpprev - global.hp[3]) * 0.17) * popularboycount);
 }
 ralseihpprev = global.hp[3];
 if (lancerspin == true && lancermarker.y > (cameray() + 40) && lancermarker.gravity > 0)
@@ -795,7 +799,7 @@
         }
     }
     var havejeviltail = false;
-    if (global.chararmor1[1] == 7 || global.chararmor2[1] == 7 || global.chararmor1[2] == 7 || global.chararmor2[2] == 7 || global.chararmor1[3] == 7 || global.chararmor2[3] == 7)
+    if (scr_armorcheck_equipped_party(7) > 0)
     {
         havejeviltail = true;
     }
@@ -806,7 +810,7 @@
         {
             msgsetloc(0, "* TENNA - Despite his size, sadly, he's quite fragile./%", "obj_tenna_enemy_slash_Step_0_gml_446_0");
         }
-        else if (global.chararmor1[1] == 21 || global.chararmor2[1] == 21 || global.chararmor1[2] == 21 || global.chararmor2[2] == 21 || global.chararmor1[3] == 21 || global.chararmor2[3] == 21)
+        else if (scr_armorcheck_equipped_party(21))
         {
             msgsetloc(0, "* TENNA - THE [Tragic businesmen] THAT [Died] AT THE [Now] OF THE [Story]/%", "obj_tenna_enemy_slash_Step_0_gml_449_0");
         }
@@ -816,30 +820,32 @@
         }
         scr_battletext_default();
     }
-    if ((acting == 2 && actcon == 0) || (acting == 3 && actcon == 0) || (acting == 4 && actcon == 0) || (acting == 5 && actcon == 0))
+    if ((acting == act_multiminigame && actcon == 0) || (acting > act_multiminigame && acting < act_solominigame && actcon == 0 && act_multiminigame != -1) || (acting == act_solominigame && actcon == 0))
     {
-        if (acting == 2)
-        {
-            strng = stringsetloc("Everyone gets", "obj_tenna_enemy_slash_Step_0_gml_387_0_b");
-        }
-        if (acting == 3)
-        {
-            strng = stringsetloc("You and Susie get", "obj_tenna_enemy_slash_Step_0_gml_385_0");
-        }
-        if (acting == 4)
-        {
-            strng = stringsetloc("You and Ralsei get", "obj_tenna_enemy_slash_Step_0_gml_386_0");
-        }
-        if (acting == 5)
-        {
-            if (global.lang != "ja")
-            {
-                strng = stringsetloc("You get", "obj_tenna_enemy_slash_Step_0_gml_384_0");
-            }
-            else
-            {
-                strng = "";
-            }
+        switch (global.actactor[myself][acting - 1])
+        {
+            case 6:
+                strng = stringsetloc("Everyone gets", "obj_tenna_enemy_slash_Step_0_gml_387_0_b");
+                break;
+            case 2:
+                strng = stringsetloc("You and Susie get", "obj_tenna_enemy_slash_Step_0_gml_385_0");
+                break;
+            case 3:
+                strng = stringsetloc("You and Ralsei get", "obj_tenna_enemy_slash_Step_0_gml_386_0");
+                break;
+            case 5:
+                strng = stringset("You and Noelle get");
+                break;
+            default:
+                if (global.lang != "ja")
+                {
+                    strng = stringsetloc("You get", "obj_tenna_enemy_slash_Step_0_gml_384_0");
+                }
+                else
+                {
+                    strng = "";
+                }
+                break;
         }
         acting = 5.1;
     }
@@ -856,9 +862,13 @@
             minigametimecon = 1;
         }
     }
-    if (acting == 6 && actcon == 0)
+    if (acting == (act_solominigame + 1) && actcon == 0)
     {
         var _ilovetvstring = stringsetloc("* You said \"I Love TV\"!", "obj_tenna_enemy_slash_Step_0_gml_502_0");
+        if (simulcountkri > 1)
+        {
+            extrapoints += 10;
+        }
         if (simultotal == 1)
         {
             msgset(0, _ilovetvstring + "/%");
@@ -885,7 +895,7 @@
         with (obj_tenna_enemy_bg)
         {
             addscore += (20 + other.extrapoints);
-            if (global.hp[2] < 1 && global.hp[3] < 1)
+            if (scr_heroupcount() <= 1)
             {
                 addscore += 20;
             }
@@ -907,7 +917,7 @@
             with (obj_tenna_enemy_bg)
             {
                 addscore += 10;
-                if (global.hp[1] < 1)
+                if (scr_havechar(1) && global.hp[1] < 1)
                 {
                     addscore += 10;
                 }
@@ -916,7 +926,7 @@
         else
         {
             extrapoints += 10;
-            if (global.hp[1] < 1)
+            if (scr_havechar(1) && global.hp[1] < 1)
             {
                 extrapoints += 10;
             }
@@ -942,13 +952,13 @@
             }
         }
     }
+    var actinghero = global.charinstance[global.currentactingchar];
     if (actingsus == 2 && actconsus == 1)
     {
         bet = true;
-        obj_herosusie.x -= 8;
-        obj_herosusie.y -= 6;
-        scr_act_charsprite("susie", spr_susie_zoosuit_Card, 1/3, 8);
-        _charinstance.actspriteobject = _charactsprite[_charnum];
+        actinghero.x -= 8;
+        actinghero.y -= 6;
+        actinghero.actspriteobject = scr_act_charsprite(global.currentactingchar, spr_susie_zoosuit_Card, 1/3, 8);
         if (havejeviltail)
         {
             betamount = 30;
@@ -980,9 +990,9 @@
     if (actingsus == 2.1 && !i_ex(obj_writer))
     {
         scr_act_charsprite_end();
-        obj_herosusie.x += 8;
+        actinghero.x += 8;
-        obj_herosusie.y += 6;
+        actinghero.y += 6;
-        scr_battle_sprite_reset("susie");
+        scr_battle_sprite_reset(global.currentactingchar);
         actingsus = 2.2;
         actcon = 1;
         actconsus = 0;
@@ -994,7 +1004,7 @@
             with (obj_tenna_enemy_bg)
             {
                 addscore += 10;
-                if (global.hp[1] < 1)
+                if (scr_havechar(1) && global.hp[1] < 1)
                 {
                     addscore += 10;
                 }
@@ -1003,7 +1013,7 @@
         else
         {
             extrapoints += 10;
-            if (global.hp[1] < 1)
+            if (scr_havechar(1) && global.hp[1] < 1)
             {
                 extrapoints += 10;
             }
@@ -1032,6 +1042,7 @@
     if (actingral == 2 && actconral == 1)
     {
         popularboy = true;
+        popularboycount++;
         saveralseihp = global.hp[3];
         instance_create(x, y, obj_popularboy_act);
         snd_play(snd_drumroll);
@@ -1041,17 +1052,10 @@
         }
         else
         {
-            if (simulorderral == 0)
-            {
-                actconral = 20;
-            }
-            else
-            {
-                actconral = 0;
-            }
             actingral = 2.3;
             msgsetloc(0, "* Ralsei's HP linked to points!/%", "obj_tenna_enemy_slash_Step_0_gml_602_0");
             scr_simultext("ralsei");
+            actconral = 20;
         }
     }
     if (actingral == 2.1 && !i_ex(obj_popularboy_act))
@@ -1061,6 +1065,16 @@
             actingral = 2.2;
             msgsetloc(0, "* Ralsei became the center of attention!/%", "obj_tenna_enemy_slash_Step_0_gml_612_0");
             scr_battletext_default();
+            global.actsimulral[myself][1] = 1;
+            var i = global.currentactingchar + 1;
+            while (i < 3)
+            {
+                if (global.char[i] == 3 && global.actingsingle[i] == 1 && global.actingchoice[i] == 1)
+                {
+                    global.actingsimul[i] = 1;
+                }
+                i++;
+            }
         }
     }
     if (actingral == 2.2 && !i_ex(obj_writer))
@@ -1078,14 +1092,57 @@
         actingral = 2.3;
         actcon = 1;
         actconral = 0;
+    }
+    if (actingnoe == 1 && actconnoe == 1)
+    {
+        if (acting != 6.1)
+        {
+            with (obj_tenna_enemy_bg)
+            {
+                addscore += 10;
+                if (scr_havechar(1) && global.hp[1] < 1)
+                {
+                    addscore += 10;
+                }
+            }
+        }
+        else
+        {
+            extrapoints += 10;
+            if (scr_havechar(1) && global.hp[1] < 1)
+            {
+                extrapoints += 10;
+            }
+        }
+        var _n_act_string = stringset("* Noelle said \"I Love TV!\"");
+        if (simultotal == 1)
+        {
+            msgset(0, _n_act_string + "/%");
+            scr_battletext_default();
+            actconnoe = 20;
+        }
+        else
+        {
+            msgset(0, _n_act_string);
+            scr_simultext("noelle");
+            if (simulordernoe == 0)
+            {
+                actconnoe = 20;
+            }
+            else
+            {
+                actconnoe = 0;
+            }
-        global.actsimulral[myself][1] = 1;
+        }
     }
-    if (actcon == 20 || actconsus == 20 || actconral == 20)
+    if (actcon == 20 || actconsus == 20 || actconral == 20 || actconnoe == 20)
     {
+        scr_nextact_simul();
         if (scr_terminate_writer() && obj_tenna_enemy_bg.addscore == 0)
         {
             actconsus = -1;
             actconral = -1;
+            actconnoe = -1;
             actcon = 1;
             with (obj_face)
             {
@@ -1332,6 +1389,13 @@
             if (global.hp[3] > 0)
             {
                 with (obj_heroralsei)
+                {
+                    instance_create(x + 30, y + 50, obj_tenna_x_hurt);
+                }
+            }
+            if (global.hp[4] > 0)
+            {
+                with (obj_heronoelle)
                 {
                     instance_create(x + 30, y + 50, obj_tenna_x_hurt);
                 }
