--- a/Code/gml_Object_obj_zapper_enemy_Step_0.gml
+++ b/Code/gml_Object_obj_zapper_enemy_Step_0.gml
@@ -3,9 +3,9 @@
     init = 1;
     if (i_ex(obj_tenna_board4_enemy))
     {
-        global.actactor[0][0] = 4;
+        global.actactor[0][0] = 7;
-        global.actactor[0][1] = 4;
+        global.actactor[0][1] = 7;
-        global.actactor[0][2] = 4;
+        global.actactor[0][2] = 7;
         scr_spellmenu_setup();
     }
 }
@@ -668,10 +668,31 @@
         actconral = 0;
         closedcaptioncon = 0;
     }
+    if (actingnoe == 1 && actconnoe == 1)
+    {
+        msgset(0, "* Noelle replaced the batteries!/%");
+        scr_mercyadd(myself, 25);
+        global.monsterhp[myself] = global.monstermaxhp[myself];
+        snd_play(snd_power);
+        healamt = instance_create(global.monsterx[myself], global.monstery[myself], obj_dmgwriter);
+        with (healamt)
+        {
+            delay = 8;
+            type = 3;
+            specialmessage = 3;
+        }
+        healanim = instance_create(x, y, obj_healanim);
+        healanim.target = id;
+        scr_battletext_default();
+        actcon = 1;
+        actconnoe = 0;
+    }
-    if (actcon == 20 || actconsus == 20 || actconral == 20)
+    if (actcon == 20 || actconsus == 20 || actconral == 20 || actconnoe == 20)
     {
+        scr_nextact_simul();
         if (scr_terminate_writer())
         {
+            actconnoe = -1;
             actconsus = -1;
             actconral = -1;
             actcon = 1;
