--- a/Code/gml_Object_obj_battlecontroller_Draw_0.gml
+++ b/Code/gml_Object_obj_battlecontroller_Draw_0.gml
@@ -130,19 +130,23 @@
 if (global.chapter == 3 && i_ex(obj_rouxls_ch3_enemy) && obj_rouxls_ch3_enemy.intro >= 2 && spadebuttonenabled == true && global.bmenuno == 0 && global.myfight == 0)
 {
     var lowhealth = false;
-    if (global.hp[1] < (global.maxhp[1] / 3))
+    if (scr_havechar(1) && global.hp[1] < (global.maxhp[1] / 3))
     {
         lowhealth = true;
     }
-    if (global.hp[2] < (global.maxhp[2] / 3))
+    if (scr_havechar(2) && global.hp[2] < (global.maxhp[2] / 3))
     {
         lowhealth = true;
     }
-    if (global.hp[3] < (global.maxhp[3] / 3))
+    if (scr_havechar(3) && global.hp[3] < (global.maxhp[3] / 3))
     {
         lowhealth = true;
     }
-    if ((global.hp[1] + global.hp[2] + global.hp[3]) < ((global.maxhp[1] + global.maxhp[2] + global.maxhp[3]) / 2))
+    if (scr_havechar(4) && global.hp[4] < (global.maxhp[4] / 3))
+    {
+        lowhealth = true;
+    }
+    if ((global.hp[global.char[0]] + global.hp[global.char[1]] + global.hp[global.char[2]]) < ((global.maxhp[global.char[0]] + global.maxhp[global.char[1]] + global.maxhp[global.char[2]]) / 2))
     {
         lowhealth = true;
     }
@@ -847,78 +851,12 @@
         }
     }
 }
-if (global.bmenuno == 2 && global.myfight == 0 && global.flag[34] == 1)
-{
-    thischar = global.char[global.charturn];
-    spellcoord = global.bmenucoord[2][global.charturn];
-    page = 0;
-    if (spellcoord > 5)
-    {
-        page = 1;
-        spellcoord -= 6;
-    }
-    icx = 10;
-    icy = 385;
-    if (spellcoord == 1 || spellcoord == 3 || spellcoord == 5)
-    {
-        icx = 230;
-    }
-    if (spellcoord > 1 && spellcoord < 4)
-    {
-        icy = 415;
-    }
-    if (spellcoord > 3)
-    {
-        icy = 445;
-    }
-    draw_sprite(spr_heart, 0, xx + icx, yy + icy);
-    scr_84_set_draw_font("mainbig");
-    for (i = 0; i < 3; i += 1)
-    {
-        var currentspell = global.spell[thischar][(page * 6) + (i * 2)];
-        draw_set_color(c_white);
-        if (global.tension < global.spellcost[thischar][(page * 6) + (i * 2)])
-        {
-            draw_set_color(c_gray);
-        }
-        else if (currentspell == 3 || currentspell == 8)
-        {
-            pacify_glow = 0;
-            for (sglowi = 0; sglowi < 3; sglowi += 1)
-            {
-                if (global.monster[sglowi] == 1 && global.monsterstatus[sglowi] == 1)
-                {
-                    if (global.encounterno != 31)
-                    {
-                        pacify_glow = 1;
-                    }
-                }
-            }
-            if (pacify_glow == 1)
-            {
-                draw_set_color(merge_color(c_aqua, c_blue, 0.3));
-            }
-        }
-        draw_text(xx + 30, yy + 375 + (i * 30), string_hash_to_newline(global.spellnameb[thischar][(page * 6) + (i * 2)]));
-        draw_set_color(c_white);
-        if (global.tension < global.spellcost[thischar][(page * 6) + (i * 2) + 1])
-        {
-            draw_set_color(c_gray);
-        }
-        draw_text(xx + 260, yy + 375 + (i * 30), string_hash_to_newline(global.spellnameb[thischar][(page * 6) + (i * 2) + 1]));
-    }
-    draw_set_color(c_gray);
-    draw_text(xx + spell_offset, yy + 375, string_hash_to_newline(global.spelldescb[thischar][(page * 6) + spellcoord]));
-    thiscost = floor((global.spellcost[thischar][(page * 6) + spellcoord] / global.maxtension) * 100);
-    draw_set_color(c_orange);
-    draw_text(xx + spell_offset, yy + 440, string_hash_to_newline(string(thiscost) + "% TP"));
-}
 var havejeviltail = false;
-if (global.chararmor1[1] == 7 || global.chararmor2[1] == 7 || global.chararmor1[2] == 7 || global.chararmor2[2] == 7 || global.chararmor1[3] == 7 || global.chararmor2[3] == 7)
+if (scr_armorcheck_equipped_party(7))
 {
     havejeviltail = true;
 }
-if (global.bmenuno == 2 && global.myfight == 0 && global.flag[34] == 0)
+if (global.bmenuno == 2 && global.myfight == 0)
 {
     thischar = global.charturn;
     spellcoord = global.bmenucoord[2][global.charturn];
@@ -1092,6 +1030,14 @@
             actname[__acti] = global.actnamenoe[thisenemy][__acti];
             actdesc[__acti] = global.actdescnoe[thisenemy][__acti];
         }
+        if (oopsallacts == 1)
+        {
+            canact[__acti] = global.canact[thisenemy][__acti];
+            acttpcost[__acti] = global.actcost[thisenemy][__acti];
+            actsimul[__acti] = global.actsimul[thisenemy][__acti];
+            actname[__acti] = global.actname[thisenemy][__acti];
+            actdesc[__acti] = global.actdesc[thisenemy][__acti];
+        }
     }
     page = 0;
     icx = 10;
@@ -1140,21 +1086,22 @@
         draw_sprite(spr_heart, 0, xx + icx, yy + icy);
     }
     scr_84_set_draw_font("mainbig");
+    var hovered_available_result = new actactor_status(true);
     for (i = 0; i < 6; i += 1)
     {
         cant = 0;
         chartime = 0;
-        if (global.char[global.charturn] == 1)
+        if (global.char[global.charturn] == 1 || oopsallacts)
         {
             chartime = global.actactor[global.bmenucoord[11][global.charturn]][i];
         }
         charoffset = 0;
-        xoffset = 0;
+        var xoffset = 0;
         if (i == 1 || i == 3 || i == 5)
         {
             xoffset = 230;
         }
-        yoffset = 0;
+        var yoffset = 0;
         if (i == 2 || i == 3)
         {
             yoffset = 30;
@@ -1166,38 +1113,47 @@
         drawsus = 0;
         drawral = 0;
         drawnoe = 0;
-        susblend = 16777215;
-        ralblend = 16777215;
-        noeblend = 16777215;
-        if (chartime == 2 || chartime == 4)
-        {
-            drawsus = 1;
-            if (havechar[1] == 0 || global.hp[2] <= 0)
-            {
-                susblend = 8421504;
-                cant = 1;
-            }
-            charoffset = 30;
-        }
-        if (chartime == 3 || chartime == 4)
-        {
-            drawral = 1;
-            if (havechar[2] == 0 || global.hp[3] <= 0)
-            {
-                ralblend = 8421504;
-                cant = 1;
-            }
-            charoffset = 30;
-        }
-        if (chartime == 5)
-        {
-            drawnoe = 1;
-            if (havechar[3] == 0 || global.hp[4] <= 0)
-            {
-                noeblend = 8421504;
-                cant = 1;
-            }
-            charoffset = 30;
+        faceblend = 16777215;
+        switch (chartime)
+        {
+            case 2:
+                drawsus = 1;
+                charoffset = 30;
+                break;
+            case 3:
+                drawral = 1;
+                charoffset = 30;
+                break;
+            case 4:
+                drawsus = 1;
+                drawral = 1;
+                charoffset = 30;
+                break;
+            case 5:
+                drawnoe = 1;
+                charoffset = 30;
+                break;
+            case 6:
+            case 7:
+                drawsus = havechar[1] == 1;
+                drawral = havechar[2] == 1;
+                drawnoe = havechar[3] == 1;
+                charoffset = 0;
+                break;
+            case 8:
+            case 9:
+                charoffset = 30;
+                break;
+        }
+        var available_result = scr_actactor_available(chartime);
+        if (i == actcoord)
+        {
+            hovered_available_result = available_result;
+        }
+        if (!available_result.available)
+        {
+            cant = 1;
+            faceblend = 8421504;
         }
         if (global.tension < acttpcost[i])
         {
@@ -1235,20 +1191,135 @@
         }
         if (chartime == 2)
         {
-            draw_sprite_ext(spr_headsusie, 0, xx + 30 + xoffset, yy + 375 + yoffset, 1, 1, 0, susblend, 1);
+            draw_sprite_ext(spr_headsusie, 0, xx + 30 + xoffset, yy + 375 + yoffset, 1, 1, 0, faceblend, 1);
         }
         if (chartime == 3)
         {
-            draw_sprite_ext(spr_headralsei, 0, xx + 30 + xoffset, yy + 375 + yoffset, 1, 1, 0, susblend, 1);
+            draw_sprite_ext(spr_headralsei, 0, xx + 30 + xoffset, yy + 375 + yoffset, 1, 1, 0, faceblend, 1);
         }
         if (chartime == 4)
         {
-            draw_sprite_ext(spr_headsusie, 0, xx + 30 + xoffset, yy + 375 + yoffset, 1, 1, 0, susblend, 1);
+            draw_sprite_ext(spr_headsusie, 0, xx + 30 + xoffset, yy + 375 + yoffset, 1, 1, 0, faceblend, 1);
-            draw_sprite_ext(spr_headralsei, 0, xx + 60 + xoffset, yy + 375 + yoffset, 1, 1, 0, susblend, 1);
+            draw_sprite_ext(spr_headralsei, 0, xx + 60 + xoffset, yy + 375 + yoffset, 1, 1, 0, faceblend, 1);
         }
         if (chartime == 5)
         {
-            draw_sprite_ext(spr_headnoelle, 0, xx + 30 + xoffset, yy + 375 + yoffset, 1, 1, 0, noeblend, 1);
+            draw_sprite_ext(spr_headnoelle, 0, xx + 30 + xoffset, yy + 375 + yoffset, 1, 1, 0, faceblend, 1);
+        }
+        if (chartime == 6 || chartime == 7)
+        {
+            var curheadsprite = 2847;
+            for (var slot = 0; slot < 3; slot++)
+            {
+                var headsprite = spr_headkris;
+                switch (global.char[slot])
+                {
+                    case 1:
+                        headsprite = spr_headkris;
+                        break;
+                    case 2:
+                        headsprite = spr_headsusie;
+                        break;
+                    case 3:
+                        headsprite = spr_headralsei;
+                        break;
+                    case 4:
+                        headsprite = spr_headnoelle;
+                        break;
+                }
+                if (global.char[slot] > 0)
+                {
+                    if (slot != global.charturn)
+                    {
+                        charoffset += 30;
+                        draw_sprite_ext(headsprite, 0, xx + charoffset + xoffset, yy + 375 + yoffset, 1, 1, 0, faceblend, 1);
+                    }
+                    else
+                    {
+                        curheadsprite = headsprite;
+                    }
+                }
+            }
+            if (charoffset == 0 && chartime == 6)
+            {
+                charoffset += 30;
+                draw_sprite_ext(curheadsprite, 8, xx + charoffset + xoffset, yy + 375 + yoffset, 1, 1, 0, faceblend, 1);
+                cant = 1;
+            }
+        }
+        if (chartime == 8)
+        {
+            var slot;
+            if (global.charturn == 0)
+            {
+                slot = 1;
+            }
+            else
+            {
+                slot = 0;
+            }
+            var headsprite = spr_headkris;
+            switch (global.char[slot])
+            {
+                case 1:
+                    headsprite = spr_headkris;
+                    break;
+                case 2:
+                    headsprite = spr_headsusie;
+                    break;
+                case 3:
+                    headsprite = spr_headralsei;
+                    break;
+                case 4:
+                    headsprite = spr_headnoelle;
+                    break;
+            }
+            if (global.char[slot] > 0)
+            {
+                draw_sprite_ext(headsprite, 0, xx + 30 + xoffset, yy + 375 + yoffset, 1, 1, 0, faceblend, 1);
+            }
+            else
+            {
+                draw_sprite_ext(spr_headkris, 8, xx + 30 + xoffset, yy + 375 + yoffset, 1, 1, 0, faceblend, 1);
+                cant = 1;
+            }
+        }
+        if (chartime == 9)
+        {
+            var slot;
+            if (global.charturn == 2)
+            {
+                slot = 1;
+            }
+            else
+            {
+                slot = 2;
+            }
+            var headsprite = spr_headkris;
+            switch (global.char[slot])
+            {
+                case 1:
+                    headsprite = spr_headkris;
+                    break;
+                case 2:
+                    headsprite = spr_headsusie;
+                    break;
+                case 3:
+                    headsprite = spr_headralsei;
+                    break;
+                case 4:
+                    headsprite = spr_headnoelle;
+                    break;
+            }
+            if (global.char[slot] > 0)
+            {
+                draw_sprite_ext(headsprite, 0, xx + 30 + xoffset, yy + 375 + yoffset, 1, 1, 0, faceblend, 1);
+            }
+            else
+            {
+                draw_sprite_ext(spr_headkris, 8, xx + 30 + xoffset, yy + 375 + yoffset, 1, 1, 0, faceblend, 1);
+                cant = 1;
+            }
         }
         var s1 = string_hash_to_newline(actname[i]);
         var s1_width = max(1, string_width(s1));
@@ -1263,8 +1334,32 @@
         }
         draw_text_transformed(xx + 30 + charoffset + xoffset, yy + 375 + yoffset, string_hash_to_newline(actname[i]), s1_xscale, 1, 0);
     }
+    var displayed_actdesc = actdesc[actcoord];
+    switch (hovered_available_result.busy_char_name)
+    {
+        case undefined:
+            break;
+        case "kris":
+            displayed_actdesc = "Kris#is busy";
+            break;
+        case "susie":
+            displayed_actdesc = "Susie#is busy";
+            break;
+        case "ralsei":
+            displayed_actdesc = "Ralsei#is busy";
+            break;
+        case "noelle":
+            displayed_actdesc = "Noelle#is busy";
+            break;
+        case "everyone":
+            displayed_actdesc = "Everyone#is busy";
+            break;
+        default:
+            displayed_actdesc = "Someone#is busy";
+            break;
+    }
     draw_set_color(c_gray);
-    draw_text(xx + 500, yy + 375, string_hash_to_newline(actdesc[actcoord]));
+    draw_text(xx + 500, yy + 375, string_hash_to_newline(displayed_actdesc));
     if (global.tensionselect > 0)
     {
         thiscost = round((acttpcost[actcoord] / global.maxtension) * 100);
@@ -1500,6 +1595,163 @@
             global.facing = 3;
             scr_keyitemremove(18);
             room_goto(room_dw_ch3_man);
+        }
+    }
+}
+if (scr_debug())
+{
+    if (keyboard_check_pressed(vk_backspace))
+    {
+        drawactdebug = !drawactdebug;
+    }
+    if (drawactdebug)
+    {
+        draw_set_color(c_black);
+        draw_set_alpha(0.8);
+        draw_rectangle(xx + 240, yy, xx + 600, yy + 45, false);
+        draw_set_color(c_aqua);
+        draw_set_alpha(1);
+        scr_84_set_draw_font("main");
+        draw_text(xx + 250, yy + 5, "global.charturn: " + string(global.charturn));
+        draw_text(xx + 400, yy + 5, "global.currentactingchar: " + string(global.currentactingchar));
+        draw_text(xx + 250, yy + 20, "global.myfight: " + string(global.myfight));
+        draw_text(xx + 400, yy + 20, "scr_heroupcount: " + string(scr_heroupcount()));
+        draw_set_color(c_black);
+        draw_set_alpha(0.8);
+        draw_rectangle(xx + 65, yy, xx + 225, yy + 90, false);
+        draw_set_color(c_white);
+        draw_set_alpha(1);
+        draw_text(xx + 75, yy + 5, "ACT Globals");
+        draw_text(xx + 75, yy + 35, "acting: ");
+        draw_text(xx + 75, yy + 50, "actingsingle: ");
+        draw_text(xx + 75, yy + 65, "actingsimul: ");
+        for (var slot = 0; slot < 3; slot++)
+        {
+            switch (global.char[slot])
+            {
+                case 1:
+                    draw_set_color(c_aqua);
+                    break;
+                case 2:
+                    draw_set_color(c_fuchsia);
+                    break;
+                case 3:
+                    draw_set_color(c_lime);
+                    break;
+                case 4:
+                    draw_set_color(c_yellow);
+                    break;
+                default:
+                    draw_set_color(c_gray);
+                    break;
+            }
+            draw_text(xx + 165 + (15 * slot), yy + 35, string(global.acting[slot]));
+            draw_text(xx + 165 + (15 * slot), yy + 50, string(global.actingsingle[slot]));
+            draw_text(xx + 165 + (15 * slot), yy + 65, string(global.actingsimul[slot]));
+            var xoffset = 175;
+            var yoffset = 125 + (115 * slot);
+            var xoffset2 = xoffset + 200;
+            with (obj_monsterparent)
+            {
+                if (myself == slot)
+                {
+                    draw_set_color(c_black);
+                    draw_set_alpha(0.8);
+                    draw_rectangle((other.xx + xoffset) - 90, (other.yy + yoffset) - 10, (other.xx + xoffset) - 20, other.yy + yoffset + 25, false);
+                    draw_set_color(c_white);
+                    draw_set_alpha(1);
+                    draw_text((other.xx + xoffset) - 80, other.yy + yoffset, "Enemy " + string(slot));
+                    draw_set_color(c_black);
+                    draw_set_alpha(0.8);
+                    draw_rectangle((other.xx + xoffset) - 10, (other.yy + yoffset) - 10, other.xx + xoffset + 175, other.yy + yoffset + 100, false);
+                    draw_set_color(c_white);
+                    draw_set_alpha(1);
+                    draw_text(other.xx + xoffset, other.yy + yoffset, "Chosen ACTs");
+                    draw_text(other.xx + xoffset, other.yy + yoffset + 30, "acting:");
+                    draw_text(other.xx + xoffset, other.yy + yoffset + 45, "actcon:");
+                    draw_text(other.xx + xoffset, other.yy + yoffset + 60, "simulorder:");
+                    draw_text(other.xx + xoffset, other.yy + yoffset + 75, "simuldone:");
+                    draw_text(other.xx + xoffset, other.yy + yoffset + 90, "simulcount:");
+                    draw_set_color(c_yellow);
+                    draw_text(other.xx + xoffset + 90, other.yy + yoffset + 60, string(simulorder));
+                    draw_text(other.xx + xoffset + 90, other.yy + yoffset + 75, string(simuldone));
+                    draw_set_color(c_aqua);
+                    draw_text(other.xx + xoffset + 90, other.yy + yoffset + 30, string(acting));
+                    draw_text(other.xx + xoffset + 90, other.yy + yoffset + 45, string(actcon));
+                    draw_text(other.xx + xoffset + 90, other.yy + yoffset + 90, string(simulcountkri));
+                    draw_set_color(c_fuchsia);
+                    draw_text(other.xx + xoffset + 105, other.yy + yoffset + 30, string(actingsus));
+                    draw_text(other.xx + xoffset + 105, other.yy + yoffset + 45, string(actconsus));
+                    draw_text(other.xx + xoffset + 105, other.yy + yoffset + 90, string(simulcountsus));
+                    draw_set_color(c_lime);
+                    draw_text(other.xx + xoffset + 120, other.yy + yoffset + 30, string(actingral));
+                    draw_text(other.xx + xoffset + 120, other.yy + yoffset + 45, string(actconral));
+                    draw_text(other.xx + xoffset + 120, other.yy + yoffset + 90, string(simulcountral));
+                    draw_set_color(c_yellow);
+                    draw_text(other.xx + xoffset + 135, other.yy + yoffset + 30, string(actingnoe));
+                    draw_text(other.xx + xoffset + 135, other.yy + yoffset + 45, string(actconnoe));
+                    draw_text(other.xx + xoffset + 135, other.yy + yoffset + 90, string(simulcountnoe));
+                    draw_set_color(c_black);
+                    draw_set_alpha(0.8);
+                    draw_rectangle((other.xx + xoffset2) - 10, (other.yy + yoffset) - 10, other.xx + xoffset2 + 200, other.yy + yoffset + 85, false);
+                    draw_set_color(c_white);
+                    draw_set_alpha(1);
+                    draw_text(other.xx + xoffset2, other.yy + yoffset, "Available ACTs");
+                    draw_text(other.xx + xoffset2, other.yy + yoffset + 30, "canact:");
+                    draw_text(other.xx + xoffset2, other.yy + yoffset + 45, "actactor:");
+                    draw_text(other.xx + xoffset2, other.yy + yoffset + 60, "actsimul:");
+                    var acttotal = 0;
+                    for (var act = 0; act < 6; act++)
+                    {
+                        var actoffset = 75 + (15 * acttotal);
+                        if (global.canact[myself][act] == 0)
+                        {
+                            break;
+                        }
+                        draw_set_color(c_aqua);
+                        draw_text(other.xx + xoffset2 + actoffset, other.yy + yoffset + 30, string(global.canact[myself][act]));
+                        draw_text(other.xx + xoffset2 + actoffset, other.yy + yoffset + 45, string(global.actactor[myself][act]));
+                        draw_text(other.xx + xoffset2 + actoffset, other.yy + yoffset + 60, string(global.actsimul[myself][act]));
+                        acttotal++;
+                    }
+                    for (var act = 0; act < 6; act++)
+                    {
+                        var actoffset = 75 + (15 * acttotal);
+                        if (global.canactsus[myself][act] == 0)
+                        {
+                            break;
+                        }
+                        draw_set_color(c_fuchsia);
+                        draw_text(other.xx + xoffset2 + actoffset, other.yy + yoffset + 30, string(global.canactsus[myself][act]));
+                        draw_text(other.xx + xoffset2 + actoffset, other.yy + yoffset + 60, string(global.actsimulsus[myself][act]));
+                        acttotal++;
+                    }
+                    for (var act = 0; act < 6; act++)
+                    {
+                        var actoffset = 75 + (15 * acttotal);
+                        if (global.canactral[myself][act] == 0)
+                        {
+                            break;
+                        }
+                        draw_set_color(c_lime);
+                        draw_text(other.xx + xoffset2 + actoffset, other.yy + yoffset + 30, string(global.canactral[myself][act]));
+                        draw_text(other.xx + xoffset2 + actoffset, other.yy + yoffset + 60, string(global.actsimulral[myself][act]));
+                        acttotal++;
+                    }
+                    for (var act = 0; act < 6; act++)
+                    {
+                        var actoffset = 75 + (15 * acttotal);
+                        if (global.canactnoe[myself][act] == 0)
+                        {
+                            break;
+                        }
+                        draw_set_color(c_yellow);
+                        draw_text(other.xx + xoffset2 + actoffset, other.yy + yoffset + 30, string(global.canactnoe[myself][act]));
+                        draw_text(other.xx + xoffset2 + actoffset, other.yy + yoffset + 60, string(global.actsimulnoe[myself][act]));
+                        acttotal++;
+                    }
+                }
+            }
         }
     }
 }
