--- a/Code/gml_Object_obj_shutta_photo_controller_Other_13.gml
+++ b/Code/gml_Object_obj_shutta_photo_controller_Other_13.gml
@@ -1,5 +1,10 @@
+var krisx = -500;
+var krisy = -500;
+if (i_ex(obj_herokris))
+{
-var krisx = (obj_herokris.x - obj_herokris.sprite_xoffset) + (obj_herokris.sprite_width / 2);
+    krisx = (obj_herokris.x - obj_herokris.sprite_xoffset) + (obj_herokris.sprite_width / 2);
-var krisy = (obj_herokris.y - obj_herokris.sprite_yoffset) + (obj_herokris.sprite_height / 2);
+    krisy = (obj_herokris.y - obj_herokris.sprite_yoffset) + (obj_herokris.sprite_height / 2);
+}
 var susiex = -1000;
 var susiey = -1000;
 if (i_ex(obj_herosusie))
@@ -14,19 +19,31 @@
     ralseix = (obj_heroralsei.x - obj_heroralsei.sprite_xoffset) + (obj_heroralsei.sprite_width / 2);
     ralseiy = (obj_heroralsei.y - obj_heroralsei.sprite_yoffset) + (obj_heroralsei.sprite_height / 2);
 }
+var noellex = -3000;
+var noelley = -3000;
+if (i_ex(obj_heronoelle))
+{
+    noellex = (obj_heronoelle.x - obj_heronoelle.sprite_xoffset) + (obj_heronoelle.sprite_width / 2);
+    noelley = (obj_heronoelle.y - obj_heronoelle.sprite_yoffset) + (obj_heronoelle.sprite_height / 2);
+}
 var krisdist = point_distance(x, y, krisx, krisy);
 var susiedist = point_distance(x, y, susiex, susiey);
 var ralseidist = point_distance(x, y, ralseix, ralseiy);
-if (krisdist < susiedist && krisdist < ralseidist)
+var noelledist = point_distance(x, y, noellex, noelley);
+if (krisdist < susiedist && krisdist < ralseidist && krisdist < noelledist)
 {
     closest_hero = "kris";
 }
-if (susiedist < krisdist && susiedist < ralseidist)
+if (susiedist < krisdist && susiedist < ralseidist && susiedist < noelledist)
 {
     closest_hero = "susie";
 }
-if (ralseidist < susiedist && ralseidist < krisdist)
+if (ralseidist < susiedist && ralseidist < krisdist && ralseidist < noelledist)
 {
     closest_hero = "ralsei";
+}
+if (noelledist < krisdist && noelledist < susiedist && noelledist < ralseidist)
+{
+    closest_hero = "noelle";
 }
 
