--- a/Code/gml_Object_obj_elnina_lanino_controller_Step_0.gml
+++ b/Code/gml_Object_obj_elnina_lanino_controller_Step_0.gml
@@ -108,17 +108,21 @@
         }
         intro = 0.2;
     }
+    var ralseipos = scr_findchar(3);
     if (intro >= 0.2 && ralseishocked == false)
     {
         extimer++;
         if (extimer == 24 || introtimer == 5)
         {
-            scr_battle_sprite_set("ralsei", spr_ralsei_shocked_right, 0, 0);
-            if (i_ex(obj_heroralsei))
+            if (ralseipos > -1)
             {
-                obj_heroralsei.x += 30;
+                scr_battle_sprite_set(ralseipos, spr_ralsei_shocked_right, 0, 0);
+                if (i_ex(global.charinstance[ralseipos]))
+                {
+                    global.charinstance[ralseipos].x += 30;
+                }
+                ralseishocked = true;
             }
-            ralseishocked = true;
         }
     }
     if (intro == 0.2 && !i_ex(obj_writer))
@@ -128,15 +132,19 @@
     if (intro == 0.3)
     {
         introtimer++;
-        if (introtimer == 6)
+        if (introtimer == 6 && ralseishocked)
         {
             msgsetloc(0, "Ninety... Ni-", "obj_elnina_lanino_controller_slash_Step_0_gml_62_0");
             global.typer = 74;
-            var heroballoon = scr_enemyblcon(obj_heroralsei.x + 50, obj_heroralsei.y + 40, 10);
+            var shockedralsei = global.charinstance[ralseipos];
+            var heroballoon = scr_enemyblcon(global.charinstance[ralseipos].x + 50, global.charinstance[ralseipos].y + 40, 10);
             heroballoon.side = -1;
+        }
+        if (introtimer == 6)
+        {
             tenna.vspeed = 50;
         }
-        if (introtimer == 24)
+        if (introtimer == 24 || (introtimer == 6 && !ralseishocked))
         {
             intro = 1;
         }
@@ -156,10 +164,17 @@
     scr_speaker("tenna");
     global.typer = 80;
     global.fc = 22;
-    scr_battle_sprite_reset("ralsei");
-    if (i_ex(obj_heroralsei))
+    if (ralseishocked)
     {
-        obj_heroralsei.x -= 30;
+        var ralseipos = scr_findchar(3);
+        if (ralseipos > -1)
+        {
+            scr_battle_sprite_reset(ralseipos);
+            if (i_ex(global.charinstance[ralseipos]))
+            {
+                global.charinstance[ralseipos].x -= 30;
+            }
+        }
     }
 }
 if (ilovetv_increase > 0 && digit_flash_timer == 0)
@@ -198,75 +213,72 @@
         ilovetv_pitch = 1;
     }
 }
-if (scr_isphase("menu"))
+if (!calculated_forecast)
 {
-    if (!calculated_forecast)
-    {
-        turn2_turns++;
+    turn2_turns++;
-        if (ilovetv > 0)
+    if (ilovetv > 0)
-        {
+    {
-            if (turn2_turns > 2 && turns < 2)
+        if (turn2_turns > 2 && turns < 2)
-            {
+        {
-                turns = 2;
+            turns = 2;
-            }
         }
+    }
-        else if (turns > 1 && ilovetv == 0 && turn2_turns < 7)
+    else if (turns > 1 && ilovetv == 0 && turn2_turns < 7)
-        {
+    {
-            turns = 0;
+        turns = 0;
-        }
+    }
-        ilovetv_increased = false;
+    ilovetv_increased = false;
-        forecast_turn = turns;
+    forecast_turn = turns;
-        var _sun = 783;
+    var _sun = 783;
-        var _moon = 1474;
+    var _moon = 1474;
-        var _rain = 4256;
+    var _rain = 4256;
-        var _snow = 1650;
+    var _snow = 1650;
-        var _hale = 14;
+    var _hale = 14;
-        var _temp_min = 50;
+    var _temp_min = 50;
-        var _temp_max = 60;
+    var _temp_max = 60;
-        switch (forecast_turn)
+    switch (forecast_turn)
-        {
+    {
-            case 0:
+        case 0:
-                forecasts[0] = [irandom_range(_temp_min, _temp_max), _moon, _snow];
+            forecasts[0] = [irandom_range(_temp_min, _temp_max), _moon, _snow];
-                forecasts[1] = [irandom_range(_temp_min, _temp_max), _moon, _snow];
+            forecasts[1] = [irandom_range(_temp_min, _temp_max), _moon, _snow];
-                forecasts[2] = [irandom_range(_temp_min, _temp_max), _moon, _snow];
+            forecasts[2] = [irandom_range(_temp_min, _temp_max), _moon, _snow];
-                break;
+            break;
-            case 1:
+        case 1:
-                forecasts[0] = [irandom_range(_temp_min, _temp_max), _sun, _rain];
+            forecasts[0] = [irandom_range(_temp_min, _temp_max), _sun, _rain];
-                forecasts[1] = [irandom_range(_temp_min, _temp_max), _sun, _rain];
+            forecasts[1] = [irandom_range(_temp_min, _temp_max), _sun, _rain];
-                forecasts[2] = [irandom_range(_temp_min, _temp_max), _sun, _rain];
+            forecasts[2] = [irandom_range(_temp_min, _temp_max), _sun, _rain];
-                break;
+            break;
-            case 2:
+        case 2:
+            forecasts[0] = [irandom_range(_temp_min, _temp_max), _moon, _rain];
+            forecasts[1] = [irandom_range(_temp_min, _temp_max), _moon, _rain];
+            forecasts[2] = [irandom_range(_temp_min, _temp_max), _moon, _rain];
+            break;
+        default:
+            if (lastchosen == "lanino")
+            {
+                _temp_min = 80 + ((forecast_turn - 3) * 100);
+                _temp_max = 89 + ((forecast_turn - 3) * 100);
+            }
+            else
+            {
+                _temp_min = 40 - ((forecast_turn - 3) * 30);
+                _temp_max = 50 - ((forecast_turn - 3) * 30);
+            }
+            if (choose(0, 1) == 1)
+            {
+                forecasts[0] = [irandom_range(_temp_min, _temp_max), _sun, _snow];
+                forecasts[1] = [irandom_range(_temp_min, _temp_max), _sun, _snow];
+                forecasts[2] = [irandom_range(_temp_min, _temp_max), _sun, _snow];
+            }
+            else
+            {
                 forecasts[0] = [irandom_range(_temp_min, _temp_max), _moon, _rain];
                 forecasts[1] = [irandom_range(_temp_min, _temp_max), _moon, _rain];
                 forecasts[2] = [irandom_range(_temp_min, _temp_max), _moon, _rain];
-                break;
-            default:
-                if (lastchosen == "lanino")
-                {
-                    _temp_min = 80 + ((forecast_turn - 3) * 100);
-                    _temp_max = 89 + ((forecast_turn - 3) * 100);
-                }
-                else
-                {
-                    _temp_min = 40 - ((forecast_turn - 3) * 30);
-                    _temp_max = 50 - ((forecast_turn - 3) * 30);
-                }
-                if (choose(0, 1) == 1)
-                {
-                    forecasts[0] = [irandom_range(_temp_min, _temp_max), _sun, _snow];
-                    forecasts[1] = [irandom_range(_temp_min, _temp_max), _sun, _snow];
-                    forecasts[2] = [irandom_range(_temp_min, _temp_max), _sun, _snow];
-                }
-                else
-                {
-                    forecasts[0] = [irandom_range(_temp_min, _temp_max), _moon, _rain];
-                    forecasts[1] = [irandom_range(_temp_min, _temp_max), _moon, _rain];
-                    forecasts[2] = [irandom_range(_temp_min, _temp_max), _moon, _rain];
-                }
-                break;
-        }
-        calculated_forecast = true;
+            }
+            break;
     }
+    calculated_forecast = true;
 }
 var lanino = obj_lanino_enemy;
 var elnina = obj_elnina_enemy;
@@ -275,7 +287,7 @@
 var _moon_atk = 69;
 var _snow_atk = 70;
 var _rain_atk = 71;
-if (scr_isphase("enemytalk") && talked == 0 && !i_ex(obj_rouxls_ch3_enemy))
+if (scr_isphase("enemytalk") && talked == 0 && !i_ex(obj_rouxls_ch3_enemy) && intro >= 4)
 {
     scr_randomtarget();
     if (!instance_exists(obj_darkener))
