--- a/Code/gml_GlobalScript_scr_actinfo_temp.gml
+++ b/Code/gml_GlobalScript_scr_actinfo_temp.gml
@@ -27,6 +27,12 @@
             acttpcost[__acti] = global.actcostnoe[arg0][__acti];
             actsimul[__acti] = global.actsimulnoe[arg0][__acti];
         }
+        if (oopsallacts)
+        {
+            canact[__acti] = global.canact[arg0][__acti];
+            acttpcost[__acti] = global.actcost[arg0][__acti];
+            actsimul[__acti] = global.actsimul[arg0][__acti];
+        }
     }
 }
 
