--- a/Code/gml_Object_obj_shadowman_enemy_Step_0.gml
+++ b/Code/gml_Object_obj_shadowman_enemy_Step_0.gml
@@ -3,26 +3,55 @@
     init = 1;
     if (i_ex(obj_tenna_board4_enemy))
     {
-        global.actactor[0][0] = 4;
+        global.actactor[0][0] = 7;
-        global.actactor[0][1] = 4;
+        global.actactor[0][1] = 7;
         scr_spellmenu_setup();
     }
 }
 if (global.monster[myself] == 0 && krisdancing)
 {
-    scr_battle_sprite_reset("kris");
+    with (obj_herokris)
+    {
+        if (sprite_index == spr_kris_dance)
+        {
+            scr_battle_sprite_reset(myself);
+        }
+    }
     krisdancing = false;
 }
 if (global.monster[myself] == 0 && susiedancing)
 {
-    scr_battle_sprite_reset("susie");
+    with (obj_herosusie)
+    {
+        if (sprite_index == spr_susie_dance)
+        {
+            scr_battle_sprite_reset(myself);
+        }
+    }
     susiedancing = false;
 }
 if (global.monster[myself] == 0 && ralseidancing)
 {
-    scr_battle_sprite_reset("ralsei");
+    with (obj_heroralsei)
+    {
+        if (sprite_index == spr_ralsei_dance)
+        {
+            scr_battle_sprite_reset(myself);
+        }
+    }
     ralseidancing = false;
 }
+if (global.monster[myself] == 0 && noelledancing)
+{
+    with (obj_heronoelle)
+    {
+        if (sprite_index == spr_noelle_dance)
+        {
+            scr_battle_sprite_reset(myself);
+        }
+    }
+    noelledancing = false;
+}
 if (global.monster[myself] == 1)
 {
     var rouxlswait = false;
@@ -206,11 +235,17 @@
             target = 0;
         }
     }
-    if ((global.mnfight == 2 && (savekrishp > global.hp[1] || savesusiehp > global.hp[2] || saveralseihp > global.hp[3]) && gainmercyovertime == 0) || (global.mnfight == 2 && global.turntimer <= 1 && gainmercyovertime == 0))
+    if ((global.mnfight == 2 && (savekrishp > global.hp[1] || savesusiehp > global.hp[2] || saveralseihp > global.hp[3] || savenoellehp > global.hp[4]) && gainmercyovertime == 0) || (global.mnfight == 2 && global.turntimer <= 1 && gainmercyovertime == 0))
     {
         gainmercyovertime = 1;
         gainmercytimer = 0;
-        scr_battle_sprite_reset("kris");
+        with (obj_herokris)
+        {
+            if (sprite_index == spr_kris_dance)
+            {
+                scr_battle_sprite_reset(myself);
+            }
+        }
         krisdancing = false;
         timberb = 15;
     }
@@ -231,12 +266,13 @@
         {
             instance_destroy();
         }
-        scr_battle_sprite_reset("kris");
+        scr_battle_sprite_reset(0);
+        scr_battle_sprite_reset(1);
+        scr_battle_sprite_reset(2);
         krisdancing = false;
-        scr_battle_sprite_reset("susie");
         susiedancing = false;
-        scr_battle_sprite_reset("ralsei");
         ralseidancing = false;
+        noelledancing = false;
     }
     if (global.mnfight > 0 && !i_ex(obj_rouxls_ch3_enemy))
     {
@@ -249,9 +285,9 @@
             if (sprite_index == spr_susie_dance)
             {
                 susiedancing = false;
-                scr_battle_sprite_reset("susie");
-                obj_herosusie.x = obj_herosusie.xstart;
-                obj_herosusie.y = obj_herosusie.ystart;
+                scr_battle_sprite_reset(myself);
+                x = xstart;
+                y = ystart;
             }
         }
         with (obj_heroralsei)
@@ -259,11 +295,19 @@
             if (sprite_index == spr_ralsei_dance)
             {
                 ralseidancing = false;
-                scr_battle_sprite_reset("ralsei");
+                scr_battle_sprite_reset(myself);
+            }
+        }
+        with (obj_heronoelle)
+        {
+            if (sprite_index == spr_noelle_dance)
+            {
+                noelledancing = false;
+                scr_battle_sprite_reset(myself);
             }
         }
     }
-    if (gainmercyovertime == 0 && savekrishp <= global.hp[1] && savesusiehp <= global.hp[2] && saveralseihp <= global.hp[3])
+    if (gainmercyovertime == 0 && (scr_havechar(1) && savekrishp <= global.hp[1]) && (scr_havechar(2) && savesusiehp <= global.hp[2]) && (scr_havechar(3) && saveralseihp <= global.hp[3]) && (scr_havechar(4) && savenoellehp <= global.hp[4]))
     {
         if (global.mnfight == 2 && global.turntimer > 1 && i_ex(obj_dmgwriter_boogie))
         {
@@ -323,8 +367,8 @@
         __mercydmgwriter.damage = 5;
         __mercydmgwriter.type = 5;
         global.hittarget[myself]++;
-        scr_battle_sprite_set("kris", spr_kris_dance, 0.25, true);
-        scr_battle_sprite_actflash("kris");
+        var krispos = scr_battle_sprite_set_simul("kris", 455, 0.25, true);
+        scr_battle_sprite_actflash(krispos);
         idlesprite = spr_shadowman_sax_b;
         msgsetloc(0, "* Kris boogies past bullets!&* SHADOWGUY gains mercy until you get hit!/%", "obj_shadowman_enemy_slash_Step_0_gml_289_0");
         scr_battletext_default();
@@ -332,6 +376,7 @@
         savekrishp = global.hp[1];
         savesusiehp = global.hp[2];
         saveralseihp = global.hp[3];
+        savenoellehp = global.hp[4];
         krisdancing = true;
     }
     if (acting == 3 && actcon == 0)
@@ -354,14 +399,33 @@
             msgsetloc(0, "* Everyone gets ready to knock the enemies' socks off!/%", "obj_shadowman_enemy_slash_Step_0_gml_304_0");
         }
         scr_battletext_default();
-        obj_herokris.x -= 8;
-        obj_herokris.y -= 12;
-        obj_herosusie.x -= 8;
-        obj_herosusie.y -= 6;
-        obj_heroralsei.x -= 4;
-        scr_act_charsprite("ralsei", spr_ralsei_zoosuit, 0, 0);
-        scr_act_charsprite("susie", spr_susie_zoosuit, 0, 0);
-        scr_act_charsprite("kris", spr_kris_zoosuit, 0, 0);
+        for (var i = 0; i < 3; i++)
+        {
+            if (!instance_exists(global.charinstance[i]))
+            {
+                continue;
+            }
+            switch (global.char[i])
+            {
+                case 1:
+                    global.charinstance[i].x -= 8;
+                    global.charinstance[i].y -= 12;
+                    scr_act_charsprite(i, spr_kris_zoosuit, 0, 0);
+                    break;
+                case 2:
+                    global.charinstance[i].x -= 8;
+                    global.charinstance[i].y -= 6;
+                    scr_act_charsprite(i, spr_susie_zoosuit, 0, 0);
+                    break;
+                case 3:
+                    global.charinstance[i].x -= 4;
+                    scr_act_charsprite(i, spr_ralsei_zoosuit, 0, 0);
+                    break;
+                case 4:
+                    scr_act_charsprite(i, spr_noelle_zoosuit, 0, 0);
+                    break;
+            }
+        }
         snd_play(snd_boost);
         with (obj_tenna_board4_gacha)
         {
@@ -470,9 +534,12 @@
         sharpshoottimer++;
         if (sharpshoottimer >= 14)
         {
+            if (instance_exists(obj_herokris))
+            {
-            obj_herokris.state = 8;
+                obj_herokris.state = 8;
-            obj_herokris.image_speed = 0;
+                obj_herokris.image_speed = 0;
-            obj_herokris.image_index = 7;
+                obj_herokris.image_index = 7;
+            }
         }
         with (obj_monsterparent)
         {
@@ -539,25 +606,21 @@
             {
                 stopshooting = true;
             }
-            for (var i = 0; i < 10; i++)
-            {
-                with (_charactsprite[i])
-                {
-                    instance_destroy();
-                }
-            }
-            with (obj_heroparent)
-            {
-                image_alpha = 1;
-                acttimer = 0;
-                state = 0;
-                global.faceaction[myself] = 0;
-            }
-            obj_herokris.x += 8;
-            obj_herokris.y += 12;
-            obj_herosusie.x += 8;
-            obj_herosusie.y += 6;
-            obj_heroralsei.x += 4;
+            scr_act_charsprite_end();
+            with (obj_herokris)
+            {
+                x += 8;
+                y += 12;
+            }
+            with (obj_herosusie)
+            {
+                x += 8;
+                y += 6;
+            }
+            with (obj_heroralsei)
+            {
+                x += 4;
+            }
             with (obj_tenna_board4_gacha)
             {
                 sharpshoot = false;
@@ -762,8 +825,8 @@
         {
             actconsus = 0;
         }
-        scr_battle_sprite_set("susie", spr_susie_dance, 0.25, true);
-        scr_battle_sprite_actflash("susie");
+        var susiepos = scr_battle_sprite_set_simul("susie", 391, 0.25, true);
+        scr_battle_sprite_actflash(susiepos);
         idlesprite = spr_shadowman_sax_b;
         msgsetloc(0, "* Susie danced!/%", "obj_shadowman_enemy_slash_Step_0_gml_575_0");
         if (global.monsterstatus[myself] != 1)
@@ -784,8 +847,8 @@
         {
             actconral = 0;
         }
-        scr_battle_sprite_set("ralsei", spr_ralsei_dance, 0.25, true);
-        scr_battle_sprite_actflash("ralsei");
+        var ralseipos = scr_battle_sprite_set_simul("ralsei", 316, 0.25, true);
+        scr_battle_sprite_actflash(ralseipos);
         idlesprite = spr_shadowman_sax_b;
         msgsetloc(0, "* Ralsei danced!/%", "obj_shadowman_enemy_slash_Step_0_gml_593_0");
         if (global.monsterstatus[myself] != 1)
@@ -794,12 +857,36 @@
         }
         scr_simultext("ralsei");
     }
+    if (actingnoe == 1 && actconnoe == 1)
+    {
+        noelledancing = true;
+        scr_mercyadd(myself, 30);
+        if (simulordernoe == 0)
+        {
+            actconnoe = 20;
+        }
+        else
+        {
+            actconnoe = 0;
+        }
+        var noellepos = scr_battle_sprite_set_simul("noelle", 4992, 0.25, true);
+        scr_battle_sprite_actflash(noellepos);
+        idlesprite = spr_shadowman_sax_b;
+        msgset(0, "* Noelle danced!/%");
+        if (global.monsterstatus[myself] != 1)
+        {
+            scr_monster_make_tired(myself);
+        }
+        scr_simultext("noelle");
+    }
-    if (actcon == 20 || actconsus == 20 || actconral == 20)
+    if (actcon == 20 || actconsus == 20 || actconral == 20 || actconnoe == 20)
     {
+        scr_nextact_simul();
         if (scr_terminate_writer())
         {
             actconsus = -1;
             actconral = -1;
+            actconnoe = -1;
             actcon = 1;
         }
     }
