--- a/Code/gml_Object_obj_ch2_cyber01_Step_0.gml
+++ b/Code/gml_Object_obj_ch2_cyber01_Step_0.gml
@@ -350,10 +350,16 @@
     obj_mainchara.cutscene = 1;
     obj_mainchara.x = kr_actor.x;
     obj_mainchara.y = kr_actor.y;
+    if (i_ex(global.cinstance[0]))
+    {
-    global.cinstance[0].x = su_actor.x;
+        global.cinstance[0].x = su_actor.x;
-    global.cinstance[0].y = su_actor.y;
+        global.cinstance[0].y = su_actor.y;
+    }
+    if (i_ex(global.cinstance[1]))
+    {
-    global.cinstance[1].x = ra_actor.x;
+        global.cinstance[1].x = ra_actor.x;
-    global.cinstance[1].y = ra_actor.y;
+        global.cinstance[1].y = ra_actor.y;
+    }
     global.flag[9] = 2;
     global.batmusic[0] = -1;
     snd_init("spamton_neo_mix_ex_wip.ogg");
@@ -387,15 +393,30 @@
 {
     con = 34;
     customcon = 0;
-    kr_actor.x = obj_herokris.x;
-    kr_actor.y = obj_herokris.y;
-    kr_actor.sprite_index = spr_krisr_dark;
-    su_actor.x = obj_herosusie.x;
-    su_actor.y = obj_herosusie.y;
-    su_actor.sprite_index = spr_susie_walk_right_dw;
-    ra_actor.x = obj_heroralsei.x;
-    ra_actor.y = obj_heroralsei.y;
-    ra_actor.sprite_index = spr_ralsei_walk_right;
+    if (i_ex(obj_herokris))
+    {
+        kr_actor.x = obj_herokris.x;
+        kr_actor.y = obj_herokris.y;
+        kr_actor.sprite_index = spr_krisr_dark;
+    }
+    if (i_ex(obj_herosusie))
+    {
+        su_actor.x = obj_herosusie.x;
+        su_actor.y = obj_herosusie.y;
+        su_actor.sprite_index = spr_susie_walk_right_dw;
+    }
+    if (i_ex(obj_heroralsei))
+    {
+        ra_actor.x = obj_heroralsei.x;
+        ra_actor.y = obj_heroralsei.y;
+        ra_actor.sprite_index = spr_ralsei_walk_right;
+    }
+    if (i_ex(obj_heronoelle))
+    {
+        no_actor.x = obj_heronoelle.x;
+        no_actor.y = obj_heronoelle.y;
+        no_actor.sprite_index = spr_noelle_walk_right_dw;
+    }
     sw_actor.x = obj_sweet_enemy.x;
     sw_actor.y = obj_sweet_enemy.y;
     sw_actor.sprite_index = obj_sweet_enemy.sprite_index;
@@ -413,6 +434,7 @@
     {
         visible = 1;
     }
+    scr_hidecharactor();
     with (obj_mainchara)
     {
         visible = 0;
@@ -458,6 +480,7 @@
     {
         visible = 1;
     }
+    scr_hidecharactor();
     with (obj_mainchara)
     {
         visible = 0;
