--- a/Code/gml_Object_obj_controller_dw_city_big_2_Step_0.gml
+++ b/Code/gml_Object_obj_controller_dw_city_big_2_Step_0.gml
@@ -31,6 +31,11 @@
     else if (!i_ex(forcefield))
     {
         forcefield = scr_forcefield(680, 1040, 2, 8, 1, 1);
+        if (obj_mainchara.x > (forcefield.x - 40))
+        {
+            obj_mainchara.x = forcefield.x - 40;
+            snd_play(snd_cantselect);
+        }
     }
 }
 toggledtimer++;
@@ -141,15 +146,23 @@
     }
     if (con == 6)
     {
+        if (instance_exists(obj_caterpillarchara))
+        {
-        noellePuzzle = instance_create(obj_caterpillarchara.x, obj_caterpillarchara.y, obj_marker);
+            noellePuzzle = instance_create(obj_caterpillarchara.x, obj_caterpillarchara.y, obj_marker);
+            noellePuzzle.depth = obj_caterpillarchara.depth;
+            obj_caterpillarchara.visible = false;
+        }
+        else
+        {
+            noellePuzzle = instance_create(758, 1064, obj_marker);
+            noellePuzzle.depth = obj_mainchara.depth + 10;
+        }
         noellePuzzle.sprite_index = spr_noelle_walk_right_dw;
         noellePuzzle.image_xscale = 2;
         noellePuzzle.image_yscale = 2;
         noellePuzzle.image_speed = 0.25;
         noellePuzzle.starty = noellePuzzle.y;
         noellePuzzle.startx = noellePuzzle.x;
-        noellePuzzle.depth = obj_caterpillarchara.depth;
-        obj_caterpillarchara.visible = false;
         con++;
     }
     if (con == 7)
@@ -617,12 +630,20 @@
     {
         if (!i_ex(d))
         {
+            if (instance_exists(obj_caterpillarchara))
+            {
-            noellePuzzle = instance_create(obj_caterpillarchara.x, obj_caterpillarchara.y, obj_marker);
+                noellePuzzle = instance_create(obj_caterpillarchara.x, obj_caterpillarchara.y, obj_marker);
+                noellePuzzle.sprite_index = obj_caterpillarchara.sprite_index;
+                obj_caterpillarchara.visible = false;
+            }
+            else
+            {
+                noellePuzzle = instance_create(476, 1054, obj_marker);
+                noellePuzzle.sprite_index = spr_noelle_walk_right_dw;
+            }
             noellePuzzle.image_xscale = 2;
             noellePuzzle.image_yscale = 2;
-            noellePuzzle.sprite_index = obj_caterpillarchara.sprite_index;
             noellePuzzle.image_speed = 0;
-            obj_caterpillarchara.visible = false;
             con++;
         }
     }
@@ -784,15 +805,12 @@
         timer++;
         if (noellePuzzle.x == newtargx && noellePuzzle.y == newtargy)
         {
-            obj_caterpillarchara.x = newtargx;
-            obj_caterpillarchara.y = newtargy;
-            obj_caterpillarchara.visible = true;
             with (obj_caterpillarchara)
             {
+                x = other.newtargx;
+                y = other.newtargy;
+                visible = true;
                 scr_caterpillar_facing(1);
-            }
-            with (obj_caterpillarchara)
-            {
                 scr_caterpillar_interpolate();
             }
             noellePuzzle.visible = false;
