--- a/Code/gml_Object_obj_battlecontroller_Create_0.gml
+++ b/Code/gml_Object_obj_battlecontroller_Create_0.gml
@@ -113,6 +113,24 @@
 global.monstergold[3] = 0;
 global.monsterexp[3] = 0;
 global.charturn = 0;
+if (global.charmove[0] == 0)
+{
+    global.charturn = 1;
+}
+if (global.charturn == 1)
+{
+    if (global.charmove[1] == 0)
+    {
+        global.charturn = 2;
+    }
+}
+if (global.charturn == 2)
+{
+    if (global.charmove[2] == 0)
+    {
+        scr_attackphase();
+    }
+}
 global.currentactingchar = 0;
 for (i = 0; i < 3; i += 1)
 {
@@ -121,6 +139,7 @@
     global.actingsimul[i] = 0;
     global.actingtarget[i] = 0;
     global.actingchoice[i] = 0;
+    global.actwaiting[i] = 0;
     global.charaction[i] = 0;
     global.charspecial[i] = 0;
     global.chartarget[i] = 0;
