--- a/Code/gml_Object_obj_ch2_city_noelle_monologue_Step_0.gml
+++ b/Code/gml_Object_obj_ch2_city_noelle_monologue_Step_0.gml
@@ -31,6 +31,21 @@
     global.flag[7] = 1;
     cutscene_master = scr_cutscene_make();
     scr_maincharacters_actors();
+    with (su_actor)
+    {
+        instance_destroy();
+    }
+    with (ra_actor)
+    {
+        instance_destroy();
+    }
+    with (obj_caterpillarchara)
+    {
+        if (visible == 0 && name != "noelle")
+        {
+            visible = 1;
+        }
+    }
     no_silhouette = instance_create(no_actor.x, cameray() - 100, obj_following_silhouette);
     no_silhouette.target_char = no_actor;
     var exit_forcefield = instance_create(20, 240, obj_forcefield);
