--- a/Code/gml_GlobalScript_scr_act_simul.gml
+++ b/Code/gml_GlobalScript_scr_act_simul.gml
@@ -6,40 +6,69 @@
         __foundsimul = 0;
         if (global.actingsingle[__ii] == 1 && instance_exists(obj_monsterparent))
         {
+            var acttarget = global.monsterinstance[global.actingtarget[__ii]];
             if (global.char[__ii] == 1 && global.actsimul[global.actingtarget[__ii]][global.actingchoice[__ii]] == 1)
             {
-                obj_monsterparent.simulorderkri = __simulcount;
-                with (global.monsterinstance[global.actingtarget[__ii]])
+                acttarget.simulorder[__ii] = __simulcount;
+                acttarget.simulorderkri = 0;
+                if (acttarget.simulcountkri == 0)
                 {
+                    var actchoice = global.actingchoice[__ii];
+                    with (acttarget)
+                    {
+                        acting = actchoice + 1;
-                    actcon = 0;
+                        actcon = 0;
+                    }
                 }
+                acttarget.simulcountkri++;
                 __foundsimul = 1;
             }
             if (global.char[__ii] == 2 && global.actsimulsus[global.actingtarget[__ii]][global.actingchoice[__ii]] == 1)
             {
-                obj_monsterparent.simulordersus = __simulcount;
-                with (global.monsterinstance[global.actingtarget[__ii]])
+                acttarget.simulorder[__ii] = __simulcount;
+                acttarget.simulordersus = 0;
+                if (acttarget.simulcountsus == 0)
                 {
+                    var actchoice = global.actingchoice[__ii];
+                    with (acttarget)
+                    {
+                        actingsus = actchoice + 1;
-                    actconsus = 1;
+                        actconsus = 1;
+                    }
                 }
+                acttarget.simulcountsus++;
                 __foundsimul = 1;
             }
             if (global.char[__ii] == 3 && global.actsimulral[global.actingtarget[__ii]][global.actingchoice[__ii]] == 1)
             {
-                obj_monsterparent.simulorderral = __simulcount;
-                with (global.monsterinstance[global.actingtarget[__ii]])
+                acttarget.simulorder[__ii] = __simulcount;
+                acttarget.simulorderral = 0;
+                if (acttarget.simulcountral == 0)
                 {
+                    var actchoice = global.actingchoice[__ii];
+                    with (acttarget)
+                    {
+                        actingral = actchoice + 1;
-                    actconral = 1;
+                        actconral = 1;
+                    }
                 }
+                acttarget.simulcountral++;
                 __foundsimul = 1;
             }
             if (global.char[__ii] == 4 && global.actsimulnoe[global.actingtarget[__ii]][global.actingchoice[__ii]] == 1)
             {
-                obj_monsterparent.simulordernoe = __simulcount;
-                with (global.monsterinstance[global.actingtarget[__ii]])
+                acttarget.simulorder[__ii] = __simulcount;
+                acttarget.simulordernoe = 0;
+                if (acttarget.simulcountnoe == 0)
                 {
+                    var actchoice = global.actingchoice[__ii];
+                    with (acttarget)
+                    {
+                        actingnoe = actchoice + 1;
-                    actconnoe = 1;
+                        actconnoe = 1;
+                    }
                 }
+                acttarget.simulcountnoe++;
                 __foundsimul = 1;
             }
             if (__foundsimul)
