--- a/Code/gml_Object_obj_kk_enemy_Step_0.gml
+++ b/Code/gml_Object_obj_kk_enemy_Step_0.gml
@@ -183,7 +183,14 @@
             }
         }
         danceCounter = 1;
+        if (global.lang != "ja")
+        {
+            scr_act_charsprite_simul("kris", 867, 0.16, 1);
+        }
+        else
+        {
-        scr_act_charsprite("kris", spr_kris_dance, 0.16, 1);
+            scr_act_charsprite(global.currentactingchar, spr_kris_dance, 0.16, 1);
+        }
         if (simultotal == 1 || (global.lang == "ja" && obj_sweet_enemy.simultotal_funny == 1))
         {
             msgsetloc(0, "* You danced!/", "obj_kk_enemy_slash_Step_0_gml_172_0");
@@ -241,9 +248,24 @@
             }
         }
         danceCounter = 2;
-        scr_act_charsprite("kris", spr_kris_dance, 0.16, 1);
-        scr_act_charsprite("ralsei", spr_ralsei_dance, 0.16, 1);
-        scr_act_charsprite("susie", spr_susie_dance, 0.16, 1);
+        for (var i = 0; i < 3; i++)
+        {
+            switch (global.char[i])
+            {
+                case 1:
+                    scr_act_charsprite(i, spr_kris_dance, 0.16, 1);
+                    break;
+                case 2:
+                    scr_act_charsprite(i, spr_susie_dance, 0.16, 1);
+                    break;
+                case 3:
+                    scr_act_charsprite(i, spr_ralsei_dance, 0.16, 1);
+                    break;
+                case 4:
+                    scr_act_charsprite(i, spr_noelle_dance, 0.16, 1);
+                    break;
+            }
+        }
         msgsetloc(0, "* Everyone danced with K_K! They're totally lost in the groove!/%", "obj_kk_enemy_slash_Step_0_gml_221_0");
         scr_battletext_default();
         actcon = 4;
@@ -258,7 +280,14 @@
     {
         dancing = true;
         danceCounter = 1;
+        if (global.lang != "ja")
+        {
+            scr_act_charsprite_simul("susie", 823, 0.16, 1);
+        }
+        else
+        {
-        scr_act_charsprite("susie", spr_susie_dance, 0.16, 1);
+            scr_act_charsprite(global.currentactingchar, spr_susie_dance, 0.16, 1);
+        }
         if (groove == 0)
         {
             if (global.lang == "ja")
@@ -298,7 +327,14 @@
     {
         dancing = true;
         danceCounter = 1;
+        if (global.lang != "ja")
+        {
+            scr_act_charsprite_simul("ralsei", 783, 0.16, 1);
+        }
+        else
+        {
-        scr_act_charsprite("ralsei", spr_ralsei_dance, 0.16, 1);
+            scr_act_charsprite(global.currentactingchar, spr_ralsei_dance, 0.16, 1);
+        }
         if (groove == 0)
         {
             if (global.lang == "ja")
@@ -334,6 +370,53 @@
             }
         }
     }
+    if (actingnoe == 1 && actconnoe == 1)
+    {
+        dancing = true;
+        danceCounter = 1;
+        if (global.lang != "ja")
+        {
+            scr_act_charsprite_simul("noelle", 3383, 0.16, 1);
+        }
+        else
+        {
+            scr_act_charsprite(global.currentactingchar, spr_noelle_dance, 0.16, 1);
+        }
+        if (groove == 0)
+        {
+            if (global.lang == "ja")
+            {
+                msgset(0, "* Noelle danced with K_K!/");
+                msgnext("* K_K got lost in the groove anyway!/%");
+            }
+            else
+            {
+                msgset(0, "* Noelle danced with K_K!/%");
+            }
+            groove = 1;
+        }
+        else
+        {
+            msgset(0, "* Noelle danced with K_K!/%");
+        }
+        if (global.lang == "ja")
+        {
+            scr_battletext_default();
+            actconnoe = 20;
+        }
+        else
+        {
+            scr_simultext("noelle");
+            if (simulordernoe == 0)
+            {
+                actconnoe = 20;
+            }
+            else
+            {
+                actconnoe = 0;
+            }
+        }
+    }
     if (actcon == 1 && !instance_exists(obj_writer))
     {
         if (obj_sweet_enemy.dancing == true && obj_kk_enemy.dancing == true && obj_hatguy_enemy.dancing == true && obj_sweet_enemy.endcon == 0)
@@ -346,7 +429,7 @@
             acting = 0;
             with (obj_marker)
             {
-                if (sprite_index == spr_kris_dance || sprite_index == spr_susie_dance || sprite_index == spr_ralsei_dance)
+                if (sprite_index == spr_kris_dance || sprite_index == spr_susie_dance || sprite_index == spr_ralsei_dance || sprite_index == spr_noelle_dance)
                 {
                     instance_destroy();
                     with (obj_heroparent)
@@ -364,8 +447,9 @@
             scr_nextact();
         }
     }
-    if (actcon == 20 || actconsus == 20 || actconral == 20)
+    if (actcon == 20 || actconsus == 20 || actconral == 20 || actconnoe == 20)
     {
+        scr_nextact_simul();
         if (scr_terminate_writer())
         {
             if (obj_sweet_enemy.dancing == true && obj_kk_enemy.dancing == true && obj_hatguy_enemy.dancing == true && obj_sweet_enemy.endcon == 0)
@@ -376,7 +460,7 @@
             {
                 with (obj_marker)
                 {
-                    if (sprite_index == spr_kris_dance || sprite_index == spr_susie_dance || sprite_index == spr_ralsei_dance)
+                    if (sprite_index == spr_kris_dance || sprite_index == spr_susie_dance || sprite_index == spr_ralsei_dance || sprite_index == spr_noelle_dance)
                     {
                         instance_destroy();
                         with (obj_heroparent)
@@ -391,6 +475,7 @@
                         }
                     }
                 }
+                actconnoe = -1;
                 actconsus = -1;
                 actconral = -1;
                 actcon = 1;
