--- a/Code/gml_GlobalScript_scr_actselect.gml
+++ b/Code/gml_GlobalScript_scr_actselect.gml
@@ -1,6 +1,6 @@
 function scr_actselect(arg0, arg1)
 {
-    if (i_ex(global.monsterinstance[arg0]))
+    if (global.charturn == 0 && i_ex(global.monsterinstance[arg0]))
     {
         if (global.char[global.charturn] == 1)
         {
@@ -9,42 +9,80 @@
         if (global.char[global.charturn] == 2)
         {
             global.monsterinstance[arg0].actingsus = arg1 + 1;
+            global.monsterinstance[arg0].actconsus = 1;
         }
         if (global.char[global.charturn] == 3)
         {
             global.monsterinstance[arg0].actingral = arg1 + 1;
+            global.monsterinstance[arg0].actconral = 1;
         }
         if (global.char[global.charturn] == 4)
         {
             global.monsterinstance[arg0].actingnoe = arg1 + 1;
+            global.monsterinstance[arg0].actconnoe = 1;
         }
     }
     if (global.char[global.charturn] == 1)
     {
-        global.actingsimul[0] = actsimul[arg1];
-        global.acting[0] = 1;
-        global.actingsingle[0] = 1;
+        global.acting[global.charturn] = 1;
+        global.actingsimul[global.charturn] = actsimul[arg1];
+        global.actingsingle[global.charturn] = 1;
         global.actingtarget[global.charturn] = arg0;
         if (global.actactor[arg0][arg1] == 2)
         {
-            global.acting[charpos[1]] = 1;
+            var pos = scr_findchar_available(2);
+            global.acting[pos] = 2;
+            global.actingsimul[pos] = 0;
+            global.actingsingle[pos] = 0;
+            if (pos == 0)
+            {
+                global.monsterinstance[arg0].acting = arg1 + 1;
+                global.monsterinstance[arg0].actingsus = 0;
+                global.monsterinstance[arg0].actconsus = 0;
+            }
         }
         if (global.actactor[arg0][arg1] == 3)
         {
-            global.acting[charpos[2]] = 1;
+            var pos = scr_findchar_available(3);
+            global.acting[pos] = 2;
+            global.actingsimul[pos] = 0;
+            global.actingsingle[pos] = 0;
+            if (pos == 0)
+            {
+                global.monsterinstance[arg0].acting = arg1 + 1;
+                global.monsterinstance[arg0].actingral = 0;
+                global.monsterinstance[arg0].actconral = 0;
+            }
         }
-        if (global.actactor[arg0][arg1] == 4)
+        if (global.actactor[arg0][arg1] == 4 || global.actactor[arg0][arg1] == 6)
         {
+            global.acting[0] = 1;
+            global.actingsimul[0] = actsimul[arg1];
+            global.actingsingle[0] = 1;
+            global.monsterinstance[arg0].acting = arg1 + 1;
-            global.acting[2] = 1;
+            global.acting[1] = 2;
-            global.acting[1] = 1;
+            global.actingsimul[1] = 0;
+            global.actingsingle[1] = 0;
+            global.acting[2] = 2;
+            global.actingsimul[2] = 0;
+            global.actingsingle[2] = 0;
         }
         if (global.actactor[arg0][arg1] == 5)
         {
-            global.acting[charpos[3]] = 1;
+            var pos = scr_findchar_available(4);
+            global.acting[pos] = 2;
+            global.actingsimul[pos] = 0;
+            global.actingsingle[pos] = 0;
+            if (pos == 0)
+            {
+                global.monsterinstance[arg0].acting = arg1 + 1;
+                global.monsterinstance[arg0].actingnoe = 0;
+                global.monsterinstance[arg0].actconnoe = 0;
+            }
         }
         for (i = 0; i < 3; i += 1)
         {
-            if (global.acting[i] == 1)
+            if (global.acting[i] > 0)
             {
                 global.faceaction[i] = 6;
                 global.charaction[i] = 9;
@@ -53,6 +91,7 @@
     }
     else
     {
+        global.acting[global.charturn] = 1;
         global.actingtarget[global.charturn] = arg0;
         global.actingsingle[global.charturn] = 1;
         global.actingsimul[global.charturn] = actsimul[arg1];
