--- a/Code/gml_Object_obj_virovirokun_enemy_Step_0.gml
+++ b/Code/gml_Object_obj_virovirokun_enemy_Step_0.gml
@@ -151,7 +151,7 @@
     }
     if (acting == 2 && actcon == 0)
     {
-        scr_act_charsprite("kris", choose(spr_krisb_virokun_nurse, spr_krisb_virokun_doctor), 0, 0);
+        scr_act_charsprite(global.currentactingchar, choose(spr_krisb_virokun_nurse, spr_krisb_virokun_doctor), 0, 0);
         if (global.mercymod[myself] < global.mercymax[myself])
         {
             scr_mercyadd(myself, 100);
@@ -164,26 +164,55 @@
     {
         if (noelle_special == 0)
         {
-            scr_act_charsprite("kris", choose(spr_krisb_virokun_nurse, spr_krisb_virokun_doctor), 0, 0);
-            if (!scr_havechar(4))
-            {
-                scr_act_charsprite("ralsei", spr_ralseib_virokun, 0, 0);
-                scr_act_charsprite("susie", spr_susieb_virokun, 0, 0);
+            for (var i = 0; i < 3; i++)
+            {
+                switch (global.char[i])
+                {
+                    case 1:
+                        scr_act_charsprite(i, choose(spr_krisb_virokun_nurse, spr_krisb_virokun_doctor), 0, 0);
+                        break;
+                    case 2:
+                        scr_act_charsprite(i, spr_susieb_virokun, 0, 0);
+                        break;
+                    case 3:
+                        scr_act_charsprite(i, spr_ralseib_virokun, 0, 0);
+                        break;
+                    case 4:
+                        if (scr_sideb_get_phase() < 2)
+                        {
+                            noelle_fall = true;
+                            noelle_fall_timer = 0;
+                            with (object_index)
+                            {
+                                noelle_splat = 1;
+                            }
+                        }
+                        else
+                        {
+                            scr_act_charsprite(i, spr_noelleb_act_nurse, 0, 1);
+                        }
+                        break;
+                }
+            }
+            if (scr_havechar(0))
+            {
+                if (scr_havechar(2))
+                {
+                    msgset(0, "* You and Susie showed the enemy tender loving care!/%");
+                }
+                if (scr_havechar(3))
+                {
+                    msgset(0, "* You and Ralsei showed the enemy tender loving care!/%");
+                }
+                if (scr_havechar(4))
+                {
+                    msgsetloc(0, "* You and Noelle showed the enemy tender loving care!/%", "obj_virovirokun_enemy_slash_Step_0_gml_183_0");
+                }
+            }
+            else
+            {
                 msgsetloc(0, "* Everyone treated the enemy with tender loving care!! All the enemies felt great!!/%", "obj_virovirokun_enemy_slash_Step_0_gml_179_0");
             }
-            else
-            {
-                if (scr_sideb_get_phase() < 2)
-                {
-                    noelle_fall = true;
-                    noelle_fall_timer = 0;
-                    with (object_index)
-                    {
-                        noelle_splat = 1;
-                    }
-                }
-                msgsetloc(0, "* You and Noelle showed the enemy tender loving care!/%", "obj_virovirokun_enemy_slash_Step_0_gml_183_0");
-            }
             with (obj_monsterparent)
             {
                 if (global.mercymod[myself] < global.mercymax[myself])
@@ -559,6 +588,7 @@
                     image_alpha = 1;
                 }
             }
+            scr_nextact_simul();
             if (scr_terminate_writer())
             {
                 actconsus = -1;
