--- a/Code/gml_Object_o_coaster_controller_sneo_Create_0.gml
+++ b/Code/gml_Object_o_coaster_controller_sneo_Create_0.gml
@@ -13,17 +13,50 @@
 mykey[2] = 67;
 for (i = 0; i < 3; i += 1)
 {
-    HeroCoaster[i] = instance_create(0, 0, o_coaster_hero_sneo);
+    HeroCoaster[i] = instance_create(0, 0, o_coaster_hero_cpm);
     HeroCoaster[i].HeroID = i;
-    HeroCoaster[i].image_index = i;
     HeroCoaster[i].depth = HeroCoaster[i].y * -100;
     HeroCoaster[i].siner = 0;
     HeroCoaster[i].mykey = mykey[i];
+    switch (scr_modconfig("coasterMode"))
+    {
+        case 0:
+            HeroCoaster[i].CoasterType = 7;
+            break;
+        case 1:
+            HeroCoaster[i].CoasterType = global.char[i];
+            break;
+        case 2:
+            var variants = [7, 1, 2, 3, 4, 8];
+            HeroCoaster[i].CoasterType = variants[scr_modconfig("coasterType")];
+            break;
+    }
     with (HeroCoaster[i])
     {
         back = instance_create(x, y, o_coaster_hero_back_sneo);
         back.parentid = id;
-        back.depth = depth + 2;
+        if (global.char[HeroID] == 0)
+        {
+            back.visible = 0;
+        }
+    }
+    switch (HeroCoaster[i].HeroID)
+    {
+        case 0:
+            HeroCoaster[i].lane_coaster_x = -15;
+            HeroCoaster[i].lane_coaster_y = 34;
+            HeroCoaster[i].lane_character_x = -5;
+            HeroCoaster[i].lane_character_y = -10;
+            break;
+        case 1:
+            HeroCoaster[i].lane_coaster_x = -5;
+            HeroCoaster[i].lane_coaster_y = 33;
+            HeroCoaster[i].lane_character_x = 5;
+            break;
+        case 2:
+            HeroCoaster[i].lane_coaster_x = -10;
+            HeroCoaster[i].lane_coaster_y = 35;
+            break;
     }
     yspot[i] = HeroCoaster[i].y;
 }
