--- a/Code/gml_Object_obj_maus_enemy_Step_0.gml
+++ b/Code/gml_Object_obj_maus_enemy_Step_0.gml
@@ -181,9 +181,10 @@
         actcon = 9;
         alarm[4] = 10;
     }
-    if (acting == 3 && actcon == 0)
+    var num = 3;
+    if (scr_havechar(2))
     {
-        if (!scr_havechar(4))
+        if (acting == 3 && actcon == 0)
         {
             scr_speaker("susie");
             msgsetloc(0, "\\E5* There's only one solution for a mouse!!/", "obj_maus_enemy_slash_Step_0_gml_177_0");
@@ -191,7 +192,11 @@
             scr_battletext();
             actcon = 30;
         }
-        else
+        num++;
+    }
+    if (scr_havechar(4))
+    {
+        if (acting == num && actcon == 0)
         {
             nact_count++;
             if (global.flag[379] == 1)
@@ -239,6 +244,7 @@
             }
             actcon = 1;
         }
+        num++;
     }
     if (actcon == 30)
     {
@@ -261,26 +267,36 @@
     }
     if (actcon == 32)
     {
-        scr_speaker("ralsei");
-        msgsetloc(0, "\\EK* ...don't we just have to fight the cat now?/", "obj_maus_enemy_slash_Step_0_gml_251_0");
-        scr_anyface_next("susie", "K");
-        msgnextloc("\\EK* Huh?/%", "obj_maus_enemy_slash_Step_0_gml_253_0");
-        scr_battletext();
         remove = true;
+        if (scr_havechar(3))
+        {
+            scr_speaker("ralsei");
+            msgsetloc(0, "\\EK* ...don't we just have to fight the cat now?/", "obj_maus_enemy_slash_Step_0_gml_251_0");
+            scr_anyface_next("susie", "K");
+            msgnextloc("\\EK* Huh?/%", "obj_maus_enemy_slash_Step_0_gml_253_0");
+            scr_battletext();
-        actcon = 33;
+            actcon = 33;
-        alarm[4] = 30;
+            alarm[4] = 30;
+        }
+        else
+        {
+            actcon = 34;
+        }
     }
     if (actcon == 34 && !instance_exists(obj_writer))
     {
         actcon = 1;
     }
-    if (acting == 4 && actcon == 0)
+    if (scr_havechar(3) || scr_havechar(4))
     {
+        if (acting == num && actcon == 0)
+        {
-        trappingX = true;
+            trappingX = true;
-        msgsetsubloc(0, "* Press ~1 to trap all the enemies!", scr_get_input_name(6), "obj_maus_enemy_slash_Step_0_gml_269_0");
+            msgsetsubloc(0, "* Press ~1 to trap all the enemies!", scr_get_input_name(6), "obj_maus_enemy_slash_Step_0_gml_269_0");
-        scr_battletext_default();
+            scr_battletext_default();
-        actcon = 9;
+            actcon = 9;
-        alarm[4] = 10;
+            alarm[4] = 10;
+        }
     }
     if (actcon == 10)
     {
@@ -338,13 +354,18 @@
     }
     if (actingral == 1 && actconral == 1)
     {
-        if (!trappedText)
+        if (!trappedText && scr_havechar(2))
         {
             with (obj_maus_enemy)
             {
                 trappedText = true;
                 global.actsimulral[myself][0] = 1;
             }
+            with (obj_mauswheel_enemy)
+            {
+                trappedText = true;
+                global.actsimulral[myself][0] = 1;
+            }
             msgsetloc(0, "* Ralsei started putting peanut butter on a spoon and put it in a trap!!/%", "obj_maus_enemy_slash_Step_0_gml_343_0");
             scr_mercyadd(myself, 50);
             actconral = 2;
@@ -362,13 +383,15 @@
     if (actconral == 2 && !instance_exists(obj_writer))
     {
         actconral = 2.5;
-        basket = instance_create(obj_herosusie.x - 5, cameray() - 100, obj_maus_basket_susie);
-        with (obj_herosusie)
+        var susieinstance = global.charinstance[scr_findchar(2)];
+        basket = instance_create(susieinstance.x - 5, cameray() - 100, obj_maus_basket_susie);
+        basket.targety = susieinstance.y;
+        with (susieinstance)
         {
             visible = 0;
         }
-        nise_susie = scr_dark_marker(obj_herosusie.x + 16, obj_herosusie.y - 1, spr_susie_shock_r);
+        nise_susie = scr_dark_marker(susieinstance.x + 16, susieinstance.y - 1, spr_susie_shock_r);
-        nise_susie.depth = obj_herosusie.depth;
+        nise_susie.depth = susieinstance.depth;
         scr_speaker("no_name");
         msgsetloc(0, "* Susie was captured!!/%", "obj_maus_enemy_slash_Step_0_gml_422_0");
         scr_battletext_default();
@@ -437,10 +460,7 @@
             {
                 if ((global.actingtarget[__i] == myself && global.char[__i] > 0 && global.charspecial[__i] == 0) || (global.chartarget[__i] == myself && (global.charspecial[__i] == 100 || global.charspecial[__i] == 3)))
                 {
-                    if (global.char[__i] == 3)
-                    {
-                        acttriggered = 1;
+                    acttriggered = 1;
-                    }
                     global.acting[__i] = 0;
                     global.actingsimul[__i] = 0;
                     global.actingsingle[__i] = 0;
@@ -458,7 +478,10 @@
             }
             if (acttriggered)
             {
-                scr_nextact();
+                with (global.monsterinstance[myself])
+                {
+                    actcon = 1;
+                }
             }
             instance_destroy();
         }
@@ -472,7 +495,7 @@
             {
                 instance_destroy();
             }
-            with (obj_herosusie)
+            with (global.charinstance[scr_findchar(2)])
             {
                 visible = 1;
             }
@@ -480,6 +503,7 @@
     }
     if (actcon == 20 || actconsus == 20 || actconral == 20 || actconnoe == 20)
     {
+        scr_nextact_simul();
         if (scr_terminate_writer())
         {
             actconsus = -1;
