--- a/Code/gml_Object_obj_sweet_enemy_Step_0.gml
+++ b/Code/gml_Object_obj_sweet_enemy_Step_0.gml
@@ -246,6 +246,16 @@
         scr_anyface_next("susie", "L");
         msgnextloc("\\EL* ... hey, wait^1. Why can't we ACT without Kris?/", "obj_sweet_enemy_slash_Step_0_gml_190_0");
     }
+    else if (turns >= 3 && !scr_havechar(1))
+    {
+        scr_speaker("susie");
+        msgsetloc(0, "\\EK* Hey, is this really working!?/", "obj_sweet_enemy_slash_Step_0_gml_280_0");
+        msgnext("\\EC* ...We can't ACT without Kris!/");
+        scr_anyface_next("ralsei", "L");
+        msgnext("\\EL* Maybe if we tried... um.../");
+        scr_anyface_next("susie", "L");
+        msgnextloc("\\EL* ... hey, wait^1. Why can't we ACT without Kris?/", "obj_sweet_enemy_slash_Step_0_gml_190_0");
+    }
     else
     {
         scr_speaker("susie");
@@ -265,8 +275,49 @@
 }
 if (dancecon == 1.1 && !instance_exists(obj_writer))
 {
-    susi = scr_act_charsprite("susie", spr_susie_dance, 0.16, 1);
-    susi.depth = obj_heroralsei.depth + 1;
+    if (scr_havechar(2) && scr_havechar(3))
+    {
+        for (var i = 0; i < 3; i++)
+        {
+            if (instance_exists(_charactsprite[i]))
+            {
+                continue;
+            }
+            switch (global.char[i])
+            {
+                case 2:
+                    scr_act_charsprite(i, spr_susie_dance, 0.16, 1);
+                    break;
+                case 4:
+                    scr_act_charsprite(i, spr_noelle_dance, 0.16, 1);
+                    break;
+            }
+            susi = _charactsprite[scr_findchar(2)];
+        }
+    }
+    else
+    {
+        learnActCutscene = false;
+        for (var i = 0; i < 3; i++)
+        {
+            if (instance_exists(_charactsprite[i]))
+            {
+                continue;
+            }
+            switch (global.char[i])
+            {
+                case 2:
+                    scr_act_charsprite(i, spr_susie_dance, 0.16, 1);
+                    break;
+                case 3:
+                    scr_act_charsprite(i, spr_ralsei_dance, 0.16, 1);
+                    break;
+                case 4:
+                    scr_act_charsprite(i, spr_noelle_dance, 0.16, 1);
+                    break;
+            }
+        }
+    }
     with (obj_afterimage)
     {
         instance_destroy();
@@ -277,38 +328,239 @@
     }
     global.flag[34] = 0;
     learnedhowtoact = 1;
+    actchange = true;
+    obj_kk_enemy.actchange = true;
+    obj_hatguy_enemy.actchange = true;
+    if (kriscount > 1)
+    {
+        global.canact[myself][1] = 1;
+        global.actname[myself][1] = stringsetloc("Dance", "scr_monstersetup_slash_scr_monstersetup_gml_1179_0");
+        global.actcost[myself][1] = 0;
+        global.actactor[myself][1] = 0;
+        global.actsimul[myself][1] = 0;
+        global.canact[obj_kk_enemy.myself][1] = 1;
+        global.actname[obj_kk_enemy.myself][1] = stringsetloc("Dance", "scr_monstersetup_slash_scr_monstersetup_gml_1145_0");
+        global.actcost[obj_kk_enemy.myself][1] = 0;
+        global.actactor[obj_kk_enemy.myself][1] = 0;
+        global.actsimul[obj_kk_enemy.myself][1] = 0;
+        global.canact[obj_hatguy_enemy.myself][1] = 1;
+        global.actname[obj_hatguy_enemy.myself][1] = stringsetloc("Dance", "scr_monstersetup_slash_scr_monstersetup_gml_1111_0");
+        global.actcost[obj_hatguy_enemy.myself][1] = 0;
+        global.actactor[obj_hatguy_enemy.myself][1] = 0;
+        global.actsimul[obj_hatguy_enemy.myself][1] = 0;
+        if (global.lang != "ja")
+        {
+            global.actsimul[myself][1] = 1;
+            global.actsimul[obj_kk_enemy.myself][1] = 1;
+            global.actsimul[obj_hatguy_enemy.myself][1] = 1;
+        }
+        global.canact[myself][2] = 1;
+        global.actname[myself][2] = stringsetloc("Dance X", "scr_monstersetup_slash_scr_monstersetup_gml_1184_0");
+        global.actcost[myself][2] = 0;
+        global.actactor[myself][2] = 6;
+        global.actsimul[myself][2] = 0;
+        global.canact[obj_kk_enemy.myself][2] = 1;
+        global.actname[obj_kk_enemy.myself][2] = stringsetloc("Dance X", "scr_monstersetup_slash_scr_monstersetup_gml_1150_0");
+        global.actcost[obj_kk_enemy.myself][2] = 0;
+        global.actactor[obj_kk_enemy.myself][2] = 6;
+        global.actsimul[obj_kk_enemy.myself][2] = 0;
+        global.canact[obj_hatguy_enemy.myself][2] = 1;
+        global.actname[obj_hatguy_enemy.myself][2] = stringsetloc("Dance X", "scr_monstersetup_slash_scr_monstersetup_gml_1116_0");
+        global.actcost[obj_hatguy_enemy.myself][2] = 0;
+        global.actactor[obj_hatguy_enemy.myself][2] = 6;
+        global.actsimul[obj_hatguy_enemy.myself][2] = 0;
+    }
+    if (danceallneeded)
+    {
+        global.canact[myself][1] = 1;
+        global.actname[myself][1] = stringset("DanceAll");
+        global.actcost[myself][1] = 0;
+        global.actactor[myself][1] = 0;
+        global.actsimul[myself][1] = 0;
+        global.actsimulsus[myself][0] = 0;
+        global.actsimulral[myself][0] = 0;
+        global.actsimulnoe[myself][0] = 0;
+        global.canact[obj_kk_enemy.myself][1] = 1;
+        global.actname[obj_kk_enemy.myself][1] = stringset("DanceAll");
+        global.actcost[obj_kk_enemy.myself][1] = 0;
+        global.actactor[obj_kk_enemy.myself][1] = 0;
+        global.actsimul[obj_kk_enemy.myself][1] = 0;
+        global.actsimulsus[obj_kk_enemy.myself][0] = 0;
+        global.actsimulral[obj_kk_enemy.myself][0] = 0;
+        global.actsimulnoe[obj_kk_enemy.myself][0] = 0;
+        global.canact[obj_hatguy_enemy.myself][1] = 1;
+        global.actname[obj_hatguy_enemy.myself][1] = stringset("DanceAll");
+        global.actcost[obj_hatguy_enemy.myself][1] = 0;
+        global.actactor[obj_hatguy_enemy.myself][1] = 0;
+        global.actsimul[obj_hatguy_enemy.myself][1] = 0;
+        global.actsimulsus[obj_hatguy_enemy.myself][0] = 0;
+        global.actsimulral[obj_hatguy_enemy.myself][0] = 0;
+        global.actsimulnoe[obj_hatguy_enemy.myself][0] = 0;
+    }
+    if (!scr_havechar(0) || danceallneeded)
+    {
-    global.canact[myself][2] = 0;
+        global.canact[myself][2] = 0;
-    global.actname[myself][2] = "";
+        global.actname[myself][2] = "";
-    global.actcost[myself][2] = 0;
+        global.actcost[myself][2] = 0;
-    global.actactor[myself][2] = 0;
+        global.actactor[myself][2] = 0;
-    global.canact[obj_kk_enemy.myself][2] = 0;
+        global.canact[obj_kk_enemy.myself][2] = 0;
-    global.actname[obj_kk_enemy.myself][2] = "";
+        global.actname[obj_kk_enemy.myself][2] = "";
-    global.actcost[obj_kk_enemy.myself][2] = 0;
+        global.actcost[obj_kk_enemy.myself][2] = 0;
-    global.actactor[obj_kk_enemy.myself][2] = 0;
+        global.actactor[obj_kk_enemy.myself][2] = 0;
-    global.canact[obj_hatguy_enemy.myself][2] = 0;
+        global.canact[obj_hatguy_enemy.myself][2] = 0;
-    global.actname[obj_hatguy_enemy.myself][2] = "";
+        global.actname[obj_hatguy_enemy.myself][2] = "";
-    global.actcost[obj_hatguy_enemy.myself][2] = 0;
+        global.actcost[obj_hatguy_enemy.myself][2] = 0;
-    global.actactor[obj_hatguy_enemy.myself][2] = 0;
+        global.actactor[obj_hatguy_enemy.myself][2] = 0;
+    }
     scr_spellmenu_setup();
     scr_speaker("none");
+    if (learnActCutscene)
+    {
+        if (scr_havechar(4))
+        {
+            msgset(0, "* Susie and Noelle learned \\cSS-Action\\cW and \\cNN-Action\\cW! They can ACT from their \\cYMAGIC\\cW menus!/");
+        }
+        else
+        {
-    msgsetloc(0, "* Susie learned \\cSS-Action\\cW! She can ACT from her \\cYMAGIC\\cW menu!/", "obj_sweet_enemy_slash_Step_0_gml_266_0");
+            msgsetloc(0, "* Susie learned \\cSS-Action\\cW! She can ACT from her \\cYMAGIC\\cW menu!/", "obj_sweet_enemy_slash_Step_0_gml_266_0");
+        }
-    scr_anyface_next("ralsei", "L");
+        scr_anyface_next("ralsei", "L");
-    msgnextloc("\\EL* S-Susie^1, I-I don't think you should.../%", "obj_sweet_enemy_slash_Step_0_gml_268_0");
+        msgnextloc("\\EL* S-Susie^1, I-I don't think you should.../%", "obj_sweet_enemy_slash_Step_0_gml_268_0");
+    }
+    else if (unique_xactions == 3)
+    {
+        var learntext = "";
+        for (var pos = 0; pos < 3; pos++)
+        {
+            if (pos > 0)
+            {
+                learntext += "&";
+            }
+            switch (global.char[pos])
+            {
+                case 2:
+                    learntext += "Susie learned \\cSS-Action\\cW!";
+                    break;
+                case 3:
+                    learntext += "Ralsei learned \\cVR-Action\\cW!";
+                    break;
+                case 4:
+                    learntext += "Noelle learned \\cNN-Action\\cW!";
+                    break;
+                default:
+                    learntext += "Someone learned something!";
+                    break;
+            }
+        }
+        msgset(0, "* " + learntext + "/");
+        msgnext("* They can ACT from their \\cYMAGIC\\cW menus!/%");
+    }
+    else if (unique_xactions == 2)
+    {
+        var total = 0;
+        var chars = [0, 0];
+        var charnames = ["", ""];
+        var actnames = ["", ""];
+        for (var pos = 0; pos < 3; pos++)
+        {
+            if (total >= 2)
+            {
+                break;
+            }
+            if (global.char[pos] <= 1)
+            {
+                continue;
+            }
+            if (total == 1 && global.char[pos] == chars[0])
+            {
+                continue;
+            }
+            switch (global.char[pos])
+            {
+                case 2:
+                    chars[total] = 2;
+                    charnames[total] = "Susie";
+                    actnames[total] = "\\cSS-Action\\cW";
+                    total++;
+                    break;
+                case 3:
+                    chars[total] = 3;
+                    charnames[total] = "Ralsei";
+                    actnames[total] = "\\cVR-Action\\cW";
+                    total++;
+                    break;
+                case 4:
+                    chars[total] = 4;
+                    charnames[total] = "Noelle";
+                    actnames[total] = "\\cNN-Action\\cW";
+                    total++;
+                    break;
+            }
+        }
+        var message = "* ";
+        message += (charnames[0] + " and " + charnames[1] + " learned ");
+        message += (actnames[0] + " and " + actnames[1] + "! ");
+        message += "They can ACT from their \\cYMAGIC\\cW menus!/%";
+        msgset(0, message);
+    }
+    else if (unique_xactions == 1)
+    {
+        var char = 0;
+        for (var pos = 0; pos < 3; pos++)
+        {
+            if (global.char[pos] > 1)
+            {
+                char = global.char[pos];
+                break;
+            }
+        }
+        switch (char)
+        {
+            case 2:
+                msgsetloc(0, "* Susie learned \\cSS-Action\\cW! She can ACT from her \\cYMAGIC\\cW menu!/", "obj_sweet_enemy_slash_Step_0_gml_266_0");
+                global.msg[0] += "%";
+                break;
+            case 3:
+                msgset(0, "* Ralsei learned \\cVR-Action\\cW! He can ACT from his \\cYMAGIC\\cW menu!/%");
+                break;
+            case 4:
+                msgset(0, "* Noelle learned \\cNN-Action\\cW! She can ACT from her \\cYMAGIC\\cW menu!/%");
+                break;
+            default:
+                msgset(0, "* Other party members can now ACT from their \\cYMAGIC\\cW menus!/%");
+                break;
+        }
+    }
+    else
+    {
+        msgset(0, "* Other party members can now ACT from their \\cYMAGIC\\cW menus!/%");
+    }
     scr_battletext();
     dancecon = 1.15;
 }
 if (dancecon == 1.15 && !instance_exists(obj_writer))
 {
-    with (susi)
+    if (learnActCutscene)
     {
+        with (susi)
+        {
-        move_towards_point(obj_heroralsei.x + 100, obj_heroralsei.y + 5, 10);
+            move_towards_point(obj_heroralsei.x + 100, obj_heroralsei.y + 5, 10);
+        }
+        if ((susi.y >= (obj_heroralsei.y - 3) && susi.ystart <= (obj_heroralsei.y - 3)) || (susi.y <= (obj_heroralsei.y + 13) && susi.ystart >= (obj_heroralsei.y + 13)))
+        {
+            susi.x = obj_heroralsei.x + 100;
+            susi.y = obj_heroralsei.y + 5;
+            susi.speed = 0;
+            dancecon = 1.2;
+        }
     }
-    if (susi.y >= (obj_heroralsei.y - 3))
+    else
     {
-        susi.x = obj_heroralsei.x + 100;
-        susi.y = obj_heroralsei.y + 5;
-        susi.speed = 0;
-        dancecon = 1.2;
+        with (obj_musical_controller)
+        {
+            happyfeetscene = 1;
+        }
+        dancecon = 4;
     }
 }
 if (dancecon == 1.2 || (dancescenetimer > 0 && dancescenetimer <= 146))
@@ -316,8 +568,8 @@
     dancescenetimer++;
     if (dancescenetimer == 1)
     {
-        scr_act_charsprite_end();
-        susi = scr_act_charsprite("susie", spr_susie_dance, 0.5, 1);
+        susi.image_index = 0;
+        susi.image_speed = 0.5;
         susi.depth = obj_heroralsei.depth + 1;
         with (obj_afterimage)
         {
@@ -335,7 +587,7 @@
     {
         susi.hspeed = 10;
         dancescenetimer2 = 1;
-        rals = scr_act_charsprite("ralsei", spr_ralsei_hurt_fixed, 0.16, 1);
+        rals = scr_act_charsprite(scr_findchar(3), spr_ralsei_hurt_fixed, 0.16, 1);
         with (obj_afterimage)
         {
             instance_destroy();
@@ -399,11 +651,8 @@
 }
 if (dancecon == 2.3 && !instance_exists(obj_writer))
 {
-    scr_act_charsprite_end();
-    susi = scr_act_charsprite("susie", spr_susie_dance, 0.5, 1);
     susi.x = obj_heroralsei.x + 80;
     susi.y = obj_heroralsei.y + 5;
-    scr_act_charsprite("ralsei", spr_ralsei_hurt_fixed, 0.16, 1);
     with (obj_afterimage)
     {
         instance_destroy();
@@ -416,12 +665,19 @@
 }
 if (dancecon == 2.4 && !instance_exists(obj_writer))
 {
-    scr_act_charsprite_end();
     scr_speaker("none");
-    susi = scr_act_charsprite("susie", spr_susie_dance, 0.16, 1);
+    susi.image_speed = 0.16;
     susi.x = obj_heroralsei.x + 100;
     susi.y = obj_heroralsei.y + 5;
-    scr_act_charsprite("ralsei", spr_ralsei_dance, 0.16, 1);
+    rals.sprite_index = spr_ralsei_dance;
+    rals.image_speed = 0.16;
+    for (var i = 0; i < 10; i++)
+    {
+        if (instance_exists(_charactsprite[i]))
+        {
+            _charactsprite[i].image_index = 0;
+        }
+    }
     msgsetloc(0, "* (Susie made Ralsei learn \\cVR-Action\\cW even though he didn't want to!)/", "obj_sweet_enemy_slash_Step_0_gml_354_0");
     scr_anyface_next("ralsei", "U");
     msgnextloc("\\EU* (I... I'm sorry^1, Kris!!!)/%", "obj_sweet_enemy_slash_Step_0_gml_356_0");
@@ -456,15 +712,16 @@
 }
 if (dancecon == 2.5 && !instance_exists(obj_writer))
 {
+    var susietarget = global.charinstance[scr_findchar(2)];
     with (susi)
     {
-        move_towards_point(obj_herosusie.x, obj_herosusie.y, 10);
-        depth = obj_heroralsei.depth + 1;
+        move_towards_point(susietarget.x, susietarget.y, 10);
+        depth = susietarget.depth;
     }
-    if (susi.y <= (obj_herosusie.y + 8))
+    if (susi.y <= (susietarget.y + 8))
     {
-        susi.x = obj_herosusie.x;
+        susi.x = susietarget.x;
-        susi.y = obj_herosusie.y;
+        susi.y = susietarget.y;
         susi.speed = 0;
         dancecon = 4;
     }
@@ -473,12 +730,44 @@
 {
     scr_act_charsprite_end();
     dancecon = 0;
+    if (danceCounter > 1)
+    {
+        danceCounter = 1;
+    }
+    if (obj_hatguy_enemy.danceCounter > 1)
+    {
+        obj_hatguy_enemy.danceCounter = 1;
+    }
+    if (obj_kk_enemy.danceCounter > 1)
+    {
+        obj_kk_enemy.danceCounter = 1;
+    }
     with (obj_musical_controller)
     {
         event_user(0);
     }
     scr_speaker("none");
 }
+if (global.myfight == 0 && danceallneeded && actchange)
+{
+    for (var i = 0; i < 3; i++)
+    {
+        if (global.actingsingle[i] > 0)
+        {
+            if (global.char[i] == 1 && global.actingchoice[i] <= 0)
+            {
+            }
+            else
+            {
+                with (obj_battlecontroller)
+                {
+                    scr_endturn();
+                }
+                break;
+            }
+        }
+    }
+}
 if (global.myfight == 3)
 {
     xx = __view_get(e__VW.XView, 0);
@@ -489,65 +778,152 @@
         msgsetloc(0, "* SWEET - The energetic one. Looks up to K_K./%", "obj_sweet_enemy_slash_Step_0_gml_403_0");
         scr_battletext_default();
     }
-    if (acting == 2 && actcon == 0)
+    if (danceallneeded && actchange)
     {
-        actCounter++;
-        danceCounter = 1;
-        scr_act_charsprite("kris", spr_kris_dance, 0.16, 1);
-        if (simultotal == 1 || (global.lang == "ja" && simultotal_funny == 1))
-        {
-            msgsetloc(0, "* You danced!/", "obj_sweet_enemy_slash_Step_0_gml_422_0");
-            msgnextloc("* Sweet got lost in the groove!/%", "obj_sweet_enemy_slash_Step_0_gml_423_0");
-            scr_battletext_default();
-            actcon = 4;
-            alarm[4] = 10;
-        }
-        else
-        {
+        if ((acting > 1 && actcon == 0) || (actingsus == 1 && actconsus == 1) || (actingral == 1 && actconral == 1) || (actingnoe == 1 && actconnoe == 1))
+        {
+            actCounter++;
             dancing = true;
-            if (groove == 0)
-            {
-                if (global.lang == "ja")
-                {
-                    msgsetloc(0, "* You danced with Sweet!/", "obj_sweet_enemy_slash_Step_0_gml_435_0");
-                    msgnextloc("* Sweet got lost in the groove!/%", "obj_sweet_enemy_slash_Step_0_gml_436_0");
-                }
-                else
-                {
-                    msgsetloc(0, "* You danced with Sweet!/%", "obj_sweet_enemy_slash_Step_0_gml_534_0");
-                }
-                groove = 1;
+            danceCounter = 1;
+            obj_hatguy_enemy.dancing = true;
+            obj_hatguy_enemy.danceCounter = 1;
+            obj_kk_enemy.dancing = true;
+            obj_kk_enemy.danceCounter = 1;
+            var name = "Someone";
+            for (var i = 0; i < 3; i++)
+            {
+                switch (global.char[i])
+                {
+                    case 1:
+                        scr_act_charsprite(i, spr_kris_dance, 0.16, 1);
+                        if (i == global.currentactingchar)
+                        {
+                            name = "You";
+                        }
+                        break;
+                    case 2:
+                        scr_act_charsprite(i, spr_susie_dance, 0.16, 1);
+                        if (i == global.currentactingchar)
+                        {
+                            name = "Susie";
+                        }
+                        break;
+                    case 3:
+                        scr_act_charsprite(i, spr_ralsei_dance, 0.16, 1);
+                        if (i == global.currentactingchar)
+                        {
+                            name = "Ralsei";
+                        }
+                        break;
+                    case 4:
+                        scr_act_charsprite(i, spr_noelle_dance, 0.16, 1);
+                        if (i == global.currentactingchar)
+                        {
+                            name = "Noelle";
+                        }
+                        break;
+                }
+                global.acting[i] = 0;
+                global.actingsingle[i] = 0;
+            }
+            if (partysize > 1)
+            {
+                name = "Everyone";
+            }
+            msgset(0, "* " + name + " danced with Sweet, Cap'n and K_K at the same time!/%");
+            scr_battletext_default();
+            actcon = 1;
+            actconsus = 0;
+            actconral = 0;
+            actconnoe = 0;
+        }
+    }
+    var actnum = 2;
+    if (kriscount <= 1 || actchange)
+    {
+        if (acting == actnum && actcon == 0)
+        {
+            actCounter++;
+            danceCounter = 1;
+            if (global.lang != "ja")
+            {
+                scr_act_charsprite_simul("kris", 867, 0.16, 1);
             }
             else
             {
-                msgsetloc(0, "* You danced with Sweet!/%", "obj_sweet_enemy_slash_Step_0_gml_441_0");
+                scr_act_charsprite(global.currentactingchar, spr_kris_dance, 0.16, 1);
             }
-            if (global.lang == "ja")
+            if (simultotal == 1 || (global.lang == "ja" && simultotal_funny == 1))
             {
+                msgsetloc(0, "* You danced!/", "obj_sweet_enemy_slash_Step_0_gml_422_0");
+                msgnextloc("* Sweet got lost in the groove!/%", "obj_sweet_enemy_slash_Step_0_gml_423_0");
                 scr_battletext_default();
-                actcon = 20;
+                actcon = 4;
+                alarm[4] = 10;
             }
             else
             {
-                scr_simultext("kris");
+                dancing = true;
-                if (simulorderkri == 0)
+                if (groove == 0)
                 {
+                    if (global.lang == "ja")
+                    {
+                        msgsetloc(0, "* You danced with Sweet!/", "obj_sweet_enemy_slash_Step_0_gml_435_0");
+                        msgnextloc("* Sweet got lost in the groove!/%", "obj_sweet_enemy_slash_Step_0_gml_436_0");
+                    }
+                    else
+                    {
+                        msgsetloc(0, "* You danced with Sweet!/%", "obj_sweet_enemy_slash_Step_0_gml_534_0");
+                    }
+                    groove = 1;
+                }
+                else
+                {
+                    msgsetloc(0, "* You danced with Sweet!/%", "obj_sweet_enemy_slash_Step_0_gml_441_0");
+                }
+                if (global.lang == "ja")
+                {
+                    scr_battletext_default();
                     actcon = 20;
                 }
                 else
                 {
+                    scr_simultext("kris");
+                    if (simulorderkri == 0)
+                    {
+                        actcon = 20;
+                    }
+                    else
+                    {
-                    actcon = 0;
+                        actcon = 0;
+                    }
                 }
             }
         }
+        actnum++;
     }
-    if (acting == 3 && actcon == 0)
+    if (acting == actnum && actcon == 0)
     {
         actCounter++;
         danceCounter = 2;
-        scr_act_charsprite("kris", spr_kris_dance, 0.16, 1);
-        scr_act_charsprite("ralsei", spr_ralsei_dance, 0.16, 1);
-        scr_act_charsprite("susie", spr_susie_dance, 0.16, 1);
+        for (var i = 0; i < 3; i++)
+        {
+            switch (global.char[i])
+            {
+                case 1:
+                    scr_act_charsprite(i, spr_kris_dance, 0.16, 1);
+                    break;
+                case 2:
+                    scr_act_charsprite(i, spr_susie_dance, 0.16, 1);
+                    break;
+                case 3:
+                    scr_act_charsprite(i, spr_ralsei_dance, 0.16, 1);
+                    break;
+                case 4:
+                    scr_act_charsprite(i, spr_noelle_dance, 0.16, 1);
+                    break;
+            }
+        }
         msgsetloc(0, "* Everyone danced with Sweet! They're totally lost in the groove!/%", "obj_sweet_enemy_slash_Step_0_gml_470_0");
         scr_battletext_default();
         actcon = 4;
@@ -562,7 +938,14 @@
     {
         dancing = true;
         danceCounter = 1;
+        if (global.lang != "ja")
+        {
+            scr_act_charsprite_simul("susie", 823, 0.16, 1);
+        }
+        else
+        {
-        scr_act_charsprite("susie", spr_susie_dance, 0.16, 1);
+            scr_act_charsprite(global.currentactingchar, spr_susie_dance, 0.16, 1);
+        }
         if (groove == 0)
         {
             if (global.lang == "ja")
@@ -602,7 +985,14 @@
     {
         dancing = true;
         danceCounter = 1;
+        if (global.lang != "ja")
+        {
+            scr_act_charsprite_simul("ralsei", 783, 0.16, 1);
+        }
+        else
+        {
-        scr_act_charsprite("ralsei", spr_ralsei_dance, 0.16, 1);
+            scr_act_charsprite(global.currentactingchar, spr_ralsei_dance, 0.16, 1);
+        }
         if (groove == 0)
         {
             if (global.lang == "ja")
@@ -638,6 +1028,53 @@
             }
         }
     }
+    if (actingnoe == 1 && actconnoe == 1)
+    {
+        dancing = true;
+        danceCounter = 1;
+        if (global.lang != "ja")
+        {
+            scr_act_charsprite_simul("noelle", 3384, 0.16, 1);
+        }
+        else
+        {
+            scr_act_charsprite(global.currentactingchar, spr_noelle_dance, 0.16, 1);
+        }
+        if (groove == 0)
+        {
+            if (global.lang == "ja")
+            {
+                msgset(0, "* Noelle danced with Sweet!/");
+                msgnext("* Sweet got lost in the groove!/%");
+            }
+            else
+            {
+                msgset(0, "* Noelle danced with Sweet!/%");
+            }
+            groove = 1;
+        }
+        else
+        {
+            msgset(0, "* Noelle danced with Sweet!/%");
+        }
+        if (global.lang == "ja")
+        {
+            scr_battletext_default();
+            actconnoe = 20;
+        }
+        else
+        {
+            scr_simultext("noelle");
+            if (simulordernoe == 0)
+            {
+                actconnoe = 20;
+            }
+            else
+            {
+                actconnoe = 0;
+            }
+        }
+    }
     if (actcon == 1 && !instance_exists(obj_writer))
     {
         if (obj_sweet_enemy.dancing == true && obj_kk_enemy.dancing == true && obj_hatguy_enemy.dancing == true && endcon == 0)
@@ -648,7 +1085,7 @@
         {
             with (obj_marker)
             {
-                if (sprite_index == spr_kris_dance || sprite_index == spr_susie_dance || sprite_index == spr_ralsei_dance)
+                if (sprite_index == spr_kris_dance || sprite_index == spr_susie_dance || sprite_index == spr_ralsei_dance || sprite_index == spr_noelle_dance)
                 {
                     instance_destroy();
                     with (obj_heroparent)
@@ -666,8 +1103,9 @@
             scr_nextact();
         }
     }
-    if (actcon == 20 || actconsus == 20 || actconral == 20)
+    if (actcon == 20 || actconsus == 20 || actconral == 20 || actconnoe == 20)
     {
+        scr_nextact_simul();
         if (scr_terminate_writer())
         {
             if (obj_sweet_enemy.dancing == true && obj_kk_enemy.dancing == true && obj_hatguy_enemy.dancing == true && endcon == 0)
@@ -678,7 +1116,7 @@
             {
                 with (obj_marker)
                 {
-                    if (sprite_index == spr_kris_dance || sprite_index == spr_susie_dance || sprite_index == spr_ralsei_dance)
+                    if (sprite_index == spr_kris_dance || sprite_index == spr_susie_dance || sprite_index == spr_ralsei_dance || sprite_index == spr_noelle_dance)
                     {
                         instance_destroy();
                         with (obj_heroparent)
@@ -693,6 +1131,7 @@
                         }
                     }
                 }
+                actconnoe = -1;
                 actconsus = -1;
                 actconral = -1;
                 actcon = 1;
