--- a/Code/gml_Object_obj_shadowMaker_Step_2.gml
+++ b/Code/gml_Object_obj_shadowMaker_Step_2.gml
@@ -53,22 +53,38 @@
 {
     if (init == 0)
     {
-        instancemax = 0;
+        instancemax = 3;
         with (obj_mainchara)
         {
-            other.listinstance[other.instancemax] = id;
-            other.instancemax++;
+            other.listinstance[0] = id;
+        }
+        if (instance_exists(global.cinstance[0]))
+        {
+            with (global.cinstance[0])
+            {
+                other.listinstance[1] = id;
+            }
+        }
+        else
+        {
+            listinstance[1] = -4;
+        }
+        if (instance_exists(global.cinstance[1]))
+        {
+            with (global.cinstance[1])
+            {
+                other.listinstance[2] = id;
+            }
+        }
+        else
+        {
+            listinstance[2] = -4;
         }
         with (obj_npc_butler)
         {
             other.listinstance[other.instancemax] = id;
             other.instancemax++;
         }
-        with (obj_caterpillarchara)
-        {
-            other.listinstance[other.instancemax] = id;
-            other.instancemax++;
-        }
         init = 1;
     }
     if (init == 1)
@@ -77,6 +93,11 @@
         {
             shadow[i] = instance_create(x, y, obj_marker);
             frontshadow[i] = instance_create(x, y, obj_marker);
+            if (!instance_exists(listinstance[i]))
+            {
+                shadow[i].visible = false;
+                frontshadow[i].visible = false;
+            }
         }
         init = 2;
     }
@@ -84,29 +105,48 @@
     {
         for (var i = 0; i < instancemax; i++)
         {
-            with (shadow[i])
+            if (instance_exists(listinstance[i]))
             {
+                var i_sprite = listinstance[i].sprite_index;
+                var i_x = listinstance[i].x;
+                var i_y = listinstance[i].y;
+                if (listinstance[i].object_index == obj_mainchara)
+                {
+                    i_sprite = listinstance[i].visual_info.sprite_index;
+                    i_x = listinstance[i].x - listinstance[i].visual_info.halign;
+                    i_y = listinstance[i].y - listinstance[i].visual_info.valign;
+                }
+                with (shadow[i])
+                {
-                sprite_index = other.listinstance[i].sprite_index;
+                    sprite_index = i_sprite;
-                image_index = other.listinstance[i].image_index;
+                    image_index = other.listinstance[i].image_index;
-                image_xscale = other.listinstance[i].image_xscale;
+                    image_xscale = other.listinstance[i].image_xscale;
-                image_yscale = other.listinstance[i].image_yscale * -1 * other.shadLength;
+                    image_yscale = other.listinstance[i].image_yscale * -1 * other.shadLength;
-                x = other.listinstance[i].x;
+                    x = i_x;
-                y = (other.listinstance[i].y + (sprite_height * image_yscale)) - 4;
+                    y = (i_y + (sprite_height * image_yscale)) - 4;
-                image_blend = other.floorShadowColor;
+                    image_blend = other.floorShadowColor;
-                image_alpha = other.floorshadOpacity;
+                    image_alpha = other.floorshadOpacity;
+                    visible = true;
-                depth = 900000;
+                    depth = 900000;
+                }
+                with (frontshadow[i])
+                {
+                    sprite_index = i_sprite;
+                    image_xscale = other.listinstance[i].image_xscale;
+                    image_yscale = other.listinstance[i].image_yscale;
+                    x = i_x;
+                    y = i_y + 4;
+                    image_blend = other.frontShadowColor;
+                    image_alpha = other.frontshadOpacity;
+                    image_index = other.listinstance[i].image_index;
+                    visible = true;
+                    depth = other.listinstance[i].depth - 1;
+                }
             }
-            with (frontshadow[i])
+            else
             {
-                sprite_index = other.listinstance[i].sprite_index;
-                image_xscale = other.listinstance[i].image_xscale;
-                image_yscale = other.listinstance[i].image_yscale;
-                x = other.listinstance[i].x;
-                y = other.listinstance[i].y + 4;
-                image_blend = other.frontShadowColor;
-                image_alpha = other.frontshadOpacity;
-                image_index = other.listinstance[i].image_index;
-                depth = other.listinstance[i].depth - 1;
+                shadow[i].visible = false;
+                frontshadow[i].visible = false;
             }
         }
     }
