--- a/Code/gml_Object_obj_sweet_enemy_Create_0.gml
+++ b/Code/gml_Object_obj_sweet_enemy_Create_0.gml
@@ -15,6 +15,7 @@
 danceCounter = 0;
 actCounter = 0;
 learnedAct = false;
+learnActCutscene = true;
 letsdance = false;
 letsdance2 = false;
 letsdance3 = false;
@@ -45,4 +46,31 @@
 endcon = 0;
 endtimer = 0;
 simultotal_funny = 0;
+partysize = 0;
+kriscount = 0;
+unique_xactions = 0;
+if (scr_havechar(2))
+{
+    unique_xactions++;
+}
+if (scr_havechar(3))
+{
+    unique_xactions++;
+}
+if (scr_havechar(4))
+{
+    unique_xactions++;
+}
+for (var i = 0; i < 3; i++)
+{
+    if (global.char[i] == 0)
+    {
+        continue;
+    }
+    partysize++;
+    if (global.char[i] == 1)
+    {
+        kriscount++;
+    }
+}
 
