--- a/Code/gml_Object_o_coaster_berdly_Step_0.gml
+++ b/Code/gml_Object_o_coaster_berdly_Step_0.gml
@@ -99,7 +99,7 @@
 }
 if (endscene == 1 && instance_exists(obj_ch2_scene11a))
 {
-    var xx = obj_ch2_scene11a.coaster_berdly.x + 22 + obj_ch2_scene11a.coaster_berdly.sinerx + obj_ch2_scene11a.coaster_berdly.coaster_offset_x;
+    var xx = obj_ch2_scene11a.coaster_berdly.x + 72 + obj_ch2_scene11a.coaster_berdly.sinerx + obj_ch2_scene11a.coaster_berdly.coaster_offset_x;
     var yy = obj_ch2_scene11a.coaster_berdly.y + 47 + obj_ch2_scene11a.coaster_berdly.sinery + obj_ch2_scene11a.coaster_berdly.coaster_offset_y;
     if (endscenetimer < 10)
     {
@@ -172,7 +172,7 @@
         {
             forceypos = 2;
             obj_berdlyb_enemy.ralseigo = 0;
-            with (o_coaster_hero)
+            with (obj_partymenu)
             {
                 if (HeroID == 2)
                 {
@@ -184,7 +184,7 @@
         {
             forceypos = 1;
             obj_berdlyb_enemy.susiego = 0;
-            with (o_coaster_hero)
+            with (obj_partymenu)
             {
                 if (HeroID == 1)
                 {
@@ -196,7 +196,7 @@
         {
             forceypos = 0;
             obj_berdlyb_enemy.krisgo = 0;
-            with (o_coaster_hero)
+            with (obj_partymenu)
             {
                 if (HeroID == 0)
                 {
