--- a/Code/gml_Object_obj_ch2_scene11a_Create_0.gml
+++ b/Code/gml_Object_obj_ch2_scene11a_Create_0.gml
@@ -27,33 +27,73 @@
 coaster_ramp.depth = 90000;
 coaster_ramp.visible = 1;
 coaster_kris = instance_create(800, 72, obj_coaster);
-with (coaster_kris)
-{
-    scr_depth();
-}
 coaster_kris.visible = 0;
+coaster_kris.HeroType = global.char[0];
+switch (scr_modconfig("coasterMode"))
+{
+    case 0:
+    case 1:
+        coaster_kris.CoasterType = global.char[0];
+        break;
+    case 2:
+        var variants = [7, 1, 2, 3, 4, 8];
+        coaster_kris.CoasterType = variants[scr_modconfig("coasterType")];
+        break;
+}
+coaster_kris.laneID = 0;
+coaster_kris.lane_coaster_y = -2;
+with (coaster_kris)
+{
+    scr_depth();
+}
 coaster_susie = instance_create(800, 118, obj_coaster);
-coaster_susie.type = 1;
-with (coaster_susie)
-{
-    scr_depth();
-}
 coaster_susie.visible = 0;
+coaster_susie.HeroType = global.char[1];
+switch (scr_modconfig("coasterMode"))
+{
+    case 0:
+    case 1:
+        coaster_susie.CoasterType = global.char[1];
+        break;
+    case 2:
+        var variants = [7, 1, 2, 3, 4, 8];
+        coaster_susie.CoasterType = variants[scr_modconfig("coasterType")];
+        break;
+}
+coaster_susie.laneID = 1;
+coaster_susie.lane_coaster_y = 12;
+with (coaster_susie)
+{
+    scr_depth();
+}
 coaster_ralsei = instance_create(800, 181, obj_coaster);
-coaster_ralsei.type = 2;
-with (coaster_ralsei)
-{
-    scr_depth();
-}
 coaster_ralsei.visible = 0;
+coaster_ralsei.HeroType = global.char[2];
+switch (scr_modconfig("coasterMode"))
+{
+    case 0:
+    case 1:
+        coaster_ralsei.CoasterType = global.char[2];
+        break;
+    case 2:
+        var variants = [7, 1, 2, 3, 4, 8];
+        coaster_ralsei.CoasterType = variants[scr_modconfig("coasterType")];
+        break;
+}
+coaster_ralsei.laneID = 2;
+coaster_ralsei.lane_coaster_y = 9;
+with (coaster_ralsei)
+{
+    scr_depth();
+}
 coaster_berdly = instance_create(1890, 126, obj_coaster);
-coaster_berdly.type = 3;
+coaster_berdly.CoasterType = 5;
 with (coaster_berdly)
 {
     scr_depth();
 }
 coaster_noelle = instance_create(1890, 105, obj_coaster);
-coaster_noelle.type = 4;
+coaster_noelle.CoasterType = 6;
 coaster_noelle.depth = 95000;
 nise_berdly = scr_dark_marker(-100, 0, spr_cutscene_17_berdly_shocked);
 with (nise_berdly)
