--- a/Code/gml_Object_obj_queen_enemy_Draw_0.gml
+++ b/Code/gml_Object_obj_queen_enemy_Draw_0.gml
@@ -3,15 +3,7 @@
 {
     flash = 0;
 }
-with (obj_herokris)
+with (obj_heroparent)
-{
-    flash = 0;
-}
-with (obj_herosusie)
-{
-    flash = 0;
-}
-with (obj_heroralsei)
 {
     flash = 0;
 }
@@ -29,17 +21,9 @@
     {
         _flash = 0;
     }
-    if ((global.bmenucoord[9][global.charturn] == 4 && !instance_exists(obj_queenshield_enemy)) || (global.bmenucoord[9][global.charturn] == 3 && instance_exists(obj_queenshield_enemy)))
+    if ((global.bmenucoord[9][global.charturn] == (3 + scr_havechar(2)) && !instance_exists(obj_queenshield_enemy)) || (global.bmenucoord[9][global.charturn] == (2 + scr_havechar(2)) && instance_exists(obj_queenshield_enemy)))
     {
-        with (obj_herokris)
+        with (obj_heroparent)
-        {
-            flash = 1;
-        }
-        with (obj_herosusie)
-        {
-            flash = 1;
-        }
-        with (obj_heroralsei)
         {
             flash = 1;
         }
