--- a/Code/gml_Object_obj_mainchara_Create_0.gml
+++ b/Code/gml_Object_obj_mainchara_Create_0.gml
@@ -67,6 +67,17 @@
     lsprite = spr_krisl_dark;
     usprite = spr_krisu_dark;
 }
+name = scr_char_id_to_name(global.char[0]) ?? "kris";
+visual_info = 
+{
+    halign: 0,
+    valign: 0
+};
+scr_set_chara_sprites(name, visual_info);
+visual_info.sprite_index = visual_info.dsprite;
+var alignment = scr_get_chara_alignment(name, visual_info);
+visual_info.halign = alignment.halign;
+visual_info.valign = alignment.valign;
 swordfacing = 1;
 swordsprite = rsprite;
 fun = 0;
@@ -260,6 +271,14 @@
             }
         }
     }
+}
+for (i = 0; i < 75; i += 1)
+{
+    remx[i] = x;
+    remy[i] = y;
+    facing[i] = global.facing;
+    sliding[i] = 0;
+    special[i] = 0;
 }
 initwd = sprite_width;
 initht = sprite_height;
