--- a/Code/gml_Object_obj_battlecontroller_Create_0.gml
+++ b/Code/gml_Object_obj_battlecontroller_Create_0.gml
@@ -113,7 +113,44 @@
 global.monstergold[3] = 0;
 global.monsterexp[3] = 0;
 global.charturn = 0;
+if (global.ambush == 0 && !instance_exists(obj_dojo_spareenemy))
+{
+    if (global.charmove[0] == 0)
+    {
+        global.charturn = 1;
+    }
+    if (global.charturn == 1)
+    {
+        if (global.charmove[1] == 0)
+        {
+            global.charturn = 2;
+        }
+    }
+    if (global.charturn == 2)
+    {
+        if (global.charmove[2] == 0)
+        {
+            scr_attackphase();
+        }
+    }
+}
 global.currentactingchar = 0;
+if (room == room_battletest)
+{
+    if (global.flag[915] == 99)
+    {
+        scr_spellget(4, 10);
+        global.charweapon[4] = 13;
+        scr_weaponinfo(13);
+        global.itemat[4][0] = weaponattemp;
+        global.itemdf[4][0] = weapondftemp;
+        global.itemmag[4][0] = weaponmagtemp;
+        global.hp[4] = 170;
+        global.maxhp[4] = 170;
+        global.at[4] = 8;
+        global.mag[4] = 16;
+    }
+}
 for (i = 0; i < 3; i += 1)
 {
     global.acting[i] = 0;
@@ -275,4 +312,5 @@
         gigaqueencon = 3;
     }
 }
+drawactdebug = false;
 
