--- a/Code/gml_GlobalScript_scr_prevhero.gml
+++ b/Code/gml_GlobalScript_scr_prevhero.gml
@@ -48,6 +48,7 @@
         }
         global.actingsingle[global.charturn] = 0;
         global.actingsimul[global.charturn] = 0;
+        global.actwaiting[global.charturn] = 0;
         global.faceaction[global.charturn] = 0;
         global.chartarget[global.charturn] = 0;
         global.charaction[global.charturn] = 0;
@@ -63,6 +64,9 @@
         global.acting[0] = 0;
         global.acting[1] = 0;
         global.acting[2] = 0;
+        global.actwaiting[0] = 0;
+        global.actwaiting[1] = 0;
+        global.actwaiting[2] = 0;
         global.faceaction[1] = 0;
         global.chartarget[1] = 0;
         global.charaction[1] = 0;
