--- a/Code/gml_Object_obj_hatguy_enemy_Step_0.gml
+++ b/Code/gml_Object_obj_hatguy_enemy_Step_0.gml
@@ -17,11 +17,15 @@
         groove = 0;
         global.typer = 50;
         rr = choose(0, 1, 2, 3);
-        if (dancing == 1 && dancedwithsusie == 0)
+        if (dancing == 1 && (dancedwithsusie == 0 && dancedwithnoelle == 0))
         {
             msgsetloc(0, "What, you wanna&see my moves?", "obj_hatguy_enemy_slash_Step_0_gml_29_0");
         }
-        else if (dancing == 1 && dancedwithsusie == 1)
+        else if (dancing == 1 && (dancedwithsusie == 1 && dancedwithnoelle == 1))
+        {
+            msgset(0, "Ladies, ladies!&One at a time!");
+        }
+        else if (dancing == 1 && (dancedwithsusie == 1 || dancedwithnoelle == 1))
         {
             msgsetloc(0, "Ah-ha, if it's&a madimoizel...", "obj_hatguy_enemy_slash_Step_0_gml_30_0");
         }
@@ -181,65 +185,140 @@
         msgsetloc(0, "* CAP'N - The smooth one. Looks up to K_K./%", "obj_hatguy_enemy_slash_Step_0_gml_143_0");
         scr_battletext_default();
     }
-    if (acting == 2 && actcon == 0)
+    if (danceallneeded && actchange)
     {
-        if (instance_exists(obj_sweet_enemy))
+        if ((acting > 1 && actcon == 0) || (actingsus == 1 && actconsus == 1) || (actingral == 1 && actconral == 1) || (actingnoe == 1 && actconnoe == 1))
         {
             with (obj_sweet_enemy)
             {
                 actCounter++;
             }
-        }
-        danceCounter = 1;
-        scr_act_charsprite("kris", spr_kris_dance, 0.16, 1);
-        if (simultotal == 1 || (global.lang == "ja" && obj_sweet_enemy.simultotal_funny == 1))
-        {
-            msgsetloc(0, "* You danced!/", "obj_hatguy_enemy_slash_Step_0_gml_165_0");
-            msgnextloc("* Cap'n got lost in the groove!/%", "obj_hatguy_enemy_slash_Step_0_gml_166_0");
-            scr_battletext_default();
-            actcon = 4;
-            alarm[4] = 10;
-        }
-        else
-        {
+            obj_sweet_enemy.dancing = true;
+            obj_sweet_enemy.danceCounter = 1;
             dancing = true;
-            if (groove == 0)
-            {
-                if (global.lang == "ja")
-                {
-                    msgsetloc(0, "* You danced with Cap'n!/", "obj_hatguy_enemy_slash_Step_0_gml_177_0");
-                    msgnextloc("* Cap'n got lost in the groove!/%", "obj_hatguy_enemy_slash_Step_0_gml_178_0");
-                }
-                else
-                {
-                    msgsetloc(0, "* You danced with Cap'n!/%", "obj_hatguy_enemy_slash_Step_0_gml_183_0");
-                }
-                groove = 1;
+            danceCounter = 1;
+            obj_kk_enemy.dancing = true;
+            obj_kk_enemy.danceCounter = 1;
+            var name = "Someone";
+            for (var i = 0; i < 3; i++)
+            {
+                switch (global.char[i])
+                {
+                    case 1:
+                        scr_act_charsprite(i, spr_kris_dance, 0.16, 1);
+                        if (i == global.currentactingchar)
+                        {
+                            name = "You";
+                        }
+                        break;
+                    case 2:
+                        scr_act_charsprite(i, spr_susie_dance, 0.16, 1);
+                        if (i == global.currentactingchar)
+                        {
+                            name = "Susie";
+                        }
+                        break;
+                    case 3:
+                        scr_act_charsprite(i, spr_ralsei_dance, 0.16, 1);
+                        if (i == global.currentactingchar)
+                        {
+                            name = "Ralsei";
+                        }
+                        break;
+                    case 4:
+                        scr_act_charsprite(i, spr_noelle_dance, 0.16, 1);
+                        if (i == global.currentactingchar)
+                        {
+                            name = "Noelle";
+                        }
+                        break;
+                }
+                global.acting[i] = 0;
+                global.actingsingle[i] = 0;
+            }
+            if (partysize > 1)
+            {
+                name = "Everyone";
+            }
+            msgset(0, "* " + name + " danced with Sweet, Cap'n and K_K at the same time!/%");
+            scr_battletext_default();
+            actcon = 1;
+            actconsus = 0;
+            actconral = 0;
+            actconnoe = 0;
+        }
+    }
+    var actnum = 2;
+    if (kriscount <= 1 || actchange)
+    {
+        if (acting == actnum && actcon == 0)
+        {
+            if (instance_exists(obj_sweet_enemy))
+            {
+                with (obj_sweet_enemy)
+                {
+                    actCounter++;
+                }
+            }
+            danceCounter = 1;
+            if (global.lang != "ja")
+            {
+                scr_act_charsprite_simul("kris", 867, 0.16, 1);
             }
             else
             {
-                msgsetloc(0, "* You danced with Cap'n!/%", "obj_hatguy_enemy_slash_Step_0_gml_183_0");
+                scr_act_charsprite(global.currentactingchar, spr_kris_dance, 0.16, 1);
             }
-            if (global.lang == "ja")
+            if (simultotal == 1 || (global.lang == "ja" && obj_sweet_enemy.simultotal_funny == 1))
             {
+                msgsetloc(0, "* You danced!/", "obj_hatguy_enemy_slash_Step_0_gml_165_0");
+                msgnextloc("* Cap'n got lost in the groove!/%", "obj_hatguy_enemy_slash_Step_0_gml_166_0");
                 scr_battletext_default();
-                actcon = 20;
+                actcon = 4;
+                alarm[4] = 10;
             }
             else
             {
-                scr_simultext("kris");
+                dancing = true;
-                if (simulorderkri == 0)
+                if (groove == 0)
                 {
+                    if (global.lang == "ja")
+                    {
+                        msgsetloc(0, "* You danced with Cap'n!/", "obj_hatguy_enemy_slash_Step_0_gml_177_0");
+                        msgnextloc("* Cap'n got lost in the groove!/%", "obj_hatguy_enemy_slash_Step_0_gml_178_0");
+                    }
+                    else
+                    {
+                        msgsetloc(0, "* You danced with Cap'n!/%", "obj_hatguy_enemy_slash_Step_0_gml_183_0");
+                    }
+                    groove = 1;
+                }
+                else
+                {
+                    msgsetloc(0, "* You danced with Cap'n!/%", "obj_hatguy_enemy_slash_Step_0_gml_183_0");
+                }
+                if (global.lang == "ja")
+                {
+                    scr_battletext_default();
                     actcon = 20;
                 }
                 else
                 {
+                    scr_simultext("kris");
+                    if (simulorderkri == 0)
+                    {
+                        actcon = 20;
+                    }
+                    else
+                    {
-                    actcon = 0;
+                        actcon = 0;
+                    }
                 }
             }
         }
+        actnum++;
     }
-    if (acting == 3 && actcon == 0)
+    if (acting == actnum && actcon == 0)
     {
         if (instance_exists(obj_sweet_enemy))
         {
@@ -249,9 +328,24 @@
             }
         }
         danceCounter = 2;
-        scr_act_charsprite("kris", spr_kris_dance, 0.16, 1);
-        scr_act_charsprite("ralsei", spr_ralsei_dance, 0.16, 1);
-        scr_act_charsprite("susie", spr_susie_dance, 0.16, 1);
+        for (var i = 0; i < 3; i++)
+        {
+            switch (global.char[i])
+            {
+                case 1:
+                    scr_act_charsprite(i, spr_kris_dance, 0.16, 1);
+                    break;
+                case 2:
+                    scr_act_charsprite(i, spr_susie_dance, 0.16, 1);
+                    break;
+                case 3:
+                    scr_act_charsprite(i, spr_ralsei_dance, 0.16, 1);
+                    break;
+                case 4:
+                    scr_act_charsprite(i, spr_noelle_dance, 0.16, 1);
+                    break;
+            }
+        }
         msgsetloc(0, "* Everyone danced with Cap'n! They're totally lost in the groove!/%", "obj_hatguy_enemy_slash_Step_0_gml_214_0");
         scr_battletext_default();
         actcon = 4;
@@ -267,7 +361,14 @@
         dancing = true;
         danceCounter = 1;
         dancedwithsusie = 1;
+        if (global.lang != "ja")
+        {
+            scr_act_charsprite_simul("susie", 823, 0.16, 1);
+        }
+        else
+        {
-        scr_act_charsprite("susie", spr_susie_dance, 0.16, 1);
+            scr_act_charsprite(global.currentactingchar, spr_susie_dance, 0.16, 1);
+        }
         if (groove == 0)
         {
             if (global.lang == "ja")
@@ -307,7 +408,14 @@
     {
         dancing = true;
         danceCounter = 1;
+        if (global.lang != "ja")
+        {
+            scr_act_charsprite_simul("ralsei", 783, 0.16, 1);
+        }
+        else
+        {
-        scr_act_charsprite("ralsei", spr_ralsei_dance, 0.16, 1);
+            scr_act_charsprite(global.currentactingchar, spr_ralsei_dance, 0.16, 1);
+        }
         if (groove == 0)
         {
             if (global.lang == "ja")
@@ -343,6 +451,54 @@
             }
         }
     }
+    if (actingnoe == 1 && actconnoe == 1)
+    {
+        dancing = true;
+        danceCounter = 1;
+        dancedwithnoelle = 1;
+        if (global.lang != "ja")
+        {
+            scr_act_charsprite_simul("noelle", 3384, 0.16, 1);
+        }
+        else
+        {
+            scr_act_charsprite(global.currentactingchar, spr_noelle_dance, 0.16, 1);
+        }
+        if (groove == 0)
+        {
+            if (global.lang == "ja")
+            {
+                msgset(0, "* Noelle danced with Cap'n!/");
+                msgnext("* Cap'n got lost in the groove!/%");
+            }
+            else
+            {
+                msgset(0, "* Noelle danced with Cap'n!/%");
+            }
+            groove = 1;
+        }
+        else
+        {
+            msgset(0, "* Noelle danced with Cap'n!/%");
+        }
+        if (global.lang == "ja")
+        {
+            scr_battletext_default();
+            actconnoe = 20;
+        }
+        else
+        {
+            scr_simultext("noelle");
+            if (simulordernoe == 0)
+            {
+                actconnoe = 20;
+            }
+            else
+            {
+                actconnoe = 0;
+            }
+        }
+    }
     if (actcon == 1 && !instance_exists(obj_writer))
     {
         if (obj_sweet_enemy.dancing == true && obj_kk_enemy.dancing == true && obj_hatguy_enemy.dancing == true && obj_sweet_enemy.endcon == 0)
@@ -351,9 +507,11 @@
         }
         else if (obj_sweet_enemy.endcon == 0)
         {
+            actcon = 0;
+            acting = 0;
             with (obj_marker)
             {
-                if (sprite_index == spr_kris_dance || sprite_index == spr_susie_dance || sprite_index == spr_ralsei_dance)
+                if (sprite_index == spr_kris_dance || sprite_index == spr_susie_dance || sprite_index == spr_ralsei_dance || sprite_index == spr_noelle_dance)
                 {
                     instance_destroy();
                     with (obj_heroparent)
@@ -371,8 +529,9 @@
             scr_nextact();
         }
     }
-    if (actcon == 20 || actconsus == 20 || actconral == 20)
+    if (actcon == 20 || actconsus == 20 || actconral == 20 || actconnoe == 20)
     {
+        scr_nextact_simul();
         if (scr_terminate_writer())
         {
             if (obj_sweet_enemy.dancing == true && obj_kk_enemy.dancing == true && obj_hatguy_enemy.dancing == true && obj_sweet_enemy.endcon == 0)
@@ -383,7 +542,7 @@
             {
                 with (obj_marker)
                 {
-                    if (sprite_index == spr_kris_dance || sprite_index == spr_susie_dance || sprite_index == spr_ralsei_dance)
+                    if (sprite_index == spr_kris_dance || sprite_index == spr_susie_dance || sprite_index == spr_ralsei_dance || sprite_index == spr_noelle_dance)
                     {
                         instance_destroy();
                         with (obj_heroparent)
@@ -398,6 +557,7 @@
                         }
                     }
                 }
+                actconnoe = -1;
                 actconsus = -1;
                 actconral = -1;
                 actcon = 1;
