--- a/Code/gml_Object_obj_virovirokun_enemy_Step_0.gml
+++ b/Code/gml_Object_obj_virovirokun_enemy_Step_0.gml
@@ -151,7 +151,7 @@
     }
     if (acting == 2 && actcon == 0)
     {
-        scr_act_charsprite("kris", choose(spr_krisb_virokun_nurse, spr_krisb_virokun_doctor), 0, 0);
+        scr_act_charsprite(global.currentactingchar, choose(spr_krisb_virokun_nurse, spr_krisb_virokun_doctor), 0, 0);
         if (global.mercymod[myself] < global.mercymax[myself])
         {
             scr_mercyadd(myself, 100);
@@ -164,26 +164,51 @@
     {
         if (noelle_special == 0)
         {
-            scr_act_charsprite("kris", choose(spr_krisb_virokun_nurse, spr_krisb_virokun_doctor), 0, 0);
-            if (!scr_havechar(4))
-            {
-                scr_act_charsprite("ralsei", spr_ralseib_virokun, 0, 0);
-                scr_act_charsprite("susie", spr_susieb_virokun, 0, 0);
+            for (var i = 0; i < 3; i++)
+            {
+                switch (global.char[i])
+                {
+                    case 1:
+                        scr_act_charsprite(i, choose(spr_krisb_virokun_nurse, spr_krisb_virokun_doctor), 0, 0);
+                        break;
+                    case 2:
+                        scr_act_charsprite(i, spr_susieb_virokun, 0, 0);
+                        break;
+                    case 3:
+                        scr_act_charsprite(i, spr_ralseib_virokun, 0, 0);
+                        break;
+                    case 4:
+                        if (scr_sideb_get_phase() < 2)
+                        {
+                            noelle_fall = true;
+                            noelle_fall_timer = 0;
+                            with (object_index)
+                            {
+                                noelle_splat = 1;
+                            }
+                        }
+                        break;
+                }
+            }
+            if (scr_havechar(0))
+            {
+                if (scr_havechar(2))
+                {
+                    msgset(0, "* You and Susie showed the enemy tender loving care!/%");
+                }
+                if (scr_havechar(3))
+                {
+                    msgset(0, "* You and Ralsei showed the enemy tender loving care!/%");
+                }
+                if (scr_havechar(4))
+                {
+                    msgsetloc(0, "* You and Noelle showed the enemy tender loving care!/%", "obj_virovirokun_enemy_slash_Step_0_gml_183_0");
+                }
+            }
+            else
+            {
                 msgsetloc(0, "* Everyone treated the enemy with tender loving care!! All the enemies felt great!!/%", "obj_virovirokun_enemy_slash_Step_0_gml_179_0");
             }
-            else
-            {
-                if (scr_sideb_get_phase() < 2)
-                {
-                    noelle_fall = true;
-                    noelle_fall_timer = 0;
-                    with (object_index)
-                    {
-                        noelle_splat = 1;
-                    }
-                }
-                msgsetloc(0, "* You and Noelle showed the enemy tender loving care!/%", "obj_virovirokun_enemy_slash_Step_0_gml_183_0");
-            }
             with (obj_monsterparent)
             {
                 if (global.mercymod[myself] < global.mercymax[myself])
@@ -367,37 +392,58 @@
             {
                 visible = 1;
                 image_alpha = 0;
+                other.nise_noelle[myself] = scr_dark_marker(x, y, spr_noelleb_act_nurse);
             }
-            nise_noelle = scr_dark_marker(obj_heronoelle.x, obj_heronoelle.y, spr_noelleb_act_nurse);
-            with (nise_noelle)
+            for (var noepos = 0; noepos < 3; noepos++)
             {
+                if (instance_exists(nise_noelle[noepos]))
+                {
+                    with (nise_noelle[noepos])
+                    {
-                depth = 20 - (__view_get(e__VW.YView, 0) / 40);
+                        depth = 20 - (__view_get(e__VW.YView, 0) / 40);
-                scr_oflash();
+                        scr_oflash();
-                a = scr_afterimage();
+                        a = scr_afterimage();
-                a.hspeed = 2.5;
+                        a.hspeed = 2.5;
-                a.depth = depth + 1;
+                        a.depth = depth + 1;
-                b = scr_afterimage();
+                        b = scr_afterimage();
-                b.image_alpha = 0.6;
+                        b.image_alpha = 0.6;
-                b.hspeed = 5;
+                        b.hspeed = 5;
-                b.depth = depth + 2;
+                        b.depth = depth + 2;
+                        image_speed = 0;
+                    }
+                }
             }
-            nise_noelle.image_speed = 0;
         }
         if (noelle_fall_timer == 30)
         {
-            nise_noelle.image_index = 1;
             snd_play(snd_noise);
-            with (nise_noelle)
+            for (var noepos = 0; noepos < 3; noepos++)
             {
+                if (instance_exists(nise_noelle[noepos]))
+                {
+                    with (nise_noelle[noepos])
+                    {
+                        image_index = 1;
-                scr_shakeobj();
+                        scr_shakeobj();
+                    }
+                }
             }
         }
         if (noelle_fall_timer == 50)
         {
             snd_play_x(snd_splat, 0.6, 0.8);
-            nise_noelle.image_index = 2;
-            nise_noelle.hspeed = 6;
-            nise_noelle.friction = 0.75;
+            for (var noepos = 0; noepos < 3; noepos++)
+            {
+                if (instance_exists(nise_noelle[noepos]))
+                {
+                    with (nise_noelle[noepos])
+                    {
+                        image_index = 2;
+                        hspeed = 6;
+                        friction = 0.75;
+                    }
+                }
+            }
         }
         if (noelle_fall_timer > 80)
         {
@@ -520,15 +566,18 @@
     {
         if (!noelle_fall)
         {
-            if (i_ex(nise_noelle))
+            for (var noepos = 0; noepos < 3; noepos++)
             {
-                with (nise_noelle)
-                {
-                    instance_destroy();
-                }
-                with (obj_heronoelle)
-                {
-                    image_alpha = 1;
+                if (instance_exists(nise_noelle[noepos]))
+                {
+                    with (nise_noelle[noepos])
+                    {
+                        instance_destroy();
+                    }
+                    with (global.charinstance[noepos])
+                    {
+                        image_alpha = 1;
+                    }
                 }
             }
             if (explosion != -4)
@@ -548,17 +597,21 @@
     {
         if (!noelle_fall)
         {
-            if (i_ex(nise_noelle))
+            for (var noepos = 0; noepos < 3; noepos++)
             {
-                with (nise_noelle)
-                {
-                    instance_destroy();
-                }
-                with (obj_heronoelle)
-                {
-                    image_alpha = 1;
+                if (instance_exists(nise_noelle[noepos]))
+                {
+                    with (nise_noelle[noepos])
+                    {
+                        instance_destroy();
+                    }
+                    with (global.charinstance[noepos])
+                    {
+                        image_alpha = 1;
+                    }
                 }
             }
+            scr_nextact_simul();
             if (scr_terminate_writer())
             {
                 actconsus = -1;
