--- a/Code/gml_Object_obj_darkslide_new_Collision_obj_mainchara.gml
+++ b/Code/gml_Object_obj_darkslide_new_Collision_obj_mainchara.gml
@@ -25,8 +25,7 @@
             with (obj_mainchara)
             {
                 sliding = 1;
-                fun = 1;
-                sprite_index = spr_krisd_slide;
+                visual_info.sprite_index = visual_info.slidesprite;
             }
             sliding = 1;
         }
