--- a/Code/gml_Object_obj_mainchara_Step_0.gml
+++ b/Code/gml_Object_obj_mainchara_Step_0.gml
@@ -512,6 +512,33 @@
     }
     x += px;
     y += py;
+    moved = 0;
+    if (x != remx[0] || y != remy[0])
+    {
+        moved = 1;
+    }
+    if (moved == 1)
+    {
+        for (var i = 75; i > 0; i -= 1)
+        {
+            remx[i] = remx[i - 1];
+            remy[i] = remy[i - 1];
+            facing[i] = facing[i - 1];
+            sliding[i] = sliding[i - 1];
+            if (global.chapter == 4)
+            {
+                special[i] = special[i - 1];
+            }
+        }
+        remx[0] = x;
+        remy[0] = y;
+        sliding[0] = sliding;
+        facing[0] = global.facing;
+        if (global.chapter == 4)
+        {
+            special[0] = climbing;
+        }
+    }
 }
 if (fun == 0)
 {
@@ -611,22 +638,27 @@
     if (global.facing == 0)
     {
         sprite_index = dsprite;
+        visual_info.sprite_index = visual_info.dsprite;
     }
     if (global.facing == 1)
     {
         sprite_index = rsprite;
+        visual_info.sprite_index = visual_info.rsprite;
     }
     if (global.facing == 2)
     {
         sprite_index = usprite;
+        visual_info.sprite_index = visual_info.usprite;
     }
     if (global.facing == 3)
     {
         sprite_index = lsprite;
+        visual_info.sprite_index = visual_info.lsprite;
     }
     if (climbing == 1)
     {
         sprite_index = climbsprite;
+        visual_info.sprite_index = visual_info.climbsprite;
     }
 }
 if (stepping == 1 && fun == 0)
@@ -865,6 +897,7 @@
             lsprite = spr_krisl;
             usprite = spr_krisu;
         }
+        scr_set_chara_sprites(name, visual_info);
     }
 }
 if (scr_debug())
