--- a/Code/gml_Object_obj_lancerboss2_Step_0.gml
+++ b/Code/gml_Object_obj_lancerboss2_Step_0.gml
@@ -1,34 +1,45 @@
-if (firstskip == 1)
+if (!init)
 {
-    global.acting[0] = 1;
-    acting = 1;
-    with (obj_writer)
-    {
-        instance_destroy();
-    }
-    with (obj_face)
-    {
-        instance_destroy();
-    }
-    with (obj_smallface)
-    {
-        instance_destroy();
-    }
-    if (global.charaction[0] == 1)
-    {
-        global.attacking = 1;
-    }
-    global.charturn = 3;
-    global.myfight = 3;
+    init = true;
+    if (firstskip == 1)
+    {
+        prevhp = global.monsterhp[myself];
+        with (obj_writer)
+        {
+            instance_destroy();
+        }
+        with (obj_face)
+        {
+            instance_destroy();
+        }
+        with (obj_smallface)
+        {
+            instance_destroy();
+        }
+        if (global.charaction[0] == 1)
+        {
+            global.attacking = 1;
+        }
+        global.charturn = 3;
+        global.myfight = 3;
+    }
+    else
+    {
+        with (obj_herosusie)
+        {
+            idlesprite = spr_susieb_idle_serious;
+            attacksprite = spr_susieb_attack_serious;
+        }
+    }
 }
 if (global.monster[myself] == 1)
 {
     global.flag[51 + myself] = 4;
-    global.acting[0] = 1;
-    acting = 1;
-    if (global.mnfight == 1 && talked == 0)
+    if (global.mnfight == 1 && talked == 0 && !bubbleactive)
     {
+        global.acting[0] = 1;
         actcon = 0;
+        sparecon = 0;
         scr_randomtarget();
         if (!instance_exists(obj_darkener))
         {
@@ -57,42 +68,51 @@
     if (global.mnfight == 2 && attacked == 0)
     {
         rtimer += 1;
+        rudebuster = false;
         if (rtimer == 12)
         {
             global.turntimer = 140;
+            if (hitcount < 3)
+            {
+                var attack = choose(0, 1, 2);
+                if (turns < 3)
+                {
+                    attack = turns;
+                }
-            if (turns == 0 || turns == 2)
+                if (attack == 0 || attack == 2)
-            {
+                {
-                dc = instance_create(x, y, obj_dbulletcontroller);
+                    dc = instance_create(x, y, obj_dbulletcontroller);
-                dc.type = 20;
+                    dc.type = 20;
-                dc.target = mytarget;
+                    dc.target = mytarget;
-                dc.damage = global.monsterat[myself] * 5;
+                    dc.damage = global.monsterat[myself] * 5;
-                if (global.hp[global.char[0]] <= 70)
+                    if (global.hp[global.char[0]] <= 70)
-                {
+                    {
-                    dc.damage = global.monsterat[myself] * 3;
+                        dc.damage = global.monsterat[myself] * 3;
-                }
+                    }
-                global.turntimer = 180;
+                    global.turntimer = 180;
-            }
+                }
-            if (turns == 1)
+                if (attack == 1)
-            {
+                {
-                dc = instance_create(x, y, obj_dbulletcontroller);
+                    dc = instance_create(x, y, obj_dbulletcontroller);
-                dc.type = 21;
+                    dc.type = 21;
-                dc.target = mytarget;
+                    dc.target = mytarget;
-                dc.damage = global.monsterat[myself] * 5;
+                    dc.damage = global.monsterat[myself] * 5;
-                if (global.hp[global.char[0]] <= 70)
+                    if (global.hp[global.char[0]] <= 70)
-                {
+                    {
-                    dc.damage = global.monsterat[myself] * 3;
+                        dc.damage = global.monsterat[myself] * 3;
-                }
+                    }
-                global.turntimer = 180;
+                    global.turntimer = 180;
+                }
             }
-            if (turns >= 3)
+            else
             {
                 dc = instance_create(x, y, obj_dbulletcontroller);
-                dc.difficulty = turns * 2;
+                dc.difficulty = hitcount * 2;
-                if (turns == 6)
+                if (hitcount == 6)
                 {
                     dc.difficulty = 30;
                 }
-                if (turns == 7)
+                if (hitcount == 7)
                 {
                     dc.difficulty = 90;
                 }
@@ -104,7 +124,10 @@
             attacked = 1;
             global.typer = 6;
             global.fc = 0;
+            if (turns > 0)
+            {
-            global.battlemsg[0] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_88_0");
+                global.battlemsg[0] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_88_0");
+            }
         }
         else
         {
@@ -125,98 +148,639 @@
     yy = __view_get(e__VW.YView, 0);
     if (acting == 1 && actcon == 0)
     {
-        firstskip = 0;
-        global.typer = 53;
-        rr = choose(0, 1, 2, 3);
-        actcon = 1;
-        if (turns == 0)
-        {
-            global.msg[0] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_120_0");
-        }
-        if (turns == 1)
-        {
-            global.msg[0] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_121_0");
-        }
-        if (turns == 2)
-        {
-            global.msg[0] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_122_0");
-            global.msg[1] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_122_1");
-        }
-        if (turns == 3)
-        {
-            global.msg[0] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_123_0");
-        }
-        if (turns == 4)
-        {
-            global.msg[0] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_124_0");
-        }
-        if (turns == 5)
-        {
-            global.msg[0] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_127_0");
-            with (obj_herosusie)
-            {
-                idlesprite = spr_susieb_idle_serious;
-                attacksprite = spr_susieb_attack_serious;
-            }
-        }
-        if (turns == 6)
-        {
-            global.msg[0] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_135_0");
-            global.msg[1] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_136_0");
-            global.msg[2] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_137_0");
-            with (obj_herosusie)
-            {
-                idlesprite = spr_susieb_idle;
-                attacksprite = spr_susieb_attack;
-            }
-        }
-        if (turns == 7)
-        {
-            global.typer = 54;
-            global.msg[0] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_145_0");
-            global.msg[1] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_145_1");
-            global.msg[2] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_145_2");
-            actcon = 2;
-        }
-        global.monsterdf[myself] -= 5;
-        scr_enemyblcon(obj_herosusie.x + 100, obj_herosusie.y, 7);
-        with (obj_writer)
-        {
-            skippable = 0;
-        }
-    }
-    if (actcon == 1 && !instance_exists(obj_writer))
-    {
-        actcon = 0;
-        scr_attackphase();
-    }
-    if (actcon == 2 && !instance_exists(obj_writer))
-    {
-        snd_free_all();
-        snd_play(snd_laz_c);
-        black = scr_dark_marker(__view_get(e__VW.XView, 0) - 20, __view_get(e__VW.YView, 0) - 20, spr_pixel_white);
-        with (black)
-        {
-            image_blend = c_black;
-            depth = -10000;
-            image_xscale = 900;
-            image_yscale = 900;
-        }
+        actcon = 1;
+        if (hitcount <= 3)
+        {
+            var teamname = "us";
+            switch (global.flag[214])
+            {
+                case 1:
+                    teamname = "the $!$? Squad";
+                    break;
+                case 2:
+                    teamname = "the Lancer Fan Club";
+                    break;
+                case 3:
+                    teamname = "the Fun Gang";
+                    break;
+            }
+            global.msg[0] = "* LANCER - AT 7 DF 1&* Turned on " + teamname + " almost instantly./";
+            global.msg[1] = "* Trying his best to resolve this peacefully^1, but reluctantly standing his ground./%";
+        }
+        else if (hitcount < 7)
+        {
+            global.msg[0] = "* LANCER - AT 7 DF 0&* Can't take many more hits./";
+            global.msg[1] = "* Still hoping for peace^1, but losing hope^1. He looks defeated..^1. is he giving up?/%";
+        }
+        else
+        {
+            global.msg[0] = "* LANCER - AT 7 DF 0&* Accepting his fate./";
+            global.msg[1] = "* He stopped fighting back a while ago^1, now seemingly delaying the inevitable./%";
+        }
+        scr_battletext_default();
+        actcon = 1;
+    }
+    if (acting == 2 && actcon == 0)
+    {
+        actcon = 1;
+        if (talkcon == 0)
+        {
+            global.msg[0] = "* You tried to tell Lancer you don't want to hurt him./";
+            if (scr_havechar(2))
+            {
+                if (scr_susiemode())
+                {
+                    global.msg[0] = "* You tried to tell Lancer you don't want to hurt him./%";
+                    actcon = 2;
+                }
+                else
+                {
+                    global.msg[1] = "* ... He cautiously glances at Susie.^1&* She looks conflicted./";
+                    global.msg[2] = "* You realise Lancer isn't the only one needing reassurance./%";
+                }
+            }
+            else
+            {
+                global.msg[1] = "* ... He looks up for a second^1, but.../%";
+            }
+        }
+        if (talkcon == 1)
+        {
+            if (scr_havechar(2))
+            {
+                if (scr_susiemode())
+                {
+                    global.msg[0] = "* You asked Lancer why he's fighting./";
+                    scr_susface(1, 1);
+                    global.msg[2] = "* ... Doesn't matter./";
+                    scr_lanface(3, "A");
+                    global.msg[4] = "* Susie^1, please-/";
+                    scr_susface(5, 3);
+                    global.msg[6] = "* Doesn't^2. Matter./%";
+                }
+                else
+                {
+                    global.msg[0] = "* You asked Lancer why he's fighting.^1&* He looks away./";
+                    scr_lanface(1, "B");
+                    global.msg[2] = "* I..^1. if I let you go^1, then you'll.../";
+                    global.msg[3] = "\\EA* ... I didn't have a choice./";
+                    scr_susface(4, "C");
+                    global.msg[5] = "* .../%";
+                }
+            }
+            else
+            {
+                global.msg[0] = "* You asked Lancer why he's fighting.^1&* He looks away./";
+                global.msg[1] = "* You might need to try something else./%";
+            }
+        }
+        if (talkcon >= 2)
+        {
+            if (scr_havechar(2))
+            {
+                if (scr_susiemode())
+                {
+                    global.msg[0] = "* It doesn't feel like talking is an option.^1&* ... mostly because of Susie./%";
+                }
+                else
+                {
+                    global.msg[0] = "* You couldn't think of anything to say.^1&* ... maybe Susie could help?/%";
+                }
+            }
+            else
+            {
+                global.msg[0] = "* You couldn't think of anything to say./";
+                global.msg[1] = "* ... is there another way to resolve things peacefully?/%";
+            }
+        }
+        talkcon++;
+        scr_battletext_default();
+    }
+    if (acting == 2 && actcon == 2 && !instance_exists(obj_writer))
+    {
+        laughsound = snd_play(snd_suslaugh);
+        global.msg[0] = "* ... Susie laughs menacingly.^1&* Lancer takes a step back./%";
+        with (obj_herosusie)
+        {
+            visible = 0;
+        }
+        suslaugh = scr_dark_marker(obj_herosusie.x, obj_herosusie.y, spr_susier_dark_laugh);
+        with (suslaugh)
+        {
+            image_speed = 0.2;
+        }
+        laughtimer = 0;
+        scr_battletext_default();
         actcon = 3;
-        acttimer = 0;
-        if (instance_exists(obj_lancerbattle2_event))
-        {
-            obj_lancerbattle2_event.black = black;
-            with (obj_lancerbattle2_event)
-            {
-                con = 52;
-                alarm[4] = 80;
-            }
-        }
-    }
-    if (actcon == 3)
-    {
-    }
+    }
+    if (acting == 2 && actcon == 3)
+    {
+        laughtimer += 1;
+        if (laughtimer >= 30)
+        {
+            actcon = 4;
+        }
+    }
+    if (acting == 2 && actcon == 4 && !instance_exists(obj_writer))
+    {
+        snd_stop(laughsound);
+        with (suslaugh)
+        {
+            instance_destroy();
+        }
+        with (obj_herosusie)
+        {
+            visible = 1;
+        }
+        actcon = 1;
+    }
+    if (acting == 3 && actcon == 0)
+    {
+        actcon = 1;
+        if (convincecon == 0)
+        {
+            if (scr_susiemode())
+            {
+                global.msg[0] = "* You tried to convince Susie to stop fighting./";
+                scr_susface(1, 0);
+                global.msg[2] = "* ... Why^1? He CHOSE to betray us!/";
+                global.msg[3] = "\\E4* I'm just making sure it NEVER happens again./%";
+            }
+            else
+            {
+                global.msg[0] = "* You tried to get Susie to lower her guard./";
+                scr_susface(1, "C");
+                global.msg[2] = "* .../";
+                scr_noface(3);
+                global.msg[4] = "* She pauses for a moment.^1&* The grip on her axe loosens slightly./%";
+            }
+        }
+        if (convincecon == 1)
+        {
+            if (scr_susiemode())
+            {
+                global.msg[0] = "* You tried to convince Susie to reason with Lancer./";
+                global.msg[1] = "* ... The grip on her axe tightens.^1&* Your words fall on deaf ears./%";
+            }
+            else
+            {
+                global.msg[0] = "* You told Susie that Lancer doesn't want to fight./";
+                scr_susface(1, "C");
+                global.msg[2] = "* ... But...^2 he's.../";
+                scr_lanface(3, "B");
+                global.msg[4] = "* S-Susie...? Your hands are shaking.../%";
+            }
+        }
+        if (convincecon >= 2)
+        {
+            if (scr_susiemode())
+            {
+                global.msg[0] = "* You kept trying to talk to Susie.&* ... She still isn't listening./%";
+            }
+            else
+            {
+                global.msg[0] = "* You convinced Susie to reason with Lancer./%";
+                actcon = 2;
+                peacecon = 1;
+            }
+        }
+        convincecon++;
+        scr_battletext_default();
+    }
+    if (acting != 99 && actcon < 2 && !instance_exists(obj_writer))
+    {
+        acting = 99;
+        actcon = 10;
+    }
+    if (acting == 99 && actcon == 10)
+    {
+        actcon = 11;
+        washit = false;
+        if (prevhp > global.monsterhp[myself])
+        {
+            hitcount++;
+            washit = true;
+            prevhp = global.monsterhp[myself];
+        }
+        var magicking = false;
+        if (global.charaction[0] == 4 || global.charaction[0] == 2)
+        {
+            magicking = true;
+        }
+        if (global.charaction[1] == 4 || global.charaction[1] == 2)
+        {
+            magicking = true;
+        }
+        if (global.charaction[2] == 4 || global.charaction[2] == 2)
+        {
+            magicking = true;
+        }
+        if (magicking || !(scr_havechar(2) && scr_susiemode()))
+        {
+            scr_attackphase();
+        }
+        else
+        {
+            bubbleactive = true;
+            bubble = 2;
+        }
+    }
+}
+if (global.myfight == 4)
+{
+    if (sparecon == 1)
+    {
+        if ((global.attacking || rudebuster) && !(scr_havechar(2) && scr_susiemode()))
+        {
+            sparecon = 0;
+        }
+        else if (instance_exists(obj_spellphase) && (obj_spellphase.char >= 3 || obj_spellphase.spelltotal == 1))
+        {
+            sparecon = 2;
+        }
+    }
+    if (sparecon == 2)
+    {
+        global.spelldelay = 999;
+        with (obj_spellphase)
+        {
+            if (spelltimer > 30)
+            {
+                spelltimer = 30;
+            }
+        }
+    }
+    if (sparecon == 2 && !instance_exists(obj_writer))
+    {
+        if (scr_havechar(2) && scr_susiemode())
+        {
+            bubbleactive = true;
+            bubble = 2;
+        }
+        else
+        {
+            bubbleactive = true;
+            bubble = 1;
+        }
+    }
+    if (sparecon == 3 && !instance_exists(obj_writer))
+    {
+        sparecon = 0;
+        sparecount++;
+        global.spelldelay = 0;
+    }
+}
+if (bubbleactive)
+{
+    if (bubble == 1)
+    {
+        if (bubblecon == 0)
+        {
+            global.typer = 56;
+            rr = choose(0, 1, 2, 3);
+            bubblecon = 3;
+            var skiplancer = false;
+            if (sparecount == 0)
+            {
+                global.msg[0] = ".../%";
+            }
+            if (sparecount == 1)
+            {
+                if (scr_havechar(2))
+                {
+                    global.msg[0] = "...?/%";
+                }
+                else
+                {
+                    global.msg[0] = "...&.../%";
+                }
+            }
+            if (sparecount == 2)
+            {
+                if (scr_havechar(2))
+                {
+                    global.msg[0] = "...S-Susie?/%";
+                }
+                else
+                {
+                    global.msg[0] = "...?/%";
+                }
+            }
+            if (sparecount == 3)
+            {
+                if (scr_havechar(2))
+                {
+                    global.msg[0] = "W-what are&you doing?/%";
+                    bubblecon = 1;
+                }
+                else
+                {
+                    global.msg[0] = "I-I'm sorry,&I.../";
+                    global.msg[1] = "...I don't&know what&else to do./%";
+                }
+            }
+            if (sparecount == 4)
+            {
+                if (scr_havechar(2))
+                {
+                    bubblecon = 1;
+                    skiplancer = true;
+                }
+                else
+                {
+                    global.msg[0] = "...I-I just&don't want&anyone to&get hurt./";
+                    global.msg[1] = "If I let&you guys &go, then.../%";
+                    if (scr_havechar(3))
+                    {
+                        bubblecon = 1;
+                    }
+                }
+            }
+            if (sparecount == 5)
+            {
+                if (scr_havechar(2))
+                {
+                    global.msg[0] = "I-If you&keep going,&then.../";
+                    global.msg[1] = "You'll&have to.../%";
+                    bubblecon = 1;
+                }
+                else if (scr_havechar(3))
+                {
+                    bubblecon = 1;
+                    skiplancer = true;
+                }
+                else
+                {
+                    global.msg[0] = "You guys&would have&to fight my&dad, and.../";
+                    global.msg[1] = "...You'd kill&each other./%";
+                }
+            }
+            if (sparecount == 6)
+            {
+                if (scr_havechar(3))
+                {
+                    bubblecon = 1;
+                    skiplancer = true;
+                }
+                else
+                {
+                    global.msg[0] = ".../%";
+                    bubblecon = 4;
+                }
+            }
+            global.monsterdf[myself] -= 5;
+            if (!skiplancer)
+            {
+                scr_enemyblcon(x - 160, y, 3);
+                with (obj_writer)
+                {
+                    skippable = 0;
+                }
+            }
+        }
+        if (bubblecon == 1 && !instance_exists(obj_writer))
+        {
+            global.typer = 53;
+            bubblecon = 3;
+            if (sparecount == 3)
+            {
+                global.msg[0] = ".../%";
+            }
+            if (sparecount == 4)
+            {
+                if (scr_havechar(2))
+                {
+                    global.msg[0] = "Just...&let us go./";
+                    global.msg[1] = "... Please./%";
+                    bubblecon = 2;
+                }
+                else
+                {
+                    global.typer = 57;
+                    global.msg[0] = "It's okay,&Lancer.../";
+                    global.msg[1] = "We don't&want to hurt&anyone./%";
+                    bubblecon = 2;
+                }
+            }
+            if (sparecount == 5)
+            {
+                if (scr_havechar(2))
+                {
+                    global.msg[0] = ".../%";
+                    bubblecon = 4;
+                }
+                else
+                {
+                    global.typer = 57;
+                    global.msg[0] = "Don't worry,&Susie will be&alright./";
+                    global.msg[1] = "She's strong!&She can take&care of&herself./%";
+                    bubblecon = 2;
+                }
+            }
+            if (sparecount == 6)
+            {
+                global.typer = 57;
+                global.msg[0] = "Lancer, you&and Susie are&friends,&right?/";
+                if (scr_susiemode())
+                {
+                    global.msg[1] = "I know she's&been a bit...&violent,&but.../";
+                }
+                else
+                {
+                    global.msg[1] = "I know she&doesn't like&taking orders,&but.../";
+                }
+                global.msg[2] = "If there's&anyone she'll&listen to,&it's you./%";
+                bubblecon = 2;
+            }
+            if (instance_exists(obj_herosusie))
+            {
+                scr_enemyblcon(obj_herosusie.x + 100, obj_herosusie.y, 7);
+            }
+            else if (instance_exists(obj_heroralsei))
+            {
+                scr_enemyblcon(obj_heroralsei.x + 100, obj_heroralsei.y, 7);
+            }
+            with (obj_writer)
+            {
+                skippable = 0;
+            }
+        }
+        if (bubblecon == 2 && !instance_exists(obj_writer))
+        {
+            global.typer = 56;
+            bubblecon = 3;
+            if (sparecount == 4)
+            {
+                if (scr_havechar(2))
+                {
+                    global.msg[0] = "I...&I can't./";
+                    global.msg[1] = "I-I have&to do this./%";
+                }
+                else
+                {
+                    global.msg[0] = "B-but, Susie&would.../%";
+                }
+            }
+            if (sparecount == 5)
+            {
+                global.msg[0] = "I-I know,&but.../";
+                global.msg[1] = "She'd have&to fight my&dad, and.../%";
+            }
+            if (sparecount == 6)
+            {
+                global.msg[0] = ".../%";
+                bubblecon = 4;
+            }
+            scr_enemyblcon(x - 160, y, 3);
+            with (obj_writer)
+            {
+                skippable = 0;
+            }
+        }
+        if (bubblecon == 3 && !instance_exists(obj_writer))
+        {
+            bubble = 0;
+        }
+        if (bubblecon == 4)
+        {
+            bubblecon = 5;
+            peacecon = 1;
+        }
+        if (bubblecon == 5)
+        {
+        }
+    }
+    if (bubble == 2)
+    {
+        if (bubblecon == 0)
+        {
+            global.typer = 53;
+            rr = choose(0, 1, 2, 3);
+            bubblecon = 1;
+            if (hitcount == 0)
+            {
+                global.msg[0] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_120_0");
+            }
+            if (hitcount == 1)
+            {
+                global.msg[0] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_121_0");
+            }
+            if (hitcount == 2)
+            {
+                global.msg[0] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_122_0");
+                global.msg[1] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_122_1");
+            }
+            if (hitcount == 3)
+            {
+                global.msg[0] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_123_0");
+            }
+            if (hitcount == 4)
+            {
+                global.msg[0] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_124_0");
+            }
+            if (hitcount == 5)
+            {
+                global.msg[0] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_127_0");
+                with (obj_herosusie)
+                {
+                    idlesprite = spr_susieb_idle_serious;
+                    attacksprite = spr_susieb_attack_serious;
+                }
+            }
+            if (hitcount == 6)
+            {
+                global.msg[0] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_135_0");
+                global.msg[1] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_136_0");
+                global.msg[2] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_137_0");
+                with (obj_herosusie)
+                {
+                    idlesprite = spr_susieb_idle;
+                    attacksprite = spr_susieb_attack;
+                }
+            }
+            if (hitcount == 7)
+            {
+                global.typer = 54;
+                global.msg[0] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_145_0");
+                global.msg[1] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_145_1");
+                global.msg[2] = scr_84_get_lang_string("obj_lancerboss2_slash_Step_0_gml_145_2");
+                bubblecon = 2;
+            }
+            global.monsterdf[myself] -= 5;
+            if (instance_exists(obj_herosusie))
+            {
+                scr_enemyblcon(obj_herosusie.x + 100, obj_herosusie.y, 7);
+            }
+            with (obj_writer)
+            {
+                skippable = 0;
+            }
+        }
+        if (bubblecon == 1 && !instance_exists(obj_writer))
+        {
+            bubble = 0;
+        }
+        if (bubblecon == 2 && !instance_exists(obj_writer))
+        {
+            killcon = 1;
+            bubblecon = 3;
+        }
+        if (bubblecon == 3)
+        {
+        }
+    }
+    if (bubble == 0)
+    {
+        bubblecon = 0;
+        bubbleactive = false;
+        if (global.myfight == 3)
+        {
+            actcon = 0;
+            scr_attackphase();
+        }
+        else
+        {
+            sparecon = 3;
+        }
+    }
+}
+if (killcon == 0 && global.monsterhp[myself] <= 0)
+{
+    killcon = 1;
+}
+if (killcon == 1)
+{
+    snd_free_all();
+    snd_play(snd_laz_c);
+    black = scr_dark_marker(__view_get(e__VW.XView, 0) - 20, __view_get(e__VW.YView, 0) - 20, spr_pixel_white);
+    with (black)
+    {
+        image_blend = c_black;
+        depth = -10000;
+        image_xscale = 900;
+        image_yscale = 900;
+    }
+    killcon = 2;
+    acttimer = 0;
+    if (instance_exists(obj_lancerbattle2_event))
+    {
+        obj_lancerbattle2_event.black = black;
+        with (obj_lancerbattle2_event)
+        {
+            con = 52;
+            alarm[4] = 80;
+        }
+    }
+}
+if (peacecon == 1 && !instance_exists(obj_writer))
+{
+    with (obj_battlecontroller)
+    {
+        skipvictory = 1;
+    }
+    scr_monsterdefeat();
+    scr_mnendturn();
+    peacecon = 2;
 }
 
 enum e__VW
