--- a/Code/gml_GlobalScript_scr_text.gml
+++ b/Code/gml_GlobalScript_scr_text.gml
@@ -276,22 +276,82 @@
             global.choicemsg[1] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_289_0");
             global.choicemsg[2] = " ";
             global.choicemsg[3] = " ";
+            if (scr_havechar(1) || scr_havechar(2))
+            {
+                if (scr_havechar(2))
+                {
+                    global.msg[0] = "\\E3* Oh^1, Susie^1!&* It's the training dummy I made!/";
+                }
+                else
+                {
-            global.msg[0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_292_0");
+                    global.msg[0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_292_0");
-            global.msg[1] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_293_0");
+                }
-            global.msg[2] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_294_0");
+                global.msg[1] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_293_0");
-            global.msg[3] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_295_0");
+                global.msg[2] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_294_0");
-            global.msg[4] = " ";
+                if (scr_havechar(2))
+                {
+                    scr_susface(3, 0);
+                    global.msg[4] = "* I'll pass./";
+                    if (scr_havechar(1))
+                    {
+                        scr_ralface(5, 1);
+                        global.msg[6] = "* OK, umm.../";
+                        global.msg[7] = "\\E6* What about you, Kris?/";
+                        global.msg[8] = "\\C2";
+                    }
+                    else
+                    {
+                        scr_ralface(5, 3);
+                        global.msg[6] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_307_0");
+                        global.msg[7] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_308_0");
+                        global.msg[8] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_309_0");
+                        global.msg[9] = "\\E6* Here!&* I made a Manual for you and Kris!/";
+                        global.msg[10] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_311_0") + scr_get_input_name(6) + scr_84_get_lang_string("scr_text_slash_scr_text_gml_311_1");
+                        scr_noface(11);
+                        global.msg[12] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_313_0");
+                    }
+                }
+                else
+                {
+                    global.msg[3] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_295_0");
+                }
+            }
+            else
+            {
+                global.msg[0] = "\\E3* Oh, I nearly forgot about the training dummy!/";
+                global.msg[1] = "\\EB* I was going to teach Kris and Susie about battles.../";
+                global.msg[2] = "\\E1* .../";
+                global.msg[3] = "\\E6* I suppose... another rehearsal couldn’t hurt, right?/";
+                global.msg[4] = "\\C2";
+            }
             break;
         case 136:
             if (global.choice == 0)
             {
+                if (scr_havechar(1) && scr_havechar(2))
+                {
+                    global.msg[0] = "\\TX\\F0\\E7\\FS\\TS* Arright, fine!/";
+                    global.msg[1] = "\\E0* Let's just crush that dummy and move on./";
+                    scr_ralface(2, 6);
+                    global.msg[3] = "* Um... There's more to battles than just attacking, Susie./";
+                    scr_susface(4, 2);
+                    global.msg[5] = "* Look, you want a battle?/";
+                    global.msg[6] = "\\E4* I'll SHOW you a battle!!/%";
+                }
+                else if (scr_havechar(1))
+                {
-                global.msg[0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_302_0");
+                    global.msg[0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_302_0");
+                }
+                else
+                {
+                    global.msg[0] = "\\E0* OK! Just like I practiced, like Kris and Susie are here./%";
+                }
                 with (obj_tutorialbattleevent)
                 {
                     con = 20;
                 }
             }
-            else
+            else if (scr_havechar(1))
             {
                 global.msg[0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_307_0");
                 global.msg[1] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_308_0");
@@ -301,6 +361,14 @@
                 scr_noface(5);
                 global.msg[6] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_313_0");
             }
+            else
+            {
+                global.msg[0] = "\\EB* On second thought, I should catch up with Kris and Susie./";
+                global.msg[1] = "\\E2* Since I have to leave the training dummy behind.../";
+                global.msg[2] = "\\E0* I need to give them the Manual I wrote when I meet them!/";
+                scr_noface(3);
+                global.msg[4] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_313_0");
+            }
             break;
         case 140:
             global.choicemsg[0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_318_0");
@@ -1150,6 +1218,10 @@
                 global.msg[4] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1218_0");
                 scr_noface(5);
                 global.msg[6] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1220_0");
+                if (!scr_havechar(3))
+                {
+                    global.msg[2] = global.msg[2] + "%";
+                }
             }
             if (global.choice == 1)
             {
@@ -1159,6 +1231,10 @@
                 global.msg[3] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1227_0");
                 scr_noface(4);
                 global.msg[5] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1229_0");
+                if (!scr_havechar(3))
+                {
+                    global.msg[1] = global.msg[1] + "%";
+                }
             }
             if (global.choice == 2)
             {
@@ -1171,8 +1247,15 @@
                 global.msg[6] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1239_0");
                 if (!scr_havechar(2))
                 {
+                    if (scr_havechar(3))
+                    {
-                    scr_ralface(3, 1);
+                        scr_ralface(3, 1);
-                    global.msg[4] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1244_0");
+                        global.msg[4] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1244_0");
+                    }
+                    else
+                    {
+                        global.msg[2] = global.msg[2] + "%";
+                    }
                 }
             }
             break;
@@ -1193,6 +1276,11 @@
                 global.msg[2] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1265_0");
                 scr_noface(3);
                 global.msg[4] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1267_0");
+                if (!scr_havechar(3))
+                {
+                    global.msg[0] = "* Ralsei's gone^1, someway^1, somehow./";
+                    global.msg[1] = "* How will you deal with babies now?/%";
+                }
             }
             if (global.choice == 1)
             {
@@ -1202,6 +1290,11 @@
                 global.msg[3] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1275_0");
                 scr_noface(4);
                 global.msg[5] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1277_0");
+                if (!scr_havechar(3))
+                {
+                    global.msg[0] = "* Ralsei's gone^1, so no more HEALS./";
+                    global.msg[1] = "* Unless you eat a delicious MEALS./%";
+                }
             }
             if (global.choice == 2)
             {
@@ -1210,6 +1303,18 @@
                 scr_ralface(2, "A");
                 global.msg[3] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1284_0");
                 global.msg[4] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1285_0");
+                if (!scr_havechar(3))
+                {
+                    global.msg[0] = "* Ralsei's gone^1, so no facts./";
+                    if (scr_havechar(2) && scr_havechar(1))
+                    {
+                        global.msg[1] = "* Will you two pick up the slackts?/%";
+                    }
+                    else
+                    {
+                        global.msg[1] = "* Think you can pick up the slackts?/%";
+                    }
+                }
             }
             break;
         case 240:
@@ -1236,7 +1341,7 @@
                     global.msg[0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1312_0");
                     global.msg[1] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1313_0");
                 }
-                if (global.plot >= 150)
+                else if (!scr_susiemode())
                 {
                     global.msg[0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1317_0");
                     global.msg[1] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1318_0");
@@ -1254,9 +1359,16 @@
                 if (!scr_havechar(2))
                 {
                     global.msg[0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1335_0");
+                    if (scr_havechar(3) && scr_havechar(1))
+                    {
-                    global.msg[1] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1336_0");
+                        global.msg[1] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1336_0");
+                    }
+                    else
+                    {
+                        global.msg[1] = "* Think you can pick up the slack?/%";
+                    }
                 }
-                if (global.plot >= 150)
+                else if (!scr_susiemode())
                 {
                     input_name = scr_get_input_name(4);
                     global.msg[0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1341_0");
@@ -1273,9 +1385,16 @@
                 if (!scr_havechar(2))
                 {
                     global.msg[0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1355_0");
+                    if (scr_havechar(3) && scr_havechar(1))
+                    {
-                    global.msg[1] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1356_0");
+                        global.msg[1] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1356_0");
+                    }
+                    else
+                    {
+                        global.msg[1] = "* Think you can pick up the slackts?/%";
+                    }
                 }
-                if (global.plot >= 150)
+                else if (scr_susiemode())
                 {
                     global.msg[0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1360_0");
                     global.msg[1] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1361_0");
@@ -1294,14 +1413,25 @@
             global.msg[0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1382_0");
             if (global.choice == 0)
             {
+                var num = 0;
+                if (!scr_havechar(1))
+                {
+                    global.msg[0] = "* Kris's gone^1, but you're in luck./";
+                    global.msg[1] = "* Everyone can use this^1, even a duck!/";
+                    num = 2;
+                }
-                global.msg[0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1385_0");
+                global.msg[num + 0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1385_0");
-                global.msg[1] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1386_0");
+                global.msg[num + 1] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1386_0");
-                global.msg[2] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1387_0");
+                global.msg[num + 2] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1387_0");
-                global.msg[3] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1388_0");
+                global.msg[num + 3] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1388_0");
-                scr_ralface(4, 0);
+                scr_ralface(num + 4, 0);
-                global.msg[5] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1390_0");
+                global.msg[num + 5] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1390_0");
-                scr_noface(6);
+                scr_noface(num + 6);
-                global.msg[7] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1392_0");
+                global.msg[num + 7] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1392_0");
+                if (!scr_havechar(3))
+                {
+                    global.msg[num + 3] = global.msg[num + 3] + "%";
+                }
             }
             if (global.choice == 1)
             {
@@ -1314,6 +1444,26 @@
                 global.msg[6] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1403_0");
                 scr_noface(7);
                 global.msg[8] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1405_0");
+                if (!scr_havechar(1))
+                {
+                    if (scr_havechar(3))
+                    {
+                        global.msg[0] = "* Without Kris to ACT^1, you might start to think -/";
+                        global.msg[3] = "* Enemies will still get TIRED when their HP drops too low.../";
+                        global.msg[4] = "* So if I'm careful^1, I can still cast PACIFY to end the battle peacefully!/";
+                        scr_noface(5);
+                        global.msg[6] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1405_0");
+                    }
+                    else
+                    {
+                        global.msg[0] = "* Kris's gone^1, so you can't use ACTs./";
+                        global.msg[1] = "* SPAREing enemies^1, will be quite the task./";
+                    }
+                }
+                if (!scr_havechar(3))
+                {
+                    global.msg[1] = global.msg[1] + "%";
+                }
             }
             if (global.choice == 2)
             {
@@ -1366,10 +1516,20 @@
             }
             break;
         case 255:
+            if (scr_havechar(3))
+            {
-            global.choicemsg[0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1451_0");
+                global.choicemsg[0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1451_0");
-            global.choicemsg[1] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1452_0");
+                global.choicemsg[1] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1452_0");
-            global.choicemsg[2] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1453_0");
+                global.choicemsg[2] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1453_0");
-            global.choicemsg[3] = " ";
+                global.choicemsg[3] = " ";
+            }
+            else
+            {
+                global.choicemsg[0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1452_0");
+                global.choicemsg[1] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1453_0");
+                global.choicemsg[2] = " ";
+                global.choicemsg[3] = " ";
+            }
             global.msg[0] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1455_0");
             scr_susface(1, 6);
             global.msg[2] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1457_0");
@@ -1390,10 +1550,21 @@
             global.msg[17] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1472_0");
             scr_ralface(18, 8);
             global.msg[19] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1474_0");
+            if (scr_havechar(3))
+            {
-            global.msg[20] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1475_0");
+                global.msg[20] = scr_84_get_lang_string("scr_text_slash_scr_text_gml_1475_0");
+            }
+            else
+            {
+                global.msg[20] = "\\C2";
+            }
             break;
         case 256:
             global.flag[232] = 1;
+            if (!scr_havechar(3))
+            {
+                global.choice++;
+            }
             if (global.choice == 0)
             {
                 scr_ralface(0, 6);
