--- a/Code/gml_Object_obj_attackpress_Create_0.gml
+++ b/Code/gml_Object_obj_attackpress_Create_0.gml
@@ -242,23 +242,17 @@
 autoed = 0;
 if (global.charauto[2] == 1)
 {
-    if (global.char[0] == 2 || global.char[1] == 2 || global.char[2] == 2)
+    for (var pos = 0; pos < 3; pos++)
     {
-        sus = 0;
-        if (global.char[1] == 2)
+        if (global.char[pos] == 2)
         {
-            sus = 1;
-        }
-        if (global.char[2] == 2)
-        {
-            sus = 2;
-        }
-        if (global.hp[2] >= 0 && global.charmove[sus] == 1)
-        {
-            haveauto = 1;
-            if (timermax == 3)
+            if (global.hp[2] >= 0 && global.charmove[pos] == 1)
             {
+                haveauto = 1;
+                if (timermax == 3)
+                {
-                timermax = 50;
+                    timermax = 50;
+                }
             }
         }
     }
