--- a/Code/gml_Object_obj_smallcheckers_enemy_Draw_0.gml
+++ b/Code/gml_Object_obj_smallcheckers_enemy_Draw_0.gml
@@ -44,7 +44,14 @@
     thissprite = spr_smallchecker_idle;
     if (global.mercymod[myself] >= global.mercymax[myself])
     {
+        if (scr_susiemode())
+        {
-        thissprite = spr_smallchecker_run;
+            thissprite = spr_smallchecker_run;
+        }
+        else
+        {
+            thissprite = spr_smallchecker_spared;
+        }
     }
     draw_sprite_ext(thissprite, siner / 3, x, y, 2, 2, 0, image_blend, 1);
     if (flash == 1)
