--- a/Code/gml_GlobalScript_scr_monstersetup.gml
+++ b/Code/gml_GlobalScript_scr_monstersetup.gml
@@ -103,7 +103,7 @@
         global.actname[myself][1] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_117_0");
         global.canact[myself][2] = 1;
         global.actname[myself][2] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_119_0");
-        if (scr_havechar(2) && global.plot < 150)
+        if (scr_havechar(2) && scr_susiemode())
         {
             global.canact[myself][3] = 1;
             global.actname[myself][3] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_123_0");
@@ -137,13 +137,13 @@
                 global.plot = 40;
             }
         }
-        if (scr_havechar(2) && global.plot < 150)
+        if (scr_havechar(2) && scr_susiemode())
         {
             global.canact[myself][3] = 1;
             global.actname[myself][3] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_154_0");
             global.actactor[myself][3] = 3;
         }
-        if (scr_havechar(2) && global.plot >= 150)
+        if (scr_havechar(2) && !scr_susiemode())
         {
             global.canact[myself][3] = 1;
             global.actname[myself][3] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_160_0");
@@ -192,7 +192,7 @@
             global.canact[myself][3] = 1;
             global.actname[myself][3] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_212_0");
         }
-        if (scr_havechar(2))
+        if (scr_havechar(2) && scr_susiemode())
         {
             global.canact[myself][4] = 1;
             global.actname[myself][4] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_218_0");
@@ -224,7 +224,7 @@
         global.monstername[myself] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_251_0");
         global.monstermaxhp[myself] = 10;
         global.monsterhp[myself] = 10;
-        global.monsterat[myself] = 5;
+        global.monsterat[myself] = 3;
         global.monsterdf[myself] = 0;
         global.monsterexp[myself] = 0;
         global.monstergold[myself] = 10;
@@ -245,9 +245,26 @@
             global.canact[myself][1] = 1;
             global.actactor[myself][1] = 2;
             global.actname[myself][1] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_271_0");
+            if (scr_susiemode())
+            {
-            global.canact[myself][2] = 1;
+                global.canact[myself][2] = 1;
-            global.actname[myself][2] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_274_0");
+                global.actname[myself][2] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_274_0");
-            global.actactor[myself][2] = 3;
+                if (scr_havechar(3))
+                {
+                    global.actactor[myself][2] = 3;
+                }
+            }
+        }
+        else if (scr_havechar(3))
+        {
+            global.canact[myself][1] = 1;
+            global.actactor[myself][1] = 3;
+            global.actname[myself][1] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_271_0");
+        }
+        else
+        {
+            global.canact[myself][1] = 1;
+            global.actname[myself][1] = "Compliment";
         }
     }
     if (global.monstertype[myself] == 10)
@@ -266,14 +283,26 @@
         global.actname[myself][0] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_294_0");
         global.canact[myself][1] = 1;
         global.actname[myself][1] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_296_0");
+        var actnum = 2;
+        if (!scr_susiemode() && scr_havechar(2))
+        {
-        global.canact[myself][2] = 1;
+            global.canact[myself][actnum] = 1;
-        global.actname[myself][2] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_298_0");
+            global.actname[myself][actnum] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_611_0");
-        global.actactor[myself][2] = 3;
+            global.actactor[myself][actnum] = 4;
+            actnum++;
+        }
-        if (scr_havechar(2))
+        else if (scr_havechar(3))
         {
-            global.canact[myself][3] = 1;
+            global.canact[myself][actnum] = 1;
-            global.actname[myself][3] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_307_0");
+            global.actname[myself][actnum] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_298_0");
-            global.actactor[myself][3] = 3;
+            global.actactor[myself][actnum] = 3;
+            actnum++;
+        }
+        if (scr_havechar(2) && scr_susiemode())
+        {
+            global.canact[myself][actnum] = 1;
+            global.actname[myself][actnum] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_307_0");
+            global.actactor[myself][actnum] = 3;
         }
     }
     if (global.monstertype[myself] == 11)
@@ -295,7 +324,7 @@
         global.canact[myself][2] = 1;
         global.actname[myself][2] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_331_0");
         global.actactor[myself][2] = 3;
-        if (scr_havechar(2) && global.plot < 150)
+        if (scr_havechar(2) && scr_susiemode())
         {
             global.canact[myself][3] = 1;
             global.actname[myself][3] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_337_0");
@@ -316,6 +345,9 @@
         global.mercymax[myself] = 100;
         global.canact[myself][0] = 1;
         global.actname[myself][0] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_358_0");
+        global.canact[myself][1] = 1;
+        global.actname[myself][1] = "death tester";
+        global.actactor[myself][1] = 2;
     }
     if (global.monstertype[myself] == 13)
     {
@@ -336,7 +368,7 @@
         global.canact[myself][2] = 1;
         global.actname[myself][2] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_381_0");
         global.actactor[myself][2] = 3;
-        if (scr_havechar(2) && global.plot < 150)
+        if (scr_havechar(2) && scr_susiemode())
         {
             global.canact[myself][3] = 1;
             global.actname[myself][3] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_386_0");
@@ -359,7 +391,7 @@
         global.actname[myself][0] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_406_0");
         global.canact[myself][1] = 1;
         global.actname[myself][1] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_408_0");
-        if (scr_havechar(2) && global.plot >= 150)
+        if (scr_havechar(2) && scr_susiemode())
         {
             global.canact[myself][2] = 1;
             global.actname[myself][2] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_412_0");
@@ -382,7 +414,7 @@
         global.actname[myself][0] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_432_0");
         global.canact[myself][1] = 1;
         global.actname[myself][1] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_434_0");
-        if (scr_havechar(2) && global.plot < 150)
+        if (scr_havechar(2) && scr_susiemode())
         {
             global.canact[myself][2] = 1;
             global.actname[myself][2] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_440_0");
@@ -529,9 +561,27 @@
         }
         global.canact[myself][1] = 1;
         global.actname[myself][1] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_609_0");
+        var actnum = 2;
+        if (!scr_susiemode() && scr_havechar(2))
+        {
-        global.canact[myself][2] = 1;
+            global.canact[myself][actnum] = 1;
-        global.actname[myself][2] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_611_0");
+            global.actname[myself][actnum] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_611_0");
-        global.actactor[myself][2] = 4;
+            global.actactor[myself][actnum] = 4;
+            actnum++;
+        }
+        else if (scr_havechar(3))
+        {
+            global.canact[myself][actnum] = 1;
+            global.actname[myself][actnum] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_298_0");
+            global.actactor[myself][actnum] = 3;
+            actnum++;
+        }
+        if (scr_havechar(2) && scr_susiemode())
+        {
+            global.canact[myself][actnum] = 1;
+            global.actname[myself][actnum] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_307_0");
+            global.actactor[myself][actnum] = 3;
+        }
     }
     if (global.monstertype[myself] == 22)
     {
@@ -549,9 +599,18 @@
         global.actname[myself][0] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_631_0");
         global.canact[myself][1] = 1;
         global.actname[myself][1] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_633_0");
+        if (scr_havechar(2) && !scr_susiemode())
+        {
-        global.canact[myself][2] = 1;
+            global.canact[myself][2] = 1;
-        global.actname[myself][2] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_635_0");
+            global.actname[myself][2] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_635_0");
-        global.actactor[myself][2] = 2;
+            global.actactor[myself][2] = 2;
+        }
+        if (scr_havechar(2) && scr_susiemode())
+        {
+            global.canact[myself][2] = 1;
+            global.actname[myself][2] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_440_0");
+            global.actactor[myself][2] = 3;
+        }
     }
     if (global.monstertype[myself] == 23)
     {
@@ -569,9 +628,18 @@
         global.actname[myself][0] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_655_0");
         global.canact[myself][1] = 1;
         global.actname[myself][1] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_657_0");
+        if (scr_havechar(2) && !scr_susiemode())
+        {
-        global.canact[myself][2] = 1;
+            global.canact[myself][2] = 1;
-        global.actname[myself][2] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_659_0");
+            global.actname[myself][2] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_659_0");
-        global.actactor[myself][2] = 2;
+            global.actactor[myself][2] = 2;
+        }
+        if (scr_havechar(2) && scr_susiemode())
+        {
+            global.canact[myself][2] = 1;
+            global.actname[myself][2] = scr_84_get_lang_string("scr_monstersetup_slash_scr_monstersetup_gml_440_0");
+            global.actactor[myself][2] = 3;
+        }
     }
     if (global.monstertype[myself] == 25)
     {
