--- a/Code/gml_GlobalScript_scr_damage.gml
+++ b/Code/gml_GlobalScript_scr_damage.gml
@@ -118,6 +118,17 @@
         {
             gameover = 0;
         }
+        if (instance_exists(obj_checkers_enemy) && gameover == 1)
+        {
+            if (obj_checkers_enemy.susie_solo)
+            {
+                gameover = 0;
+                with (obj_checkers_enemy)
+                {
+                    susie_end = 1;
+                }
+            }
+        }
         if (gameover == 1)
         {
             scr_gameover();
