--- a/Code/gml_Object_obj_battlecontroller_Create_0.gml
+++ b/Code/gml_Object_obj_battlecontroller_Create_0.gml
@@ -1,3 +1,6 @@
+var __cam = view_get_camera(0);
+xx = camera_get_view_x(__cam);
+yy = camera_get_view_y(__cam);
 if (global.flag[9] == 1)
 {
     global.batmusic[1] = mus_loop_ext(global.batmusic[0], 0.7, 1);
@@ -102,7 +105,30 @@
         }
     }
 }
+firstskip = 0;
+scr_susiemode();
 global.charturn = 0;
+if (global.charmove[0] == 0 || global.charauto[global.char[0]] == 1)
+{
+    global.charturn = 1;
+}
+if (global.charturn == 1)
+{
+    if (global.charmove[1] == 0 || global.charauto[global.char[1]] == 1)
+    {
+        global.charturn = 2;
+    }
+}
+if (global.charturn == 2)
+{
+    if (global.charmove[2] == 0 || global.charauto[global.char[2]] == 1)
+    {
+        if (!instance_exists(obj_lancerboss2))
+        {
+            firstskip = 1;
+        }
+    }
+}
 for (i = 0; i < 3; i += 1)
 {
     global.acting[i] = 0;
@@ -193,6 +219,10 @@
         global.charinstance[i].char = 3;
         global.charinstance[i].depth = 200 - (i * 20);
     }
+}
+if (firstskip)
+{
+    scr_attackphase();
 }
 slmxx = 0;
 slmyy = 0;
