--- a/Code/gml_Object_obj_heroparent_Draw_0.gml
+++ b/Code/gml_Object_obj_heroparent_Draw_0.gml
@@ -1,4 +1,13 @@
+var alive;
+if (instance_exists(obj_joker))
+{
+    alive = global.hp[obj_joker.hpchar[myself]] > 0;
+}
+else
+{
-if (global.hp[global.char[myself]] > 0)
+    alive = global.hp[global.char[myself]] > 0;
+}
+if (alive)
 {
     if (global.myfight == 3 && global.faceaction[myself] == 6)
     {
@@ -48,7 +57,7 @@
     if (state == 1 && hurt == 0)
     {
         siner += 1;
-        if (attacked == 0)
+        if (attacked == 0 && attacktimer >= 0)
         {
             snd_stop(snd_laz_c);
             if (object_index == obj_herokris)
@@ -84,16 +93,19 @@
             attacked = 1;
             alarm[1] = 10;
         }
-        if (attacktimer < attackframes)
-        {
-            image_index = attacktimer;
-        }
-        else
-        {
-            image_index = attackframes;
-        }
-        thissprite = attacksprite;
-        index = image_index;
+        if (attacktimer >= 0)
+        {
+            if (attacktimer < attackframes)
+            {
+                image_index = attacktimer;
+            }
+            else
+            {
+                image_index = attackframes;
+            }
+            thissprite = attacksprite;
+            index = image_index;
+        }
         attacktimer += attackspeed;
     }
     if (state == 2 && hurt == 0)
@@ -146,7 +158,8 @@
     }
     if (state == 6)
     {
-        if (global.myfight == 3)
+        var lancercon = instance_exists(obj_lancerboss2) && obj_lancerboss2.bubbleactive;
+        if (global.myfight == 3 && !lancercon)
         {
             if (acttimer < actframes)
             {
