--- a/Code/gml_Object_obj_mainchara_Step_0.gml
+++ b/Code/gml_Object_obj_mainchara_Step_0.gml
@@ -367,6 +367,25 @@
     }
     x += px;
     y += py;
+    moved = 0;
+    if (x != remx[0] || y != remy[0])
+    {
+        moved = 1;
+    }
+    if (moved == 1)
+    {
+        for (var i = 75; i > 0; i -= 1)
+        {
+            remx[i] = remx[i - 1];
+            remy[i] = remy[i - 1];
+            facing[i] = facing[i - 1];
+            wassliding[i] = wassliding[i - 1];
+        }
+        remx[0] = x;
+        remy[0] = y;
+        facing[0] = global.facing;
+        wassliding[0] = sliding;
+    }
 }
 walk = 0;
 if (x != nowx && nopress == 0)
@@ -377,79 +396,90 @@
 {
     walk = 1;
 }
+if (sliding == 1)
+{
+    walk = 1;
+}
 if (walk == 1)
 {
     walkbuffer = 6;
 }
-if (walkbuffer > 3 && fun == 0)
+if (fun == 0 || sliding == 1)
 {
-    walktimer += 1.5;
-    if (runmove == 1)
+    if (walkbuffer > 3)
     {
         walktimer += 1.5;
-    }
-    if (walktimer >= 40)
-    {
-        walktimer -= 40;
-    }
-    if (walktimer < 10)
-    {
+        if (runmove == 1)
+        {
+            walktimer += 1.5;
+        }
+        if (walktimer >= 40)
+        {
+            walktimer -= 40;
+        }
+        if (walktimer < 10)
+        {
+            image_index = 0;
+        }
+        if (walktimer >= 10)
+        {
+            image_index = 1;
+        }
+        if (walktimer >= 20)
+        {
+            image_index = 2;
+        }
+        if (walktimer >= 30)
+        {
+            image_index = 3;
+        }
+    }
+    else if (walkbuffer <= 0)
+    {
+        if (walktimer < 10)
+        {
+            walktimer = 9.5;
+        }
+        if (walktimer >= 10 && walktimer < 20)
+        {
+            walktimer = 19.5;
+        }
+        if (walktimer >= 20 && walktimer < 30)
+        {
+            walktimer = 29.5;
+        }
+        if (walktimer >= 30)
+        {
+            walktimer = 39.5;
+        }
         image_index = 0;
     }
-    if (walktimer >= 10)
-    {
-        image_index = 1;
-    }
-    if (walktimer >= 20)
-    {
-        image_index = 2;
-    }
-    if (walktimer >= 30)
-    {
-        image_index = 3;
-    }
-}
-else if (walkbuffer <= 0 && fun == 0)
-{
-    if (walktimer < 10)
-    {
-        walktimer = 9.5;
-    }
-    if (walktimer >= 10 && walktimer < 20)
-    {
-        walktimer = 19.5;
-    }
-    if (walktimer >= 20 && walktimer < 30)
-    {
-        walktimer = 29.5;
-    }
-    if (walktimer >= 30)
-    {
-        walktimer = 39.5;
-    }
-    image_index = 0;
 }
 walkbuffer -= 0.75;
-if (fun == 0)
+if (fun == 0 && sliding == 0)
 {
     if (global.facing == 0)
     {
         sprite_index = dsprite;
+        visual_info.sprite_index = visual_info.dsprite;
     }
     if (global.facing == 1)
     {
         sprite_index = rsprite;
+        visual_info.sprite_index = visual_info.rsprite;
     }
     if (global.facing == 2)
     {
         sprite_index = usprite;
+        visual_info.sprite_index = visual_info.usprite;
     }
     if (global.facing == 3)
     {
         sprite_index = lsprite;
+        visual_info.sprite_index = visual_info.lsprite;
     }
 }
-if (stepping == 1)
+if (stepping == 1 && sliding == 0)
 {
     if (image_index == 1 && stepped == 0)
     {
