--- a/Code/gml_Object_obj_heroparent_Draw_0.gml
+++ b/Code/gml_Object_obj_heroparent_Draw_0.gml
@@ -1,4 +1,13 @@
+var alive;
+if (instance_exists(obj_joker))
+{
+    alive = global.hp[obj_joker.hpchar[myself]] > 0;
+}
+else
+{
-if (global.hp[global.char[myself]] > 0)
+    alive = global.hp[global.char[myself]] > 0;
+}
+if (alive)
 {
     if (global.myfight == 3 && global.faceaction[myself] == 6)
     {
