--- a/Code/gml_GlobalScript_scr_spell.gml
+++ b/Code/gml_GlobalScript_scr_spell.gml
@@ -43,7 +43,16 @@
                     delay = 8;
                     type = 3;
                 }
+                var full = false;
+                if (instance_exists(obj_joker))
+                {
+                    full = global.hp[obj_joker.hpchar[myself]] >= global.maxhp[obj_joker.hpchar[myself]];
+                }
+                else
+                {
-                if (global.hp[global.char[myself]] >= global.maxhp[global.char[myself]])
+                    full = global.hp[global.char[myself]] >= global.maxhp[global.char[myself]];
+                }
+                if (full)
                 {
                     with (dmgwr)
                     {
@@ -66,7 +75,7 @@
                 {
                     with (global.monsterinstance[star])
                     {
-                        if (global.monstertype[myself] != 19 && global.monstertype[myself] != 3)
+                        if (global.monstertype[myself] != 19 && global.monstertype[myself] != 3 && global.monstertype[myself] != 21 && global.monstertype[myself] != 10)
                         {
                             global.flag[51 + myself] = 3;
                             event_user(10);
@@ -75,7 +84,6 @@
                         else
                         {
                             pacifycon = 1;
-                            global.spelldelay = 999;
                         }
                     }
                 }
@@ -99,11 +107,27 @@
             {
                 global.spelldelay = 70;
                 damage = ceil(((global.battlemag[arg1] * 5) + (global.battleat[arg1] * 11)) - (global.monsterdf[star] * 3));
+                if (global.monstertype[global.chartarget[myself]] == 19)
+                {
+                    damage = ceil(damage * 0.3);
+                }
                 if (global.automiss[star] == 1)
                 {
                     damage = 0;
                 }
-                attack = instance_create(obj_herosusie.x, obj_herosusie.y, obj_rudebuster_anim);
+                with (global.monsterinstance[star])
+                {
+                    if (global.monstertype[myself] == 12)
+                    {
+                        rudebuster = true;
+                    }
+                }
+                attack = instance_create(global.charinstance[caster].x, global.charinstance[caster].y, obj_rudebuster_anim);
+                depth = global.charinstance[caster].depth;
+                with (global.charinstance[caster])
+                {
+                    visible = 0;
+                }
                 attack.damage = damage;
                 attack.star = star;
                 attack.caster = caster;
@@ -121,11 +145,20 @@
             {
                 global.spelldelay = 70;
                 damage = ceil(((global.battlemag[arg1] * 6) + (global.battleat[arg1] * 13)) - (global.monsterdf[star] * 6));
+                if (global.monstertype[global.chartarget[myself]] == 19)
+                {
+                    damage = ceil(damage * 0.3);
+                }
                 if (global.automiss[star] == 1)
                 {
                     damage = 0;
                 }
-                attack = instance_create(obj_herosusie.x, obj_herosusie.y, obj_rudebuster_anim);
+                attack = instance_create(global.charinstance[caster].x, global.charinstance[caster].y, obj_rudebuster_anim);
+                depth = global.charinstance[caster].depth;
+                with (global.charinstance[caster])
+                {
+                    visible = 0;
+                }
                 attack.damage = damage;
                 attack.star = star;
                 attack.caster = caster;
@@ -190,6 +223,13 @@
                 }
                 else
                 {
+                    if (global.monstertype[star] == 12)
+                    {
+                        with (global.monsterinstance[star])
+                        {
+                            sparecon = 1;
+                        }
+                    }
                     scr_mercyadd(star, global.sparepoint[star]);
                     _pspell = instance_create(0, 0, obj_pacifyspell);
                     _pspell.target = global.monsterinstance[star];
