--- a/Code/gml_Object_obj_rudinnranger_Step_0.gml
+++ b/Code/gml_Object_obj_rudinnranger_Step_0.gml
@@ -207,7 +207,7 @@
         scr_battletext_default();
         actcon = 1;
     }
-    if (acting == 3 && actcon == 0)
+    if (acting == 3 && actcon == 0 && !scr_susiemode())
     {
         global.msg[0] = scr_84_get_lang_string("obj_rudinnranger_slash_Step_0_gml_181_0");
         scr_susface(1, 2);
@@ -252,9 +252,24 @@
             complimented = 1;
         }
     }
+    if (acting == 3 && actcon == 0 && scr_susiemode())
+    {
+        actcon = 1;
+        global.msg[0] = "* You and Ralsei warned Rudinn Ranger about Susie.&* The enemy went on guard.../%";
+        if (scr_monsterpop() > 1)
+        {
+            global.msg[0] = scr_84_get_lang_string("obj_diamondenemy_slash_Step_0_gml_190_0");
+        }
+        for (i = 0; i < 3; i += 1)
+        {
+            global.monstercomment[i] = scr_84_get_lang_string("obj_diamondenemy_slash_Step_0_gml_193_0");
+            global.automiss[i] = 1;
+        }
+        scr_battletext_default();
+    }
     if (actcon == 1 && !instance_exists(obj_writer))
     {
-        if (acting == 3)
+        if (acting == 3 && !scr_susiemode())
         {
             scr_mercyadd(myself, 100);
         }
