--- a/Code/gml_Object_obj_caterpillarchara_Create_0.gml
+++ b/Code/gml_Object_obj_caterpillarchara_Create_0.gml
@@ -8,37 +8,47 @@
 }
 walk = 0;
 fun = 0;
+slided = 0;
 runmove = 0;
 dir = 0;
 walkbuffer = 0;
 walktimer = 0;
 blushtimer = 0;
+name = "susie";
 target = 12;
+var alignment = scr_get_chara_alignment(name, self);
+halign = alignment.halign;
+valign = alignment.valign;
 usprite = spr_susieu_dark;
 dsprite = spr_susied_dark;
 rsprite = spr_susier_dark;
 lsprite = spr_susiel_dark;
-if (global.plot >= 240)
-{
-    usprite = spr_susieu_dark;
-    dsprite = spr_susied_dark_eyes;
-    rsprite = spr_susier_dark_eyes;
-    lsprite = spr_susiel_dark_eyes;
-}
+slidesprite = spr_krisd_slide;
 usprite_blush = spr_ralseiu;
 dsprite_blush = spr_ralseid;
 rsprite_blush = spr_ralseir_blush;
 lsprite_blush = spr_ralseil_blush;
+scr_set_chara_sprites(name, self);
 parent = obj_mainchara;
-pd = parent.dsprite;
-pr = parent.rsprite;
-pl = parent.lsprite;
-pu = parent.usprite;
-for (i = 0; i < 25; i += 1)
-{
-    remx[i] = obj_mainchara.x;
-    remy[i] = obj_mainchara.y;
-    facing[i] = global.facing;
+if (!instance_exists(obj_lancerslideevent))
+{
+    for (i = 0; i < 25; i += 1)
+    {
+        remx[i] = parent.remx[i];
+        remy[i] = parent.remy[i];
+        facing[i] = parent.facing[i];
+        sliding[i] = parent.wassliding[i];
+    }
+}
+else
+{
+    for (i = 0; i < 25; i += 1)
+    {
+        remx[i] = x;
+        remy[i] = y;
+        facing[i] = global.facing;
+        sliding[i] = slided;
+    }
 }
 depth = obj_mainchara.depth + 5;
 mywidth = sprite_width;
