--- a/Code/gml_Object_obj_joker_Step_0.gml
+++ b/Code/gml_Object_obj_joker_Step_0.gml
@@ -481,65 +481,94 @@
         {
             global.msg[0] = scr_84_get_lang_string("obj_joker_slash_Step_0_gml_314_0");
             star = choose(0, 1, 2);
+            if (!(global.char[0] == 0 && global.char[1] == 0 && global.char[2] == 0))
+            {
+                while (global.char[star] == 0)
+                {
+                    star = choose(0, 1, 2);
+                }
-            scr_healitemspell(floor(random(31) + 25));
+                scr_healitemspell(floor(random(31) + 25));
+            }
         }
         if (chaosdance == 6)
         {
+            var chartotal = 0;
             global.msg[0] = scr_84_get_lang_string("obj_joker_slash_Step_0_gml_320_0");
-            swap1 = 1;
-            swap2 = 1;
-            swap1 = choose(2, 3);
-            if (swap1 == 2)
-            {
-                swap2 = 3;
-            }
-            if (swap1 == 3)
-            {
-                swap2 = 2;
-            }
-            for (i = 0; i < 4; i += 1)
-            {
-                remhp[i] = global.hp[i];
-                curmaxhp[i] = global.maxhp[i];
-            }
-            global.maxhp[1] = curmaxhp[swap1];
-            global.maxhp[swap1] = curmaxhp[swap2];
-            global.maxhp[swap2] = curmaxhp[1];
-            global.hp[1] = remhp[swap1];
-            global.hp[swap1] = remhp[swap2];
-            global.hp[swap2] = remhp[1];
-            remhpcolor[0] = obj_battlecontroller.hpcolor[0];
-            remhpcolor[1] = obj_battlecontroller.hpcolor[1];
-            remhpcolor[2] = obj_battlecontroller.hpcolor[2];
-            obj_battlecontroller.hpcolor[0] = remhpcolor[swap1 - 1];
-            obj_battlecontroller.hpcolor[swap1 - 1] = remhpcolor[swap2 - 1];
-            obj_battlecontroller.hpcolor[swap2 - 1] = remhpcolor[0];
-            for (i = 1; i <= 3; i += 1)
-            {
-                if (global.hp[i] < 1)
-                {
-                    global.hp[1] += floor(global.hp[i] / 3);
-                    global.hp[2] += floor(global.hp[i] / 3);
-                    global.hp[3] += floor(global.hp[i] / 3);
-                    global.hp[i] = 1;
-                }
-            }
-            if (global.hp[1] < 1)
-            {
-                global.hp[1] = 1;
-            }
-            if (global.hp[2] < 1)
-            {
-                global.hp[2] = 1;
-            }
-            if (global.hp[3] < 1)
-            {
-                global.hp[3] = 1;
-            }
-            scr_revive(0);
-            scr_revive(1);
-            scr_revive(2);
-            snd_play(snd_weirdeffect);
+            for (t = 0; t < 3; t++)
+            {
+                if (global.char[t] > 0)
+                {
+                    chartotal++;
+                }
+            }
+            if (chartotal > 1)
+            {
+                hpshift = 0 + choose(-1, 1);
+                for (i = 0; i < 3; i += 1)
+                {
+                    remhpcolor[i] = hpcolor[i];
+                }
+                for (i = 0; i < 3; i++)
+                {
+                    if (global.char[i] != 0)
+                    {
+                        var otherslot = shufflepos[i] + hpshift;
+                        if (otherslot < 0)
+                        {
+                            otherslot = 2;
+                        }
+                        else if (otherslot > 2)
+                        {
+                            otherslot = 0;
+                        }
+                        if (global.char[otherslot] == 0)
+                        {
+                            otherslot += hpshift;
+                            if (otherslot < 0)
+                            {
+                                otherslot = 2;
+                            }
+                            else if (otherslot > 2)
+                            {
+                                otherslot = 0;
+                            }
+                            if (global.char[otherslot] == 0)
+                            {
+                                otherslot = shufflepos[i];
+                            }
+                        }
+                        shufflepos[i] = otherslot;
+                        hpcolor[i] = remhpcolor[otherslot];
+                        hpchar[i] = global.char[otherslot];
+                    }
+                }
+                for (i = 1; i <= 3; i += 1)
+                {
+                    if (global.hp[i] < 1)
+                    {
+                        global.hp[1] += floor(global.hp[i] / 3);
+                        global.hp[2] += floor(global.hp[i] / 3);
+                        global.hp[3] += floor(global.hp[i] / 3);
+                        global.hp[i] = 1;
+                    }
+                }
+                if (global.hp[1] < 1)
+                {
+                    global.hp[1] = 1;
+                }
+                if (global.hp[2] < 1)
+                {
+                    global.hp[2] = 1;
+                }
+                if (global.hp[3] < 1)
+                {
+                    global.hp[3] = 1;
+                }
+                scr_revive(0);
+                scr_revive(1);
+                scr_revive(2);
+                snd_play(snd_weirdeffect);
+            }
         }
         if (chaosdance == 7)
         {
@@ -573,7 +602,19 @@
         hypnofx = instance_create(global.monsterx[myself], global.monstery[myself], obj_hypnofx);
         snd_play(snd_hypnosis);
         actcon = 1;
+        aaa = 2;
+        if (scr_havechar(2) && scr_havechar(3))
+        {
-        aaa = choose(0, 1, 2);
+            aaa = choose(0, 1, 2);
+        }
+        else if (scr_havechar(2))
+        {
+            aaa = choose(1, 2);
+        }
+        else if (scr_havechar(3))
+        {
+            aaa = choose(0, 2);
+        }
         if (aaa == 0)
         {
             txtpart1 = scr_84_get_lang_string("obj_joker_slash_Step_0_gml_400_0");
