--- a/Code/gml_Object_obj_battlecontroller_Step_0.gml
+++ b/Code/gml_Object_obj_battlecontroller_Step_0.gml
@@ -59,6 +59,10 @@
     }
     with (obj_tensionbar)
     {
+        if (instance_exists(obj_dummyenemy) && obj_dummyenemy.should_die)
+        {
+            continue;
+        }
         alarm[5] = 15;
         hspeed = -10;
         friction = -0.4;
@@ -75,6 +79,43 @@
             instance_destroy();
         }
     }
+    if (instance_exists(obj_lancerbattle2_event))
+    {
+        if (obj_lancerbattle2_event.con == 52)
+        {
+            victoried = 2;
+            with (obj_face_parent)
+            {
+                instance_destroy();
+            }
+            with (lastbattlewriter)
+            {
+                instance_destroy();
+            }
+        }
+    }
+    if (instance_exists(obj_dummyenemy))
+    {
+        if (obj_dummyenemy.should_die)
+        {
+            with (obj_dummyenemy)
+            {
+                susie_kill = 1;
+            }
+            with (lastbattlewriter)
+            {
+                instance_destroy();
+            }
+            for (i = 0; i < 3; i += 1)
+            {
+                with (global.charinstance[i])
+                {
+                    state = 0;
+                }
+            }
+            victoried = 2;
+        }
+    }
 }
 if (victoried == 1)
 {
@@ -189,6 +230,7 @@
     }
     if (global.bmenuno == 2)
     {
+        thischar = global.charturn;
         with (battlewriter)
         {
             skipme = 1;
@@ -205,14 +247,13 @@
         {
             depth = 10;
         }
-        thischar = global.char[global.charturn];
         if (right_p())
         {
             cango = 1;
             spellcoord = global.bmenucoord[2][global.charturn];
             if (spellcoord < 11)
             {
-                if (global.spell[thischar][global.bmenucoord[2][global.charturn] + 1] == 0)
+                if (global.battlespell[thischar][spellcoord + 1] == 0)
                 {
                     cango = 0;
                     if ((spellcoord % 2) == 1 && spellcoord > 0)
@@ -242,7 +283,7 @@
         {
             cango = 1;
             spellcoord = global.bmenucoord[2][global.charturn];
-            if (global.spell[thischar][1] != 0)
+            if (global.battlespell[thischar][1] != 0)
             {
                 if ((spellcoord % 2) == 0)
                 {
@@ -264,11 +305,11 @@
             }
             else
             {
-                if (global.spell[thischar][spellcoord + 2] == 0)
+                if (global.battlespell[thischar][spellcoord + 2] == 0)
                 {
                     cango = 0;
                 }
-                if (spellcoord == 5 && global.spell[thischar][6] != 0 && global.spell[thischar][7] == 0)
+                if (spellcoord == 5 && global.battlespell[thischar][6] != 0 && global.battlespell[thischar][7] == 0)
                 {
                     cango = 2;
                 }
@@ -298,26 +339,33 @@
                 global.bmenucoord[2][global.charturn] -= 2;
             }
         }
-        global.tensionselect = global.spellcost[thischar][global.bmenucoord[2][global.charturn]];
+        global.tensionselect = global.battlespellcost[thischar][global.bmenucoord[2][global.charturn]];
-        if (button1_p() && global.spell[thischar][global.bmenucoord[2][global.charturn]] != 0 && onebuffer < 0)
+        if (button1_p() && global.battlespell[thischar][global.bmenucoord[2][global.charturn]] != 0 && onebuffer < 0)
         {
-            if (global.spellcost[thischar][global.bmenucoord[2][global.charturn]] <= global.tension)
+            if (global.battlespellcost[thischar][global.bmenucoord[2][global.charturn]] <= global.tension)
             {
                 onebuffer = 2;
                 global.bmenuno = 0;
                 selnoise = 1;
-                scr_spellinfo(global.spell[thischar][global.bmenucoord[2][global.charturn]]);
-                if (spelltarget == 0)
-                {
-                    scr_spellconsumeb();
-                }
-                if (spelltarget == 1)
-                {
-                    global.bmenuno = 8;
-                }
-                if (spelltarget == 2)
-                {
-                    global.bmenuno = 3;
+                if (global.battlespell[thischar][global.bmenucoord[2][global.charturn]] == -2)
+                {
+                    scr_actwait();
+                }
+                else
+                {
+                    scr_spellinfo(global.battlespell[global.charturn][global.bmenucoord[2][global.charturn]]);
+                    if (spelltarget == 0)
+                    {
+                        scr_spellconsumeb();
+                    }
+                    if (spelltarget == 1)
+                    {
+                        global.bmenuno = 8;
+                    }
+                    if (spelltarget == 2)
+                    {
+                        global.bmenuno = 3;
+                    }
                 }
             }
         }
@@ -555,20 +603,8 @@
         }
         global.tensionselect = global.actcost[thisenemy][global.bmenucoord[9][global.charturn]];
         canpress = 1;
-        if (global.actactor[global.bmenucoord[11][global.charturn]][global.bmenucoord[9][global.charturn]] == 2 || global.actactor[global.bmenucoord[11][global.charturn]][global.bmenucoord[9][global.charturn]] == 4)
-        {
-            if (havechar[1] == 0 || global.hp[2] <= 0)
-            {
-                canpress = 0;
-            }
-        }
-        if (global.actactor[global.bmenucoord[11][global.charturn]][global.bmenucoord[9][global.charturn]] == 3 || global.actactor[global.bmenucoord[11][global.charturn]][global.bmenucoord[9][global.charturn]] == 4)
-        {
-            if (havechar[2] == 0 || global.hp[3] <= 0)
-            {
-                canpress = 0;
-            }
-        }
+        var actactor = global.actactor[global.bmenucoord[11][global.charturn]][global.bmenucoord[9][global.charturn]];
+        canpress = scr_actactor_available(actactor).available;
         if (canpress == 1)
         {
             if (button1_p() && global.canact[thisenemy][global.bmenucoord[9][global.charturn]] == 1 && global.tension >= global.tensionselect && onebuffer < 0)
@@ -576,29 +612,44 @@
                 onebuffer = 2;
                 global.bmenuno = 0;
                 selnoise = 1;
+                global.actingtarget[global.charturn] = thisenemy;
+                global.actingchoice[global.charturn] = global.bmenucoord[9][global.charturn];
                 global.tension -= global.actcost[thisenemy][global.bmenucoord[9][global.charturn]];
                 global.tensionselect = 0;
-                if (instance_exists(global.monsterinstance[thisenemy]))
-                {
-                    global.monsterinstance[thisenemy].acting = global.bmenucoord[9][global.charturn] + 1;
-                }
-                global.acting[0] = 1;
-                if (global.actactor[global.bmenucoord[11][global.charturn]][global.bmenucoord[9][global.charturn]] == 2)
-                {
-                    global.acting[charpos[1]] = 1;
-                }
-                if (global.actactor[global.bmenucoord[11][global.charturn]][global.bmenucoord[9][global.charturn]] == 3)
-                {
-                    global.acting[charpos[2]] = 1;
-                }
-                if (global.actactor[global.bmenucoord[11][global.charturn]][global.bmenucoord[9][global.charturn]] == 4)
-                {
-                    global.acting[2] = 1;
-                    global.acting[1] = 1;
+                var actindex = global.bmenucoord[9][global.charturn] + 1;
+                if (global.charturn == 0 && instance_exists(global.monsterinstance[thisenemy]))
+                {
+                    global.monsterinstance[thisenemy].acting = actindex;
+                }
+                global.acting[global.charturn] = 1;
+                if (actactor == 2)
+                {
+                    var pos = scr_findchar_available(2);
+                    global.acting[pos] = 2;
+                    if (pos == 0)
+                    {
+                        global.monsterinstance[thisenemy].acting = actindex;
+                    }
+                }
+                if (actactor == 3)
+                {
+                    var pos = scr_findchar_available(3);
+                    global.acting[pos] = 2;
+                    if (pos == 0)
+                    {
+                        global.monsterinstance[thisenemy].acting = actindex;
+                    }
+                }
+                if (actactor == 4 || actactor == 6)
+                {
+                    global.acting[0] = 1;
+                    global.acting[1] = 2;
+                    global.acting[2] = 2;
+                    global.monsterinstance[thisenemy].acting = actindex;
                 }
                 for (i = 0; i < 3; i += 1)
                 {
-                    if (global.acting[i] == 1)
+                    if (global.acting[i] > 0)
                     {
                         global.faceaction[i] = 6;
                         global.charaction[i] = 9;
