--- a/Code/gml_GlobalScript_scr_spell.gml
+++ b/Code/gml_GlobalScript_scr_spell.gml
@@ -43,7 +43,16 @@
                     delay = 8;
                     type = 3;
                 }
+                var full = false;
+                if (instance_exists(obj_joker))
+                {
+                    full = global.hp[obj_joker.hpchar[myself]] >= global.maxhp[obj_joker.hpchar[myself]];
+                }
+                else
+                {
-                if (global.hp[global.char[myself]] >= global.maxhp[global.char[myself]])
+                    full = global.hp[global.char[myself]] >= global.maxhp[global.char[myself]];
+                }
+                if (full)
                 {
                     with (dmgwr)
                     {
@@ -66,7 +75,7 @@
                 {
                     with (global.monsterinstance[star])
                     {
-                        if (global.monstertype[myself] != 19 && global.monstertype[myself] != 3)
+                        if (global.monstertype[myself] != 19 && global.monstertype[myself] != 3 && global.monstertype[myself] != 21 && global.monstertype[myself] != 10)
                         {
                             global.flag[51 + myself] = 3;
                             event_user(10);
@@ -75,7 +84,6 @@
                         else
                         {
                             pacifycon = 1;
-                            global.spelldelay = 999;
                         }
                     }
                 }
@@ -103,6 +111,13 @@
                 {
                     damage = 0;
                 }
+                with (global.monsterinstance[star])
+                {
+                    if (global.monstertype[myself] == 12)
+                    {
+                        rudebuster = true;
+                    }
+                }
                 attack = instance_create(obj_herosusie.x, obj_herosusie.y, obj_rudebuster_anim);
                 attack.damage = damage;
                 attack.star = star;
@@ -190,6 +205,13 @@
                 }
                 else
                 {
+                    if (global.monstertype[star] == 12)
+                    {
+                        with (global.monsterinstance[star])
+                        {
+                            sparecon = 1;
+                        }
+                    }
                     scr_mercyadd(star, global.sparepoint[star]);
                     _pspell = instance_create(0, 0, obj_pacifyspell);
                     _pspell.target = global.monsterinstance[star];
