--- a/Code/gml_Object_o_coaster_hero_Step_0.gml
+++ b/Code/gml_Object_o_coaster_hero_Step_0.gml
@@ -6,32 +6,95 @@
 {
     nitro = 0;
 }
+if (!init)
+{
+    init = true;
+    HeroType = global.char[HeroID];
+    if (HeroType > 0)
+    {
+        heroInstance = global.charinstance[HeroID];
+    }
+    switch (CoasterType)
+    {
+        case 0:
+            default_sprite_front = spr_sneo_car_front;
+            if (HeroType > 0)
+            {
+                default_sprite_back = spr_sneo_car_back;
+            }
+            back_offset_x = 1;
+            back_offset_y = -35;
+            coaster_offset_x = 0;
+            coaster_offset_y = -19;
+            break;
+        case 1:
+            default_sprite_front = spr_kris_coaster_front;
+            hurt_sprite_front = spr_kris_coaster_hurt_front;
+            default_sprite_back = spr_kris_coaster_back;
+            back_offset_x = 0;
+            back_offset_y = -37;
+            coaster_offset_x = 0;
+            coaster_offset_y = -21;
+            break;
+        case 2:
+            default_sprite_front = spr_susie_coaster_front;
+            hurt_sprite_front = spr_susie_coaster_hurt_front;
+            default_sprite_back = spr_susie_coaster_back;
+            back_offset_x = 0;
+            back_offset_y = -35;
+            coaster_offset_x = 0;
+            coaster_offset_y = -21;
+            character_offset_x = -4;
+            break;
+        case 3:
+            default_sprite_front = spr_ralsei_coaster_front;
+            hurt_sprite_front = spr_ralsei_coaster_hurt_front;
+            default_sprite_back = spr_ralsei_coaster_back;
+            back_offset_x = -10;
+            back_offset_y = -43;
+            coaster_offset_x = -10;
+            coaster_offset_y = -24;
+            break;
+        case 4:
+            default_sprite_front = spr_noelle_coaster_front;
+            hurt_sprite_front = spr_noelle_coaster_front;
+            default_sprite_back = -1;
+            character_offset_x = 4;
+            character_offset_y = -10;
+            coaster_offset_x = -17;
+            coaster_offset_y = 35;
+            coaster_offset_x += (2 * sprite_get_width(default_sprite_front));
+            coaster_offset_y -= (2 * sprite_get_height(default_sprite_front));
+            coaster_scale_x = -1;
+            break;
+    }
+    switch (HeroType)
+    {
+        case 0:
+            visible = 0;
+            break;
+        case 1:
+            character_offset_x += 2;
+            character_offset_y += 1;
+            break;
+        case 4:
+            character_offset_x += 4;
+            character_offset_y += -12;
+            break;
+    }
+}
 if (o_coaster_controller.actcon == 1 && o_coaster_controller.timer < o_coaster_controller.timermax && bump == 1)
 {
-    if ((HeroID == 0 && gooffscreen == 0) || (HeroID == 1 && gooffscreen == 0) || (HeroID == 2 && gooffscreen == 0))
+    if (HeroType > 0 && gooffscreen == 0)
     {
         if (visible == 1)
         {
             bump = 0;
             mykeybuffer = 3;
             o_coaster_controller.playerinput = 1;
-            if ((HeroID == 0 && sprite_index != spr_krisb_act) || (HeroID == 0 && sprite_index == spr_krisb_act && image_index > 9))
-            {
-                sprite_index = spr_krisb_act;
-                image_index = 0;
-                forceact = 1;
-                image_speed = 0.5;
-            }
-            if ((HeroID == 1 && sprite_index != spr_susie_act) || (HeroID == 1 && sprite_index == spr_susie_act && image_index > 8))
-            {
-                sprite_index = spr_susie_act;
-                image_index = 0;
-                forceact = 1;
-                image_speed = 0.5;
-            }
-            if ((HeroID == 2 && sprite_index != spr_ralsei_act) || (HeroID == 2 && sprite_index == spr_ralsei_act && image_index > 11))
-            {
-                sprite_index = spr_ralsei_act;
+            if (sprite_index != heroInstance.actsprite || (sprite_index == heroInstance.actsprite && image_index > heroInstance.actframes))
+            {
+                sprite_index = heroInstance.actsprite;
                 image_index = 0;
                 forceact = 1;
                 image_speed = 0.5;
@@ -47,235 +110,151 @@
 {
     actoncondelay = 0;
 }
-if (o_coaster_controller.actcon == 0 && actoncondelay == 0 && endscene == 0)
-{
-    if (HeroID == 0 && forceact == 0)
-    {
-        sprite_index = obj_herokris.sprite_index;
-        image_index = obj_herokris.image_index - 1;
-    }
-    if (HeroID == 1 && forceact == 0 && instance_exists(obj_herosusie))
-    {
-        sprite_index = obj_herosusie.sprite_index;
-        image_index = obj_herosusie.image_index - 1;
-    }
-    if (HeroID == 2 && forceact == 0 && instance_exists(obj_heroralsei))
-    {
-        sprite_index = obj_heroralsei.sprite_index;
-        image_index = obj_heroralsei.image_index - 1;
-    }
-}
-if (sprite_index == spr_krisb_act && image_index > 10)
-{
-    obj_herokris.sprite_index = spr_krisb_idle;
-    image_speed = 0.25;
-    sprite_index = spr_krisb_idle;
-}
-if (sprite_index == spr_susie_act && image_index > 9)
-{
-    sprite_index = spr_susieb_idle;
-    image_speed = 0.25;
-}
-if (sprite_index == spr_susieb_act && image_index > 9)
-{
-    sprite_index = spr_susieb_idle;
-}
-if (sprite_index == spr_ralsei_act && image_index > 11)
-{
-    sprite_index = spr_ralsei_idle;
-    image_speed = 0.25;
-    forceact = 0;
-}
-if (sprite_index == spr_krisb_attack && forceact == 1 && introanim == 1)
-{
-    sprite_index = spr_krisb_idle;
-    image_speed = 0;
-    image_index = 0;
-    forceact = 0;
-    introanim = 0;
-}
-if (sprite_index == spr_susie_attack && forceact == 1 && introanim == 1)
-{
-    sprite_index = spr_susieb_idle;
-    image_speed = 0;
-    image_index = 0;
-    forceact = 0;
-    introanim = 0;
-}
-if (sprite_index == spr_ralsei_battleintro && forceact == 1 && introanim == 1)
-{
-    sprite_index = spr_ralsei_idle;
-    image_speed = 0;
-    image_index = 9;
-    forceact = 0;
-    introanim = 0;
-}
-if (sprite_index == spr_krisb_idle)
-{
-    var xx = 4;
-    var yy = -11;
-}
-else if (sprite_index == spr_krisb_attack)
-{
-    var xx = 0;
-    var yy = 8;
-}
-else if (sprite_index == spr_krisb_attackready)
-{
-    var xx = 0;
-    var yy = 8;
-}
-else if (sprite_index == spr_susie_idle)
-{
-    var xx = 0;
-    var yy = -9;
-}
-else if (sprite_index == spr_susieb_idle)
-{
-    var xx = 0;
-    var yy = -9;
-}
-else if (sprite_index == spr_susieb_itemready)
-{
-    var xx = 0;
-    var yy = -9;
-}
-else if (sprite_index == spr_susieb_item)
-{
-    var xx = 0;
-    var yy = -9;
-}
-else if (sprite_index == spr_ralsei_idle)
-{
-    var xx = -12;
-    var yy = -9;
-}
-else if (sprite_index == spr_ralsei_act)
-{
-    var xx = -12;
-    var yy = 0;
-}
-else if (sprite_index == spr_ralsei_walk_right)
-{
-    var xx = 0;
-    var yy = -16;
-}
-else if (sprite_index == spr_ralsei_right)
-{
-    var xx = 0;
-    var yy = -16;
-}
-else if (sprite_index == spr_krisb_victory)
-{
-    var xx = 0;
-    var yy = -11;
-}
-else if (sprite_index == spr_ralsei_victory)
-{
-    var xx = 0;
-    var yy = -9;
-}
-else if (sprite_index == spr_krisr_dark)
-{
-    var xx = 5;
-    var yy = -10;
-}
-else if (sprite_index == spr_susie_right_dw)
-{
-    var xx = 10;
-    var yy = -13;
-}
-else if (sprite_index == spr_ralsei_walk_right)
-{
-    var xx = 1;
-    var yy = -2;
-}
-else
-{
-    var xx = 0;
-    var yy = 0;
-}
-if (HeroID == 1)
-{
-    var xx = xx - 10;
-}
-var a = 2;
-if (sprite_index == spr_susieb_actready)
-{
-    a = 1.5;
-}
-if (sprite_index == spr_susieb_act)
-{
-    a = 1.5;
-}
-if (sprite_index == spr_susieb_attackready)
-{
-    a = 1.5;
-}
-if (sprite_index == spr_susieb_attack)
-{
-    a = 1.5;
-}
-if (sprite_index == spr_susieb_defend)
-{
-    a = 1.5;
-}
-if (sprite_index == spr_susie_rudebuster)
-{
-    a = 1.5;
-}
-if (sprite_index == spr_susie_spell)
-{
-    a = 1.5;
-}
-if (sprite_index == spr_susie_spellready)
-{
-    a = 1.5;
-}
-if (sprite_index == spr_susie_actready)
-{
-    a = 1.5;
-}
-if (sprite_index == spr_susie_act)
-{
-    a = 1.5;
-}
-if (sprite_index == spr_susie_attackready)
-{
-    a = 1.5;
-}
-if (sprite_index == spr_susie_attack)
-{
-    a = 1.5;
-}
-if (sprite_index == spr_susie_defend)
-{
-    a = 1.5;
+if (instance_exists(heroInstance))
+{
+    if (o_coaster_controller.actcon == 0 && actoncondelay == 0 && endscene == 0)
+    {
+        if (forceact == 0 && instance_exists(heroInstance))
+        {
+            sprite_index = heroInstance.sprite_index;
+            image_index = heroInstance.image_index - 1;
+        }
+    }
+    if (sprite_index == heroInstance.actsprite && image_index > heroInstance.actframes)
+    {
+        heroInstance.sprite_index = heroInstance.idlesprite;
+        image_speed = 0.25;
+        sprite_index = heroInstance.idlesprite;
+        forceact = 0;
+    }
+}
+switch (HeroType)
+{
+    case 1:
+        if (sprite_index == spr_krisb_idle)
+        {
+            anim_offset_x = 4;
+            anim_offset_y = -11;
+        }
+        else if (sprite_index == spr_krisb_attack || sprite_index == spr_krisb_attackready)
+        {
+            anim_offset_x = 0;
+            anim_offset_y = 8;
+        }
+        else if (sprite_index == spr_krisb_victory)
+        {
+            anim_offset_x = 0;
+            anim_offset_y = -11;
+        }
+        else if (sprite_index == spr_krisr_dark)
+        {
+            anim_offset_x = 5;
+            anim_offset_y = -10;
+        }
+        else
+        {
+            anim_offset_x = 0;
+            anim_offset_y = 0;
+        }
+        break;
+    case 2:
+        if (sprite_index == spr_susieb_idle || sprite_index == spr_susieb_item || sprite_index == spr_susieb_itemready)
+        {
+            anim_offset_x = 0;
+            anim_offset_y = -9;
+        }
+        else if (sprite_index == spr_susie_right_dw)
+        {
+            anim_offset_x = 10;
+            anim_offset_y = -13;
+        }
+        else
+        {
+            anim_offset_x = 0;
+            anim_offset_y = 0;
+        }
+        break;
+    case 3:
+        if (sprite_index == spr_ralsei_idle)
+        {
+            anim_offset_x = -12;
+            anim_offset_y = -9;
+        }
+        else if (sprite_index == spr_ralsei_act || sprite_index == spr_ralsei_actready)
+        {
+            anim_offset_x = -12;
+            anim_offset_y = 0;
+        }
+        else if (sprite_index == spr_ralsei_walk_right || sprite_index == spr_ralsei_right)
+        {
+            anim_offset_x = 0;
+            anim_offset_y = -16;
+        }
+        else if (sprite_index == spr_ralsei_victory)
+        {
+            anim_offset_x = 0;
+            anim_offset_y = -9;
+        }
+        else if (sprite_index == spr_ralsei_walk_right)
+        {
+            anim_offset_x = 1;
+            anim_offset_y = -2;
+        }
+        else
+        {
+            anim_offset_x = 0;
+            anim_offset_y = 0;
+        }
+        break;
+}
+if (HeroType == 2)
+{
+    if (sprite_index == spr_susieb_actready || sprite_index == spr_susieb_act || sprite_index == spr_susie_actready || sprite_index == spr_susie_act)
+    {
+        factor_y = 1.5;
+    }
+    else if (sprite_index == spr_susieb_attackready || sprite_index == spr_susieb_attack || sprite_index == spr_susie_attackready || sprite_index == spr_susie_attack)
+    {
+        factor_y = 1.5;
+    }
+    else if (sprite_index == spr_susieb_defend || sprite_index == spr_susie_defend)
+    {
+        factor_y = 1.5;
+    }
+    else if (sprite_index == spr_susie_spell || sprite_index == spr_susie_spellready)
+    {
+        factor_y = 1.5;
+    }
+    else if (sprite_index == spr_susie_rudebuster)
+    {
+        factor_y = 1.5;
+    }
+    else
+    {
+        factor_y = 2;
+    }
 }
 if (endscene == 1 && instance_exists(obj_ch2_scene11a))
 {
-    var xx, yy;
+    var temp_offset_x = 0;
+    var temp_offset_y = 0;
     if (HeroID == 0)
     {
-        xx = obj_ch2_scene11a.coaster_kris.x + obj_ch2_scene11a.coaster_kris.sinerx + obj_ch2_scene11a.coaster_kris.coaster_offset_x;
+        temp_offset_x = obj_ch2_scene11a.coaster_kris.x + obj_ch2_scene11a.coaster_kris.sinerx + obj_ch2_scene11a.coaster_kris.coaster_offset_x;
-        yy = obj_ch2_scene11a.coaster_kris.y + 37 + obj_ch2_scene11a.coaster_kris.sinery + obj_ch2_scene11a.coaster_kris.coaster_offset_y;
     }
     if (HeroID == 1)
     {
-        xx = obj_ch2_scene11a.coaster_susie.x + obj_ch2_scene11a.coaster_susie.sinerx + obj_ch2_scene11a.coaster_susie.coaster_offset_x;
+        temp_offset_x = obj_ch2_scene11a.coaster_susie.x + obj_ch2_scene11a.coaster_susie.sinerx + obj_ch2_scene11a.coaster_susie.coaster_offset_x;
-        yy = obj_ch2_scene11a.coaster_susie.y + 37 + obj_ch2_scene11a.coaster_susie.sinery + obj_ch2_scene11a.coaster_susie.coaster_offset_y;
     }
     if (HeroID == 2)
     {
-        xx = obj_ch2_scene11a.coaster_ralsei.x + 8 + obj_ch2_scene11a.coaster_ralsei.sinerx + obj_ch2_scene11a.coaster_ralsei.coaster_offset_x;
+        temp_offset_x = obj_ch2_scene11a.coaster_ralsei.x + obj_ch2_scene11a.coaster_ralsei.sinerx + obj_ch2_scene11a.coaster_ralsei.coaster_offset_x;
-        yy = obj_ch2_scene11a.coaster_ralsei.y + 44 + obj_ch2_scene11a.coaster_ralsei.sinery + obj_ch2_scene11a.coaster_ralsei.coaster_offset_y;
     }
     if (endscenetimer < 10)
     {
         endscenetimer++;
     }
-    x = lerp(x, xx, endscenetimer / 10);
-    y = lerp(y, yy, endscenetimer / 10);
+    x = lerp(x, temp_offset_x - coaster_offset_x, endscenetimer / 10);
 }
 if (disabled == 1)
 {
@@ -394,11 +373,4 @@
     }
 }
 mykeybuffer -= 1;
-if (gooffscreen == 2)
-{
-    if (HeroID == 0)
-    {
-        x -= 2;
-    }
-}
 
