--- a/Code/gml_Object_o_coaster_hero_Create_0.gml
+++ b/Code/gml_Object_o_coaster_hero_Create_0.gml
@@ -1,6 +1,4 @@
 HeroID = 0;
-xx = 0;
-yy = 0;
 forceact = 1;
 gooffscreen = 0;
 userealsprite = 0;
@@ -34,4 +32,22 @@
 draw_button_press = 0;
 draw_button_press_timer = 0;
 bump = 0;
+HeroType = 0;
+CoasterType = 0;
+heroInstance = 0;
+normalsprite = spr_krisr_dark;
+init = 0;
+default_sprite_front = spr_sneo_car_front;
+default_sprite_back = -1;
+hurt_sprite_front = spr_sneo_car_front;
+back_offset_x = 0;
+back_offset_y = 0;
+coaster_offset_x = 0;
+coaster_offset_y = 0;
+coaster_scale_x = 1;
+character_offset_x = 0;
+character_offset_y = 0;
+anim_offset_x = 0;
+anim_offset_y = 0;
+factor_y = 2;
 
