--- a/Code/gml_GlobalScript_scr_actselect.gml
+++ b/Code/gml_GlobalScript_scr_actselect.gml
@@ -24,7 +24,7 @@
         global.actingsimul[0] = actsimul[arg1];
         global.acting[0] = 1;
         global.actingsingle[0] = 1;
-        global.actingtarget[global.charturn] = arg0;
+        global.actingtarget[0] = arg0;
         if (global.actactor[arg0][arg1] == 2)
         {
             global.acting[charpos[1]] = 1;
@@ -41,6 +41,11 @@
         if (global.actactor[arg0][arg1] == 5)
         {
             global.acting[charpos[3]] = 1;
+        }
+        if (global.actactor[arg0][arg1] == 6)
+        {
+            global.acting[2] = 1;
+            global.acting[1] = 1;
         }
         for (i = 0; i < 3; i += 1)
         {
