--- a/Code/gml_Object_o_coaster_controller_Step_0.gml
+++ b/Code/gml_Object_o_coaster_controller_Step_0.gml
@@ -1,15 +1,3 @@
-if (endcoaster == 0)
-{
-    if (obj_berdlyb_enemy.kriscoaster == 0 && krisgooffscreen == 1)
-    {
-    }
-    if (obj_berdlyb_enemy.suscoaster == 0 && susiegooffscreen == 1)
-    {
-    }
-    if (obj_berdlyb_enemy.ralcoaster == 0 && ralseigooffscreen == 1)
-    {
-    }
-}
 if (endcoaster == 1)
 {
     if (HeroCoaster[0].x < HeroCoaster[0].xstart && HeroCoaster[0].gooffscreen == 1)
